## Introduction

How do gene expression patterns highlight disease etiology and drug mechanisms in genetic studies and functional genomics? To address this question, this paper proposes projecting genetic associations through gene expression patterns to identify therapeutic targets, potential drug repurposing opportunities, and the clustering of complex traits.
Genes function in networks that are context-specific, influencing biological processes relevant to disease.
Understanding how genes regulate expression across different cell types is crucial for improving treatments.
By integrating functional genomics data with GWAS data, researchers can better identify transcriptional mechanisms that, when disrupted, lead to tissue- and cell lineage-specific pathology.
Previous studies have described various regulatory DNA elements and genetic effects on gene expression in different tissues, laying the groundwork for this research.


The research problem addressed in this paper is the need to identify biological processes by leveraging gene expression data from various tissues.
The proposed solution is the use of a transcription-wide association study (TWAS), which integrates expression quantitative trait loci (eQTLs) data to provide a mechanistic interpretation for findings from genetic studies.
TWAS aims to determine if gene regulatory mechanisms mediate the association between genetic variants and human diseases.
These approaches have proven successful in understanding disease etiology at the transcriptome level and in tasks such as disease-risk prediction and drug repurposing.
However, TWAS operates at the individual gene level, which may not capture more complex interactions at the network level.


The omnigenic model suggests that methods must incorporate gene-gene interactions to disentangle disease-relevant mechanisms.
Widespread gene pleiotropy reveals the interconnected nature of transcriptional networks, where all genes expressed in disease-relevant cell types have an effect on the trait.
One approach to learn these interactions is through gene modules: groups of genes with similar expression profiles across conditions.
Unsupervised approaches, such as matrix factorization techniques like ICA/PCA, have shown superior performance in inferring gene-gene connections by extracting gene modules from co-expression patterns.
Integrating genetic studies with gene modules extracted using unsupervised learning could enhance our understanding of disease etiology and progression.


This paper introduces PhenoPLIER, an omnigenic approach that offers a gene module perspective to genetic studies.
The method's flexibility allows for the integration of various data modalities into a unified representation for joint analysis.
This module perspective can help infer how groups of functionally-related genes influence complex traits, identify shared and distinct transcriptomic properties among traits, and predict how pharmacological perturbations impact gene activity to produce their effects.
PhenoPLIER maps gene-trait associations and drug-induced transcriptional responses into a common latent representation by integrating thousands of gene-trait associations (utilizing TWAS from PhenomeXcan) and drug transcriptional profiles (from LINCS L1000) into a low-dimensional space learned from public gene expression data on tens of thousands of RNA-seq samples (recount2).
The latent representation is defined by a matrix factorization approach that extracts gene modules with specific sparsity constraints and preferences aligning with prior knowledge (pathways). 

When mapping gene-trait associations to this reduced expression space, the study observes that diseases significantly correlate with gene modules expressed in relevant cell types.
For example, hypothyroidism is associated with T cells, corneal endothelial cells with keratometry measurements, hematological assays with specific blood cell types, plasma lipids with adipose tissue, and neuropsychiatric disorders with various brain cell types.
Furthermore, significant module-trait associations in PhenomeXcan (the discovery cohort) are replicated in the Electronic Medical Records and Genomics (eMERGE) network phase III (the replication cohort).
Additionally, a CRISPR screen is conducted to analyze lipid regulation in HepG2 cells, revealing more robust trait associations with modules than with individual genes, even when known single genes related to lipid metabolism did not reach genome-wide significance. 

Compared to a single-gene approach, the module-based method better predicts FDA-approved drug-disease links by capturing tissue-specific pathophysiological mechanisms linked with the drugs' mechanism of action.
This improved drug-disease prediction indicates that modules may offer a more effective way to examine drug-disease relationships than individual genes.
Exploring the phenotype-module space uncovers stable trait clusters associated with relevant tissues, including a complex branch involving lipids with cardiovascular, autoimmune, and neuropsychiatric disorders.
In summary, PhenoPLIER incorporates groups of genes that work together in specific cell types to carry out various functions, enhancing the robustness of detecting and interpreting genetic associations.
This approach can prioritize alternative and potentially more promising candidate targets, even when known single gene associations are not detected, representing a conceptual shift in the interpretation of genetic studies.
It has the potential to extract mechanistic insights from statistical associations to improve the understanding of complex diseases and their therapeutic modalities.
