## Introduction

How do genes influence complex traits through gene expression patterns, and how can this understanding be used to identify disease etiology and potential drug mechanisms? Genes function in networks that are context-specific, with variations in these genes impacting disease-relevant biological processes.
To determine the influence of genes on complex traits, it is crucial to mechanistically understand expression regulation across various cell types.
This understanding can lead to improved treatments.
Previous studies have identified regulatory DNA elements and genetic effects on gene expression across different tissues.
Integrating functional genomics data with GWAS data has enhanced the identification of transcriptional mechanisms that, when dysregulated, can lead to tissue- and cell lineage-specific pathology.


How can genetic associations be projected through gene expression patterns to highlight disease etiology and drug mechanisms? This paper proposes a solution by utilizing transcription-wide association studies (TWAS) that integrate expression quantitative trait loci (eQTLs) data to provide a mechanistic interpretation for GWAS findings.
TWAS tests whether perturbations in gene regulatory mechanisms mediate the association between genetic variants and human diseases, leading to a deeper understanding of disease etiology at the transcriptome level and facilitating tasks such as disease-risk prediction and drug repurposing.
Despite the success of TWAS at the individual gene level, it falls short in capturing more complex interactions at the network level.


How do gene-gene interactions impact the architecture of complex traits and disease mechanisms? The omnigenic model suggests that understanding these interactions is crucial for unraveling disease-relevant mechanisms.
Gene pleiotropy highlights the interconnected nature of transcriptional networks, where all genes expressed in disease-relevant cell types may affect the trait.
One approach to uncover these interactions is through gene modules, groups of genes with similar expression profiles across different conditions.
Unsupervised methods, like matrix factorization techniques such as ICA/PCA, have shown promise in extracting gene modules and capturing local expression effects.
Integrating genetic studies with gene modules extracted through unsupervised learning could enhance our comprehension of disease etiology and progression.


This paper introduces PhenoPLIER, an omnigenic approach that offers a gene module perspective to genetic studies.
Our method's flexibility enables the integration of various data modalities into a unified representation for joint analysis.
We demonstrate that this module perspective can deduce how functionally-related gene groups influence complex traits, identify shared and distinct transcriptomic properties among traits, and forecast how pharmacological perturbations impact gene activity to produce their effects.
PhenoPLIER maps gene-trait associations and drug-induced transcriptional responses to a common latent representation.
To achieve this, we combine numerous gene-trait associations (utilizing TWAS from PhenomeXcan) and drug transcriptional profiles (from LINCS L1000) into a low-dimensional space derived from public gene expression data on tens of thousands of RNA-seq samples (recount2).
We employ a latent representation defined by a matrix factorization approach that extracts gene modules with specific sparsity constraints and preferences aligning with prior knowledge (pathways).
When correlating gene-trait associations to this reduced expression space, we observe diseases significantly linked to gene modules expressed in relevant cell types, such as hypothyroidism with T cells, corneal endothelial cells with keratometry measurements, and neuropsychiatric disorders with various brain cell types.
Additionally, PhenoPLIER can identify modules associated with cell types under specific stimuli or disease states using models from large and diverse RNA-seq datasets.
Notably, significant module-trait associations in PhenomeXcan replicated in the eMERGE network phase III.
Furthermore, a CRISPR screen analyzing lipid regulation in HepG2 cells revealed more robust trait associations with modules than individual genes, even when known lipid metabolism genes did not reach genome-wide significance.
Compared to a single-gene approach, our module-based method better predicts FDA-approved drug-disease links by capturing tissue-specific pathophysiological mechanisms related to drug mechanisms.
This enhanced drug-disease prediction indicates that modules may offer a superior approach to investigating drug-disease relationships compared to individual genes.
Exploring the phenotype-module space uncovers stable trait clusters associated with relevant tissues, including a complex branch involving lipids with cardiovascular, autoimmune, and neuropsychiatric disorders.
In conclusion, PhenoPLIER shifts from examining single genes linked to various complex traits to incorporating groups of genes that collaborate to perform different functions in specific cell types.
This approach enhances the robustness of detecting and interpreting genetic associations, prioritizing alternative and potentially more promising candidate targets even when known single gene associations are undetected.
This approach represents a conceptual shift in genetic studies interpretation, offering mechanistic insights from statistical associations to improve the understanding of complex diseases and their therapeutic approaches.
