## Introduction

How do genes influence complex traits and how can this knowledge lead to improved treatments? By understanding the regulation of gene expression across different cell types, we can gain insight into the mechanisms underlying disease etiology and potential therapeutic targets.
Previous studies have identified genetic effects on gene expression and regulatory DNA elements in various tissues.
Integrating functional genomics and GWAS data has enhanced our ability to identify transcriptional mechanisms that, when disrupted, contribute to tissue-specific pathology.
This paper aims to explore the relationships between genetic associations, gene expression patterns, disease etiology, and drug mechanisms to highlight potential therapeutic targets and strategies for drug repurposing.


The research problem addressed in this paper is how to effectively identify biological processes using gene expression data from multiple tissues.
The proposed solution is the transcription-wide association study (TWAS), which integrates expression quantitative trait loci (eQTLs) data to provide a mechanistic interpretation for GWAS findings by testing whether perturbations in gene regulatory mechanisms mediate the association between genetic variants and human diseases.
These approaches have been successful in understanding disease etiology at the transcriptome level and in disease-risk prediction (polygenic scores) and drug repurposing tasks.
However, TWAS operates at the individual gene level, which may not capture more complex interactions at the network level.


The omnigenic model suggests that current methods must incorporate gene-gene interactions to unravel disease-relevant mechanisms.
Gene pleiotropy highlights the interconnected nature of transcriptional networks, where all genes expressed in disease-relevant cell types may impact the trait.
One effective way to uncover these interactions is through gene modules, groups of genes with similar expression profiles across different conditions.
Various unsupervised approaches, such as matrix factorization techniques like ICA/PCA, have been proposed to infer these connections by extracting gene modules from co-expression patterns.
Integrating genetic studies with gene modules extracted using unsupervised learning could enhance our understanding of disease etiology and progression.


This paper introduces PhenoPLIER, an omnigenic approach that offers a gene module perspective to genetic studies.
The method's flexibility enables the integration of various data modalities into a unified representation for joint analysis.
This module perspective can deduce how groups of functionally-related genes influence complex traits, identify shared and distinct transcriptomic properties among traits, and predict how pharmacological perturbations impact gene activity to produce their effects.
PhenoPLIER maps gene-trait associations and drug-induced transcriptional responses into a common latent representation by integrating thousands of gene-trait associations (via TWAS from PhenomeXcan) and drug transcriptional profiles (from LINCS L1000) into a low-dimensional space learned from public gene expression data on tens of thousands of RNA-seq samples (recount2).
A latent representation defined by a matrix factorization approach extracts gene modules with specific sparsity constraints and preferences aligning with prior knowledge (pathways).
The mapping of gene-trait associations to this reduced expression space reveals diseases significantly associated with gene modules expressed in relevant cell types, such as hypothyroidism with T cells, corneal endothelial cells with keratometry measurements, and neuropsychiatric disorders with various brain cell types.
Notably, using models derived from large and diverse RNA-seq datasets, PhenoPLIER identifies modules linked to cell types under specific stimuli or disease states.
The significant module-trait associations observed in PhenomeXcan are replicated in the Electronic Medical Records and Genomics (eMERGE) network phase III.
Additionally, a CRISPR screen analyzing lipid regulation in HepG2 cells demonstrates more robust trait associations with modules than with individual genes, even when known lipid metabolism genes did not reach genome-wide significance.
Compared to single-gene approaches, the module-based method better predicts FDA-approved drug-disease links by capturing tissue-specific pathophysiological mechanisms related to drug mechanisms, such as niacin with cardiovascular traits via an immune mechanism.
This enhanced drug-disease prediction indicates that modules may offer a more effective means of examining drug-disease relationships than individual genes.
Exploring the phenotype-module space reveals stable trait clusters associated with relevant tissues, including a complex branch involving lipids with cardiovascular, autoimmune, and neuropsychiatric disorders.
In summary, PhenoPLIER shifts from considering single genes associated with complex traits to incorporating groups of genes that collaborate to perform various functions in specific cell types.
This approach enhances the robustness of detecting and interpreting genetic associations, prioritizing alternative and potentially more promising candidate targets even when known single gene associations are undetected.
This approach represents a conceptual shift in the interpretation of genetic studies, offering the potential to extract mechanistic insights from statistical associations to enhance the understanding of complex diseases and their therapeutic modalities.
