## Introduction

How do genes influence complex traits through gene expression regulation across different cell types, and how can this understanding lead to improved treatments? Previous studies have highlighted the importance of context-specific gene networks in biological processes and disease etiology.
By integrating functional genomics and GWAS data, this research aims to identify genetic associations that impact gene expression patterns and ultimately highlight potential therapeutic targets and drug repurposing opportunities.
This approach focuses on clustering complex traits to understand disease mechanisms and drug actions more effectively.


The research problem addressed in this paper is the need to identify biological processes using gene expression data and to understand the mechanistic interpretation of genetic associations with human diseases.
The proposed solution is the transcription-wide association study (TWAS), which integrates expression quantitative trait loci (eQTLs) data to investigate how gene regulatory mechanisms mediate the relationship between genetic variants and diseases.
TWAS has been successful in uncovering disease etiology at the transcriptome level, predicting disease risk, and repurposing drugs.
However, TWAS operates at the individual gene level, limiting its ability to capture more complex interactions at the network level.


The architecture of complex traits, such as the omnigenic model, emphasizes the importance of gene-gene interactions in understanding disease mechanisms.
Gene pleiotropy highlights the interconnected nature of transcriptional networks, where all genes expressed in disease-relevant cell types may impact the trait.
One effective approach to uncover these interactions is through gene modules, groups of genes with similar expression profiles across various conditions.
Unsupervised methods, like matrix factorization techniques (e.g., ICA/PCA), excel in identifying gene modules from co-expression patterns.
Integrating genetic studies with gene modules extracted through unsupervised learning can enhance our comprehension of disease etiology and progression.


This paper introduces PhenoPLIER, an omnigenic approach that offers a gene module perspective to genetic studies.
The method's flexibility enables the integration of various data modalities into a unified representation for joint analysis.
This perspective allows for the inference of how functionally-related gene groups influence complex traits, identification of shared and distinct transcriptomic properties among traits, and prediction of how pharmacological perturbations impact gene activity to produce their effects.
PhenoPLIER maps gene-trait associations and drug-induced transcriptional responses to a common latent representation.
This integration involves thousands of gene-trait associations (via TWAS from PhenomeXcan) and drug transcriptional profiles (from LINCS L1000) into a low-dimensional space derived from public gene expression data on tens of thousands of RNA-seq samples (recount2).
The latent representation is defined through a matrix factorization approach that extracts gene modules with specific sparsity constraints and preferences aligned with prior knowledge (pathways).
The mapping of gene-trait associations to this reduced expression space reveals significant associations between diseases and gene modules expressed in relevant cell types.
Notably, associations observed in PhenomeXcan were replicated in the eMERGE network phase III.
Additionally, a CRISPR screen analyzing lipid regulation in HepG2 cells demonstrates more robust trait associations with modules than with individual genes, even when known lipid metabolism genes did not reach genome-wide significance.
The module-based method outperforms single-gene approaches in predicting FDA-approved drug-disease links by capturing tissue-specific pathophysiological mechanisms linked to drug mechanisms.
This improved drug-disease prediction suggests that modules offer a more effective means of examining drug-disease relationships than individual genes.
Overall, PhenoPLIER's exploration of the phenotype-module space reveals stable trait clusters associated with relevant tissues, indicating a conceptual shift in interpreting genetic studies.
This approach enhances the detection and interpretation of genetic associations, prioritizes alternative candidate targets, and provides mechanistic insights to improve the understanding of complex diseases and therapeutic modalities.
