### PhenoPLIER: an integration framework based on gene co-expression patterns

![
**Schematic of the PhenoPLIER framework.**
**a)** High-level schematic of PhenoPLIER (a gene module-based method) in the context of TWAS (single-gene) and GWAS (single-variant).
In GWAS, we identify variants associated with traits.
In TWAS, first, we identify variants that are associated with gene expression levels (eQTLs); then, prediction models based on eQTLs are used to impute gene expression, which is used to compute gene-trait associations.
Resources such as LINCS L1000 provide information about how a drug perturbs gene expression; at the bottom-right corner, we show how a drug downregulates two genes (A and C).
In PhenoPLIER, these data types are integrated using groups of genes co-expressed across one or more conditions (such as cell types) that we call gene modules or latent variables/LVs. Created with BioRender.com.
**b)** The integration process in PhenoPLIER uses low-dimensional representations (matrices $\mathbf{Z}$ and $\mathbf{B}$) learned from large gene expression datasets (top).
We used gene-drug information $\mathbf{L}$ from LINCS L1000 and gene-trait associations $\mathbf{M}$ from TWAS: PhenomeXcan was used as the discovery cohort, and eMERGE as replication (middle).
PhenoPLIER provides three computational components (bottom):
1) an LV-based regression model that associates an LV $j$ ($\mathbf{Z}_j$) with a trait $i$ ($\mathbf{M}_i$),
2) a clustering framework that learns groups of traits from TWAS associations projected into the LV space ($\hat{\mathbf{M}}$),
and 3) an LV-based drug repurposing approach that uses the projection of TWAS ($\hat{\mathbf{M}}$) and LINCS L1000 ($\hat{\mathbf{L}}$) into the LV space.
**c)** Genes that are part of LV603, termed as a neutrophil signature [@doi:10.1016/j.cels.2019.04.003], were expressed in relevant cell types (top), with 53 independent samples expressed in Neutrophils, 59 in Granulocytes, and 20 in Whole blood, 56 in PBMC, 8 in mDCs, 29 in Monocytes, and 5 in Epithelial cells (the boxplot shows the 25th, 50th and 75th percentiles while the whiskers extend to the minimum/maximum values).
LV603 was associated in PhenoPLIER with neutrophil counts and other white blood cells (bottom, showing the top 10 traits for LV603 after projecting gene-trait associations in PhenomeXcan).
eQTLs: expression quantitative trait loci;
MVN: multivariate normal distribution;
PBMC: peripheral blood mononuclear cells;
mDCs: myeloid dendritic cells.
](images/entire_process/entire_process.svg "PhenoPLIER framework"){#fig:entire_process width="100%"}


PhenoPLIER is a computational framework that combines gene-trait and gene-drug associations with gene modules expressed in specific contexts.
The approach uses a latent representation derived from a large gene expression compendium to integrate TWAS with drug-induced transcriptional responses for a joint analysis.
The approach consists of three main components: an LV-based regression model to compute an association between an LV and a trait, a clustering framework to learn groups of traits with shared transcriptomic properties, and an LV-based drug repurposing approach that links diseases to potential treatments.
Extensive simulations were performed for the regression model and clustering framework to ensure proper calibration and expected results under a model of no association.


TWAS results from PhenomeXcan and the eMERGE network were utilized as discovery and replication cohorts, respectively.
PhenomeXcan offers gene-trait associations for 4,091 diseases and traits from various studies, while eMERGE analyzed 309 phecodes.
TWAS results were obtained using Summary-MultiXcan (S-MultiXcan) for regression and clustering, and Summary-PrediXcan (S-PrediXcan) for drug repurposing.
Colocalization results were also considered to assess overlap between GWAS and eQTL signals.
In the drug repurposing analysis, transcriptional responses to small molecule perturbations from LINCS L1000, involving 1,170 compounds, were employed.


The latent gene expression representation was derived from the MultiPLIER models, which utilized a matrix factorization method known as the pathway-level information extractor (PLIER).
The models were applied to recount2, a comprehensive collection of transcript-level gene expression data obtained through RNA-seq from a diverse set of samples representing different disease states, cell types, differentiation stages, and stimuli.
A total of 987 latent variables (LVs) were extracted by the MultiPLIER models, optimizing both data reconstruction and alignment with prior knowledge and pathways.
Further details on the methodology can be found in the Methods section.


Each latent variable (LV) or gene module represents a group of genes expressed together in the same tissues and cell types.
We hypothesize that projecting transcriptome-wide association studies (TWAS) and pharmacologic perturbations data into this latent space can provide a better mechanistic understanding.
PhenoPLIER converts gene associations to an LV score by multiplying standardized effect sizes for a trait or differential expression values for a drug by the LV genes' weights and summing them.
This process connects different traits and drugs to functionally-related groups of genes or LVs.
PhenoPLIER utilizes LV annotations to specify conditions where the genes are expressed, such as cell types, tissues, disease stages, or stimuli.
MultiPLIER's models link LVs to samples annotated for experimental conditions.
For example, LV603 was associated with a neutrophil pathway and correlated with neutrophil count estimates.
Our analysis using PhenoPLIER revealed that neutrophil counts and other white blood cell traits were among the top 10 traits associated with LV603.
Additionally, basophils count and percentage were significantly linked to this LV.
The genes in LV603 were expressed in relevant cell types.
These findings suggest that functionally related and co-expressed genes correspond to trait-associated genes and can link transcriptional mechanisms to complex traits.


Therefore, PhenoPLIER allows users to investigate if disease-associated genes are part of modules expressed in specific tissues and cell types.
These cell type-specific modules may be linked to different diseases, potentially illustrating a concept of "network pleiotropy" from an omnigenic perspective.
Additionally, it can identify a subset of genes within modules that directly impact traits without mediated regulation by other genes, known as "core" genes.
These core genes could serve as alternative and potentially more effective candidate targets.
Furthermore, PhenoPLIER can analyze if drugs disrupt these transcriptional mechanisms and propose potential mechanisms of action.
