### LVs predict drug-disease pairs better than single genes

We assessed the ability of substituting Latent Variables (LVs) for individual genes to predict known treatment-disease relationships using transcriptional responses to small molecule perturbations from LINCS L1000.
These responses were processed and mapped to DrugBank IDs.
By employing a drug repurposing strategy that matches reversed transcriptome patterns between genes and drug-induced perturbations, we utilized imputed transcriptomes from Transcriptome-Wide Association Studies (TWAS) to prioritize drug candidates.
A drug-disease score was computed by calculating the negative dot product between the z-scores for a disease (from TWAS) and the z-scores for a drug (from LINCS) across sets of genes of varying sizes.
A high score for a drug-disease pair indicated that the drug down-regulated the predicted expression values of disease-associated genes, suggesting a potential treatment.
Additionally, we projected the expression profiles of drugs into our latent representation to estimate how pharmacological perturbations affected gene module activity.
The prediction performance was evaluated using a manually-curated gold standard set of drug-disease medical indications for 322 drugs across 53 diseases.


![
**Drug-disease prediction performance for gene-based and LV-based approaches.**
The receiver operating characteristic (ROC) (left) and the precision-recall curves (right) for a gene-based and LV-based approach.
"Random" refers to the average precision of a hundred classifiers with randomly permuted scores, where the error band represents the 95% confidence interval.
AUC: area under the curve; AP: average precision.
](images/drug_disease_prediction/roc_pr_curves.svg "ROC-PR curves for drug-disease prediction"){#fig:drug_disease:roc_pr width="80%"}


It is important to note that the gene-trait associations and drug-induced expression profiles projected into the latent space represent a condensed summary of the results.
Despite this simplification, the LV-based method performed better than the gene-based method, with an area under the curve of 0.632 and an average precision of 0.858 (see Figure 1).
The prediction outcomes indicate that this reduced-dimensional space captures biologically relevant patterns that connect pathophysiological processes with drug mechanisms.


We investigated the relationship between niacin and coronary artery disease (CAD) using gene expression patterns.
Niacin is a B vitamin commonly used to treat lipid disorders, with conflicting evidence on its effectiveness in preventing cardiovascular disease.
The drug affects various tissues, but its mechanisms are not fully understood.
Niacin raises high-density lipoprotein (HDL) levels by inhibiting an HDL catabolism receptor in the liver and also inhibits diacylglycerol acyltransferaseâ€“2 (DGAT2), reducing low-density lipoproteins (LDL) production.
In our analysis, niacin was predicted as a potential treatment for CAD and atherosclerosis (AT) by both gene-based and LV-based approaches.
LV246, associated with plasma lipids and high cholesterol, was identified as a key factor in predicting niacin's therapeutic effects for AT (Figure 1, Table 1).
Additionally, LV246 was significant in predicting other cardiovascular diseases like ischaemic heart disease and high cholesterol.

![
**Top cell types/tissues where LV116's genes are expressed.**
Values in the $y$-axis come from matrix $\mathbf{B}$ in the MultiPLIER models (Figure {@fig:entire_process}b).
In the $x$-axis, cell types/tissues are sorted by the maximum sample value.
The figure shows a clear immune response with cell types under different stimuli.
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP045500 -->
MS: multiple sclerosis;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP045569 -->
<!-- PBMCs: peripheral blood mononuclear cells; -->
HSV: treated with herpes simplex virus;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP015670 -->
WNV: infected with West Nile virus;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP062958 -->
IFNa: treated with interferon-alpha;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP039361 -->
HMDM: human peripheral blood mononuclear cell-derived macrophages;
<!-- IPSDM: human induced pluripotent stem cell-derived macrophages; -->
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP056733 -->
Salm: infected with *Salmonella typhimurium*;
Yers: infected with *Yersinia pseudotuberculosis*;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP062966 -->
ISM: Interferon Signature Metric;
SLE: Systemic lupus erythematosus.
](images/lvs_analysis/lv116/lv116-cell_types.svg "LV116 cell types"){#fig:lv116:cell_types width="100%"}



Analysis of key niacin-contributing LVs in various cardiovascular diseases revealed additional mechanisms of action.
For instance, GPR109A/HCAR2 encodes a high-affinity niacin receptor in adipocytes and immune cells, including monocytes, macrophages, neutrophils, and dendritic cells.
Initially, niacin's antiatherogenic effects were attributed to adipose tissue lipolysis inhibition.
However, it has been shown that nicotinic acid can independently reduce atherosclerosis progression by activating GPR109A in immune cells, enhancing anti-inflammatory processes.
Additionally, flushing, a common niacin side effect, is triggered by GPR109A activation in Langerhans cells.
This alternative niacin mechanism could have been predicted by examining the cell types expressing top-contributing modules, such as LV116 and LV931.
These modules showed a strong immune response signature in monocytes, macrophages, neutrophils, dendritic cells, among others.
LV116's genes are expressed in immune responses to various stimuli, including diarrhea from different pathogens, samples from multiple sclerosis or systemic lupus erythematosus, and infections with various viruses.
LV246, LV116, and LV931 were among the top 20 modules contributing to niacin prediction in different cardiovascular traits.


<!-- niacin:cardiovascular:top_lvs:start DISABLE NOW, BUT HAS TO BE ADDED THE "end" version just below to update table -->
| LV    | Cell type             | Disease                                      |
|:------|:----------------------|:---------------------------------------------|
| LV116 | Immune cells, skin    | Atherosclerosis (ICD10 I70)                  |
|       |                       | Chronic ischaemic heart disease (ICD10 I25)  |
|       |                       | Heart attack, angina, stroke or hypertension |
|       |                       | Ischaemic heart disease (wide definition)    |
| LV931 | Immune cells          | Atherosclerosis (ICD10 I70)                  |
|       |                       | Heart attack, angina, stroke or hypertension |
|       |                       | Ischaemic heart disease (wide definition)    |
| LV246 | Adipose tissue, liver | Atherosclerosis (ICD10 I70)                  |
|       |                       | High cholesterol (self-reported)             |
|       |                       | Ischaemic heart disease (wide definition)    |

Table: **LVs among the top 20 contributors to the prediction of niacin for five cardiovascular diseases.** "Heart attack, angina, stroke or hypertension" refers to the UK Biobank data-field 6150. GWAS sample size: Atherosclerosis (361,194 in total and 566 cases), Chronic ischaemic heart disease (361,194 in total and 12,769 cases), Heart attack, angina, stroke or hypertension (360,420 in total and 253,565 cases), Ischaemic heart disease/wide definition (361,194 in total and 20,857 cases), High cholesterol/self-reported (361,141 in total and 43,957 cases).  {#tbl:niacin:cardio:top_lvs}


In addition to cardiovascular traits, other LVs could provide insights into niacin mechanisms.
For instance, LV66, a top LV influenced by niacin (Figure 1), showed high expression in ovarian granulosa cells.
Recent studies have suggested niacin as a potential therapy for ovarian diseases, as it has been shown to enhance follicle growth and prevent granulosa cell apoptosis in animal models.
