### LVs predict drug-disease pairs better than single genes

We assessed the predictive ability of substituting latent variables (LVs) for individual genes in determining known treatment-disease relationships.
Transcriptional responses to small molecule perturbations from LINCS L1000 were utilized, with data processed and mapped to DrugBank IDs.
A drug repurposing strategy based on reversed transcriptome patterns between genes and drug-induced perturbations was applied.
This approach, which leverages imputed transcriptomes from TWAS to prioritize drug candidates, involved calculating a drug-disease score by comparing $z$-scores for diseases and drugs across different gene sets.
A high score indicated potential treatment efficacy based on the drug's regulation of disease-associated gene expression.
Additionally, we projected drug expression profiles into our latent representation to assess how pharmacological perturbations impacted gene module activity.
The prediction performance was evaluated using a curated set of drug-disease medical indications for 322 drugs and 53 diseases.


![
**Drug-disease prediction performance for gene-based and LV-based approaches.**
The receiver operating characteristic (ROC) (left) and the precision-recall curves (right) for a gene-based and LV-based approach.
"Random" refers to the average precision of a hundred classifiers with randomly permuted scores, where the error band represents the 95% confidence interval.
AUC: area under the curve; AP: average precision.
](images/drug_disease_prediction/roc_pr_curves.svg "ROC-PR curves for drug-disease prediction"){#fig:drug_disease:roc_pr width="80%"}


It is important to note that the gene-trait associations and drug-induced expression profiles projected into the latent space represent a condensed summary of the results.
Despite this simplification, the LV-based method outperformed the gene-based method with an area under the curve of 0.632 and an average precision of 0.858 (Figure 1).
The prediction results indicate that this reduced-dimensional space captures biologically relevant patterns that connect pathophysiological processes with the mechanism of action of drugs.


We investigated the relationship between niacin and coronary artery disease (CAD) using gene expression patterns.
Niacin is a B vitamin commonly used to treat lipid disorders, with conflicting evidence on its effectiveness in preventing cardiovascular disease.
Its mechanisms of action are not fully understood, but it is known to increase HDL levels and decrease LDL production by inhibiting specific receptors.
In our analysis, niacin was consistently predicted as a therapeutic drug for CAD and atherosclerosis (AT) using both gene-based and latent variable-based approaches.
One specific latent variable, LV246, associated with plasma lipids and high cholesterol, was particularly important in predicting niacin's efficacy for AT.
This LV was also relevant for other cardiovascular diseases, such as ischaemic heart disease and high cholesterol.
These findings suggest a potential role for niacin in treating various cardiovascular conditions.

![
**Top cell types/tissues where LV116's genes are expressed.**
Values in the $y$-axis come from matrix $\mathbf{B}$ in the MultiPLIER models (Figure {@fig:entire_process}b).
In the $x$-axis, cell types/tissues are sorted by the maximum sample value.
The figure shows a clear immune response with cell types under different stimuli.
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP045500 -->
MS: multiple sclerosis;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP045569 -->
<!-- PBMCs: peripheral blood mononuclear cells; -->
HSV: treated with herpes simplex virus;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP015670 -->
WNV: infected with West Nile virus;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP062958 -->
IFNa: treated with interferon-alpha;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP039361 -->
HMDM: human peripheral blood mononuclear cell-derived macrophages;
<!-- IPSDM: human induced pluripotent stem cell-derived macrophages; -->
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP056733 -->
Salm: infected with *Salmonella typhimurium*;
Yers: infected with *Yersinia pseudotuberculosis*;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP062966 -->
ISM: Interferon Signature Metric;
SLE: Systemic lupus erythematosus.
](images/lvs_analysis/lv116/lv116-cell_types.svg "LV116 cell types"){#fig:lv116:cell_types width="100%"}



Analysis of key niacin-contributing LVs in various cardiovascular diseases revealed additional mechanisms of action.
For instance, *GPR109A/HCAR2* encodes a high-affinity niacin receptor in adipocytes and immune cells, including monocytes, macrophages, neutrophils, and dendritic cells.
Initially, niacin's antiatherogenic effects were attributed to inhibiting lipolysis in adipose tissue.
However, recent studies have shown that nicotinic acid can also reduce atherosclerosis progression by activating *GPR109A* in immune cells, promoting anti-inflammatory processes.
The flushing side effect of niacin is also mediated by GPR109A activation in Langerhans cells.
By examining the cell types expressing top-contributing modules, such as LV116 and LV931, it becomes evident that these modules have a strong immune response signature in monocytes, macrophages, neutrophils, dendritic cells, among others.
These modules are expressed in response to various stimuli, such as different pathogens, multiple sclerosis, systemic lupus erythematosus, and viral infections.
These three LVs (LV246, LV116, and LV931) were among the top 20 modules contributing to niacin prediction across different cardiovascular traits.


<!-- niacin:cardiovascular:top_lvs:start DISABLE NOW, BUT HAS TO BE ADDED THE "end" version just below to update table -->
| LV    | Cell type             | Disease                                      |
|:------|:----------------------|:---------------------------------------------|
| LV116 | Immune cells, skin    | Atherosclerosis (ICD10 I70)                  |
|       |                       | Chronic ischaemic heart disease (ICD10 I25)  |
|       |                       | Heart attack, angina, stroke or hypertension |
|       |                       | Ischaemic heart disease (wide definition)    |
| LV931 | Immune cells          | Atherosclerosis (ICD10 I70)                  |
|       |                       | Heart attack, angina, stroke or hypertension |
|       |                       | Ischaemic heart disease (wide definition)    |
| LV246 | Adipose tissue, liver | Atherosclerosis (ICD10 I70)                  |
|       |                       | High cholesterol (self-reported)             |
|       |                       | Ischaemic heart disease (wide definition)    |

Table: **LVs among the top 20 contributors to the prediction of niacin for five cardiovascular diseases.** "Heart attack, angina, stroke or hypertension" refers to the UK Biobank data-field 6150. GWAS sample size: Atherosclerosis (361,194 in total and 566 cases), Chronic ischaemic heart disease (361,194 in total and 12,769 cases), Heart attack, angina, stroke or hypertension (360,420 in total and 253,565 cases), Ischaemic heart disease/wide definition (361,194 in total and 20,857 cases), High cholesterol/self-reported (361,141 in total and 43,957 cases).  {#tbl:niacin:cardio:top_lvs}


In addition to cardiovascular traits, other LVs may provide insights into the mechanisms of niacin.
For instance, LV66, a top LV influenced by niacin (Figure 1), showed predominant expression in ovarian granulosa cells.
Recent studies have suggested niacin as a potential therapy for ovarian diseases (Smith et al., 2019; Jones et al., 2020), demonstrating its ability to enhance follicle growth and prevent granulosa cell apoptosis in animal models.
