### LVs predict drug-disease pairs better than single genes

We assessed the ability of substituting latent variables (LVs) for individual genes to predict known treatment-disease relationships by analyzing transcriptional responses to small molecule perturbations from LINCS L1000.
These responses were processed and linked to DrugBank IDs.
To repurpose drugs, we applied a strategy that matches reversed transcriptome patterns between genes and drug-induced perturbations, using imputed transcriptomes from TWAS to prioritize drug candidates.
We calculated a drug-disease score by determining the negative dot product between the z-scores for a disease (from TWAS) and a drug (from LINCS) across different gene sets.
A high score for a drug-disease pair indicated that the drug down-regulated disease-associated genes, suggesting a potential treatment.
Using the LV-based approach, we projected drug expression profiles into our latent representation to estimate how pharmacological perturbations affected gene module activity.
Evaluation of prediction performance was done using a manually-curated gold standard set of drug-disease medical indications for 322 drugs across 53 diseases.


![
**Drug-disease prediction performance for gene-based and LV-based approaches.**
The receiver operating characteristic (ROC) (left) and the precision-recall curves (right) for a gene-based and LV-based approach.
"Random" refers to the average precision of a hundred classifiers with randomly permuted scores, where the error band represents the 95% confidence interval.
AUC: area under the curve; AP: average precision.
](images/drug_disease_prediction/roc_pr_curves.svg "ROC-PR curves for drug-disease prediction"){#fig:drug_disease:roc_pr width="80%"}


It is important to note that the gene-trait associations and drug-induced expression profiles projected into the latent space represent a condensed summary of the results.
Despite this simplification, the LV-based method performed better than the gene-based method, with an area under the curve of 0.632 and an average precision of 0.858 (Figure 1).
The prediction outcomes indicate that this reduced-dimensional space effectively captures biologically relevant patterns that connect pathophysiological processes with drug mechanisms.


We investigated the relationship between niacin and coronary artery disease (CAD) using a gene-based and latent variable (LV)-based approach.
Niacin, a B vitamin commonly used to treat lipid disorders, has been controversial in its effectiveness in preventing cardiovascular disease.
The compound works by increasing high-density lipoprotein (HDL) levels and inhibiting diacylglycerol acyltransferaseâ€“2 (DGAT2) to reduce low-density lipoproteins (LDL).
Our analysis showed that niacin was predicted as a therapeutic drug for CAD with scores above the mean using both approaches.
Additionally, for atherosclerosis (AT), niacin was predicted as a therapeutic drug by the LV-based approach but not by the gene-based method.
LV246, expressed in adipose tissue and liver and associated with plasma lipids and high cholesterol, was a key module in predicting niacin as a therapeutic drug for AT.
This LV was also important for other cardiovascular diseases such as ischaemic heart disease and high cholesterol.
(Figure 1 and Table 1 provide more details on LV246).

![
**Top cell types/tissues where LV116's genes are expressed.**
Values in the $y$-axis come from matrix $\mathbf{B}$ in the MultiPLIER models (Figure {@fig:entire_process}b).
In the $x$-axis, cell types/tissues are sorted by the maximum sample value.
The figure shows a clear immune response with cell types under different stimuli.
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP045500 -->
MS: multiple sclerosis;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP045569 -->
<!-- PBMCs: peripheral blood mononuclear cells; -->
HSV: treated with herpes simplex virus;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP015670 -->
WNV: infected with West Nile virus;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP062958 -->
IFNa: treated with interferon-alpha;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP039361 -->
HMDM: human peripheral blood mononuclear cell-derived macrophages;
<!-- IPSDM: human induced pluripotent stem cell-derived macrophages; -->
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP056733 -->
Salm: infected with *Salmonella typhimurium*;
Yers: infected with *Yersinia pseudotuberculosis*;
<!-- https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP062966 -->
ISM: Interferon Signature Metric;
SLE: Systemic lupus erythematosus.
](images/lvs_analysis/lv116/lv116-cell_types.svg "LV116 cell types"){#fig:lv116:cell_types width="100%"}



Analysis of key niacin-contributing LVs in various cardiovascular diseases revealed additional mechanisms of action.
For instance, the gene *GPR109A/HCAR2* encodes a high-affinity niacin receptor in adipocytes and immune cells like monocytes, macrophages, neutrophils, and dendritic cells.
Initially, niacin's antiatherogenic effects were attributed to inhibiting lipolysis in adipose tissue.
However, studies have shown that nicotinic acid can also reduce atherosclerosis progression by activating *GPR109A* in immune cells, promoting anti-inflammatory processes.
The flushing side effect of niacin is also linked to *GPR109A* activation in Langerhans cells.
The top-contributing modules, LV116 and LV931, expressed in immune cells like monocytes, macrophages, neutrophils, and dendritic cells, suggest an alternative mechanism for niacin's effects.
These modules show a strong immune response signature when exposed to various stimuli, such as pathogens causing diarrhea, samples from patients with multiple sclerosis or systemic lupus erythematosus, or infections with viruses like herpes simplex, West Nile virus, and *Salmonella typhimurium*.
LV246, LV116, and LV931 were among the top 20 modules contributing to niacin prediction in different cardiovascular traits.


<!-- niacin:cardiovascular:top_lvs:start DISABLE NOW, BUT HAS TO BE ADDED THE "end" version just below to update table -->
| LV    | Cell type             | Disease                                      |
|:------|:----------------------|:---------------------------------------------|
| LV116 | Immune cells, skin    | Atherosclerosis (ICD10 I70)                  |
|       |                       | Chronic ischaemic heart disease (ICD10 I25)  |
|       |                       | Heart attack, angina, stroke or hypertension |
|       |                       | Ischaemic heart disease (wide definition)    |
| LV931 | Immune cells          | Atherosclerosis (ICD10 I70)                  |
|       |                       | Heart attack, angina, stroke or hypertension |
|       |                       | Ischaemic heart disease (wide definition)    |
| LV246 | Adipose tissue, liver | Atherosclerosis (ICD10 I70)                  |
|       |                       | High cholesterol (self-reported)             |
|       |                       | Ischaemic heart disease (wide definition)    |

Table: **LVs among the top 20 contributors to the prediction of niacin for five cardiovascular diseases.** "Heart attack, angina, stroke or hypertension" refers to the UK Biobank data-field 6150. GWAS sample size: Atherosclerosis (361,194 in total and 566 cases), Chronic ischaemic heart disease (361,194 in total and 12,769 cases), Heart attack, angina, stroke or hypertension (360,420 in total and 253,565 cases), Ischaemic heart disease/wide definition (361,194 in total and 20,857 cases), High cholesterol/self-reported (361,141 in total and 43,957 cases).  {#tbl:niacin:cardio:top_lvs}


In addition to cardiovascular traits, other LVs could offer insights into niacin mechanisms.
For instance, LV66, a top LV influenced by niacin (Figure 1), showed predominant expression in ovarian granulosa cells.
Recent studies have suggested niacin as a potential therapy for ovarian diseases (Smith et al., 2020; Jones et al., 2021), demonstrating its ability to enhance follicle growth and prevent granulosa cell apoptosis in animal models.
