### LVs reveal trait clusters with shared transcriptomic properties

![
**Cluster analysis on traits using the latent gene expression representation.**
**a)** The projection of TWAS results on 3,752 traits into the latent gene expression representation is the input data to the clustering process.
A linear (PCA) and non-linear (UMAP) dimensionality reduction techniques were applied to the input data, and five different clustering algorithms processed all data versions.
These algorithms derive partitions from the data using different parameters (such as the number of clusters), leading to an ensemble of 4,428 partitions.
Then, a distance matrix is derived by counting how many times a pair of traits was grouped in different clusters across the ensemble.
Finally, a consensus function is applied to the distance matrix to generate consolidated partitions with different numbers of clusters (from 2 to $\sqrt{n}\approx$ 60).
These final solutions were represented in the clustering tree (Figure @fig:clustering:tree).
**b)** The clusters found by the consensus function were used as labels to train a decision tree classifier on the original input data, which detects the LVs that better differentiate groups of traits.
](images/clustering/clustering_design.svg "Cluster analysis on traits"){#fig:clustering:design width="100%"}


Gene-trait associations were projected into the latent space to identify clusters connected by similar transcriptional processes.
To address biases in individual clustering algorithms, a consensus clustering framework was developed to combine solutions from different methods.
Different data versions were processed using various methods and parameters to ensure diversity in the ensemble (Figure 1a).
The resulting 15 final consensus clustering solutions (Figure 2) had varying numbers of clusters (5 to 29) determined by selecting partitions with the highest agreement within the ensemble.
Instead of choosing a single solution with a specific number of clusters, a clustering tree was utilized to identify stable trait groups across different resolutions (Figure 3).
A decision tree classifier was then trained on the input data using the clusters as labels to determine which latent variables distinguished the trait groups (Figure 1b).


![
**Clustering tree using multiple resolutions for clusters of traits.**
Each row represents a partition/grouping of the traits, and each circle is a cluster from that partition.
The number of clusters goes from 5 to 29.
Arrows indicate how traits in one cluster move across clusters from different partitions.
Most of the clusters are preserved across different resolutions, showing highly stable solutions even with independent runs of the clustering algorithm.
<!--  -->
RDW: red cell (erythrocyte) distribution width;
BMI: body mass index;
WC: waist circumference;
HC: hip circumference;
RA: rheumatoid arthritis;
SLE: systemic lupus erythematosus;
HTN: Hypertension;
IBD: inflammatory bowel disease;
SCZ: Schizophrenia;
CAD: Coronary artery disease;
AD: Alzheimer's disease;
<!--  -->
The full lists of traits in each cluster in the last five partitions of the tree (from $k=16$ to $k=29$) are in Supplementary Data 3-7.
](images/clustering/clustering_tree.svg "Clustering tree on groups of traits"){#fig:clustering:tree width="100%"}


Phenotypes were grouped into five branches in our analysis.
The first branch, labeled as "large," encompassed most traits and only subdivided at $k$=16, including asthma, subjective well-being traits, and nutrient intake clusters.
The second branch included heel bone-densitometry measurements, while the third branch consisted of hematological assays on red blood cells, physical measures like spirometry and body impedance, and anthropometric traits with fat-free and fat mass measures in separate sub-branches.
The fourth branch, labeled as "complex," included keratometry measurements, assays on white blood cells and platelets, skin and hair color traits, autoimmune disorders, and cardiovascular diseases, along with other cardiovascular-related traits such as hand-grip strength and environmental/behavioral factors like physical activity and diet.
Results were consistent within branches, with the same traits often clustered together across different resolutions.
Arrows between clusters indicated traits moving between groups, primarily within the "complex" branch and from the "large" branch to the "complex" branch.
This behavior is expected given the shared genetic and environmental factors associated with complex diseases, making them challenging to categorize into a single cluster.
(See Supplementary Data 3-7 for all clustering results).


![
**Cluster-specific and general transcriptional processes associated with different diseases.**
The plot shows a submatrix of $\hat{\mathbf{M}}$ for the main trait clusters at $k$=29, considering only LVs (rows) that are well-aligned with at least one pathway.
](images/clustering/global_clustermap-plain.svg "Heatmap with gene modules and traits"){#fig:clustering:heatmap width="100%"}


We identified the key Latent Variables (LVs) driving clusters of traits by training decision tree classifiers on input data with each cluster at $k$=29 as labels.
This analysis revealed the most discriminative LVs for each cluster.
Some of these LVs aligned with known pathways, while others did not align with prior knowledge but were expressed in relevant tissues.
Specific LVs were associated with certain traits, while others were linked to a broad range of phenotypes, suggesting involvement in general biological functions.
For instance, LV928 and LV30, aligned with early progenitors of the erythrocyte lineage, were predominantly expressed in early erythropoiesis stages and strongly associated with red blood cell assays.
In contrast, LV730 was specific to thrombocytes from cancer samples and associated with platelet assays, while LV598, expressed in corneal endothelial cells, was linked to keratometry measurements.
These associations were statistically significant (FDR < 0.05).


The autoimmune and cardiovascular disease sub-branches converged at $k=10$ (Figure 1), suggesting shared genetic influences.
LV57, predominantly expressed in T cells, showed strong associations with autoimmune disorders, particularly hypothyroidism, in PhenomeXcan (Table 1), with consistent findings in eMERGE (Table 2).
Additionally, LV57 was linked to deep venous thrombosis.
In contrast, LV844 exhibited a more autoimmune-specific pattern, associating with diseases like polymyalgia rheumatica, type 1 diabetes, rheumatoid arthritis, and celiac disease in PhenomeXcan (Table 3).
However, these associations were not replicated in eMERGE.
LV844 showed broad expression across various cell types (Figure 2 and Table 4).


The cardiovascular sub-branch showed 129 significant LV-trait associations in PhenomeXcan and 23 in eMERGE.
LV136, related to collagen formation and muscle contraction pathways, was linked to coronary artery disease and keratometry measurements in PhenomeXcan.
In eMERGE, LV136 was associated with coronary atherosclerosis.
LV136 was expressed in various cell types, including fibroblasts, mesenchymal stem cells, osteoblasts, pancreatic stellate cells, cardiomyocytes, and adipocytes.
Within the cardiovascular sub-branch, neuropsychiatric and neurodevelopmental disorders like Alzheimer's disease, schizophrenia, and ADHD were identified.
These disorders share risk factors such as hypertension, high cholesterol, obesity, and smoking.
However, our results highlighted potentially shared transcriptional processes in specific tissues/cell types.
Alzheimer's disease was significantly associated with LV21 in PhenomeXcan.
LV21 was strongly expressed in soft tissue sarcomas, monocytes/macrophages (including microglia from cortex samples), and aortic valves.
This LV was also associated with lipids, high cholesterol in PhenomeXcan, and hyperlipidemia in eMERGE.
Macrophages play a crucial role in reverse cholesterol transport and atherogenesis, while lipid metabolism in microglia is important in neurodegenerative diseases.
