### LVs reveal trait clusters with shared transcriptomic properties

![
**Cluster analysis on traits using the latent gene expression representation.**
**a)** The projection of TWAS results on 3,752 traits into the latent gene expression representation is the input data to the clustering process.
A linear (PCA) and non-linear (UMAP) dimensionality reduction techniques were applied to the input data, and five different clustering algorithms processed all data versions.
These algorithms derive partitions from the data using different parameters (such as the number of clusters), leading to an ensemble of 4,428 partitions.
Then, a distance matrix is derived by counting how many times a pair of traits was grouped in different clusters across the ensemble.
Finally, a consensus function is applied to the distance matrix to generate consolidated partitions with different numbers of clusters (from 2 to $\sqrt{n}\approx$ 60).
These final solutions were represented in the clustering tree (Figure @fig:clustering:tree).
**b)** The clusters found by the consensus function were used as labels to train a decision tree classifier on the original input data, which detects the LVs that better differentiate groups of traits.
](images/clustering/clustering_design.svg "Cluster analysis on traits"){#fig:clustering:design width="100%"}


The projection of gene-trait associations into the latent space revealed clusters linked by similar transcriptional processes.
To address biases in individual clustering algorithms, we developed a consensus clustering framework that combines solutions from different methods.
Diversity in the ensemble is essential, so we processed various data versions with different methods and parameters.
A consensus function then consolidated the ensemble into a superior solution.
Our clustering pipeline produced 15 final consensus clustering solutions.
The number of clusters (ranging from 5 to 29) was determined by selecting partitions with the highest agreement.
Instead of choosing a specific solution, we utilized a clustering tree to identify stable trait groups across multiple resolutions.
To identify the distinguishing latent variables, we trained a decision tree classifier on the input data using the clusters as labels.


![
**Clustering tree using multiple resolutions for clusters of traits.**
Each row represents a partition/grouping of the traits, and each circle is a cluster from that partition.
The number of clusters goes from 5 to 29.
Arrows indicate how traits in one cluster move across clusters from different partitions.
Most of the clusters are preserved across different resolutions, showing highly stable solutions even with independent runs of the clustering algorithm.
<!--  -->
RDW: red cell (erythrocyte) distribution width;
BMI: body mass index;
WC: waist circumference;
HC: hip circumference;
RA: rheumatoid arthritis;
SLE: systemic lupus erythematosus;
HTN: Hypertension;
IBD: inflammatory bowel disease;
SCZ: Schizophrenia;
CAD: Coronary artery disease;
AD: Alzheimer's disease;
<!--  -->
The full lists of traits in each cluster in the last five partitions of the tree (from $k=16$ to $k=29$) are in Supplementary Data 3-7.
](images/clustering/clustering_tree.svg "Clustering tree on groups of traits"){#fig:clustering:tree width="100%"}


Phenotypes were grouped into five clear branches in our study.
The first branch includes most traits and subdivides starting at $k$=16, including asthma, subjective well-being traits, and nutrient intake clusters.
The second branch includes heel bone-densitometry measurements, the third branch includes hematological assays on red blood cells and physical measures like spirometry and body impedance, and the fourth branch includes keratometry measurements, assays on white blood cells and platelets, skin and hair color traits, autoimmune disorders, and cardiovascular diseases.
Arrows between clusters show traits moving between groups, mainly within the "complex" branch and from the "large" branch to the "complex" branch.
This behavior is expected due to shared genetic and environmental factors in complex diseases.
Results were stable across different resolutions, with the same traits often clustered together.
See Supplementary Data 3-7 for all clustering results.


![
**Cluster-specific and general transcriptional processes associated with different diseases.**
The plot shows a submatrix of $\hat{\mathbf{M}}$ for the main trait clusters at $k$=29, considering only LVs (rows) that are well-aligned with at least one pathway.
](images/clustering/global_clustermap-plain.svg "Heatmap with gene modules and traits"){#fig:clustering:heatmap width="100%"}


We identified the key latent variables (LVs) driving clusters of traits by training decision tree classifiers on the input data with each cluster at $k$=29 serving as labels.
The top discriminative LVs for each cluster were determined through this process.
Some of these LVs aligned with known pathways, while others did not align with prior knowledge but were expressed in relevant tissues.
Certain LVs were highly specific to particular traits, while others were associated with a broad range of phenotypes, suggesting involvement in general biological functions.
Using our regression framework, we assessed the association of these LVs with different traits.
For instance, LVs like LV928 and LV30, aligned with early erythrocyte lineage progenitors, were predominantly expressed in early erythropoiesis differentiation stages and strongly associated with red blood cell assays.
In contrast, LV730, expressed in thrombocytes from various cancer samples, was highly specific and associated with platelet assays.
Similarly, LV598, expressed in corneal endothelial cells, was associated with keratometry measurements.


The autoimmune and cardiovascular disease sub-branches converged at $k=10$ in Figure 1, indicating potential shared genetic influences.
LV57, predominantly expressed in T cells (Figure 2 and Table 1), showed strong associations with autoimmune disorders in PhenomeXcan (Table 2) and replicated associations with hypothyroidism in eMERGE (Table 3).
Additionally, LV57 was linked to deep venous thrombosis in both datasets.
Conversely, LV844 exhibited a more autoimmune-specific profile, with associations to various autoimmune diseases in PhenomeXcan (Table 4), but these associations did not replicate in eMERGE.
LV844 showed broad expression across different cell types (Figure 3 and Table 5), including blood, breast organoids, myeloma cells, lung fibroblasts, and various brain cell types.


In the cardiovascular sub-branch, we identified 129 significant LV-trait associations in PhenomeXcan and 23 in eMERGE.
LV136, related to collagen formation and muscle contraction pathways, was associated with coronary artery disease and keratometry measurements in PhenomeXcan.
In eMERGE, LV136 was linked to coronary atherosclerosis.
This LV showed expression in various cell types, including fibroblasts, mesenchymal stem cells, osteoblasts, pancreatic stellate cells, cardiomyocytes, and adipocytes.
Within this sub-branch, we observed neuropsychiatric disorders like Alzheimer's disease, schizophrenia, and ADHD, which have been previously connected to the cardiovascular system and share risk factors such as hypertension, high cholesterol, obesity, and smoking.
Our results indicated potential shared transcriptional processes among these diseases.
For instance, Alzheimer's disease was significantly associated with LV21 in PhenomeXcan.
LV21, not aligned with known pathways, showed strong expression in soft tissue sarcomas, monocytes/macrophages (including microglia from cortex samples), and aortic valves.
This LV was also strongly linked to lipids and high cholesterol in PhenomeXcan, as well as hyperlipidemia in eMERGE.
Macrophages play a crucial role in reverse cholesterol transport and atherogenesis, while lipid metabolism in microglia is emerging as a key factor in neurodegenerative disease development.
