### LVs reveal trait clusters with shared transcriptomic properties

![
**Cluster analysis on traits using the latent gene expression representation.**
**a)** The projection of TWAS results on 3,752 traits into the latent gene expression representation is the input data to the clustering process.
A linear (PCA) and non-linear (UMAP) dimensionality reduction techniques were applied to the input data, and five different clustering algorithms processed all data versions.
These algorithms derive partitions from the data using different parameters (such as the number of clusters), leading to an ensemble of 4,428 partitions.
Then, a distance matrix is derived by counting how many times a pair of traits was grouped in different clusters across the ensemble.
Finally, a consensus function is applied to the distance matrix to generate consolidated partitions with different numbers of clusters (from 2 to $\sqrt{n}\approx$ 60).
These final solutions were represented in the clustering tree (Figure @fig:clustering:tree).
**b)** The clusters found by the consensus function were used as labels to train a decision tree classifier on the original input data, which detects the LVs that better differentiate groups of traits.
](images/clustering/clustering_design.svg "Cluster analysis on traits"){#fig:clustering:design width="100%"}


The gene-trait associations were projected into a latent space to identify clusters connected by similar transcriptional processes.
To address biases in individual clustering algorithms, a consensus clustering framework was developed to combine trait partitions from different methods.
This ensemble approach, which enhances diversity, involved generating various data versions processed with different methods and parameters.
A consensus function then merged the ensemble into a superior solution.
A total of 15 final consensus clustering solutions were produced, with the number of clusters ranging from 5 to 29.
Rather than selecting a single solution with a specific cluster number, a clustering tree was utilized to explore stable trait groups across different resolutions.
To determine the distinguishing latent variables among trait groups, a decision tree classifier was trained on the input data with the identified clusters as labels.


![
**Clustering tree using multiple resolutions for clusters of traits.**
Each row represents a partition/grouping of the traits, and each circle is a cluster from that partition.
The number of clusters goes from 5 to 29.
Arrows indicate how traits in one cluster move across clusters from different partitions.
Most of the clusters are preserved across different resolutions, showing highly stable solutions even with independent runs of the clustering algorithm.
<!--  -->
RDW: red cell (erythrocyte) distribution width;
BMI: body mass index;
WC: waist circumference;
HC: hip circumference;
RA: rheumatoid arthritis;
SLE: systemic lupus erythematosus;
HTN: Hypertension;
IBD: inflammatory bowel disease;
SCZ: Schizophrenia;
CAD: Coronary artery disease;
AD: Alzheimer's disease;
<!--  -->
The full lists of traits in each cluster in the last five partitions of the tree (from $k=16$ to $k=29$) are in Supplementary Data 3-7.
](images/clustering/clustering_tree.svg "Clustering tree on groups of traits"){#fig:clustering:tree width="100%"}


Phenotypes were grouped into five branches in our study.
The first branch, labeled as "large," encompassed most traits and only subdivided at $k$=16, including asthma, subjective well-being traits, and nutrient intake clusters.
The second branch consisted of heel bone-densitometry measurements.
The third branch included hematological assays on red blood cells, physical measures such as spirometry and body impedance, and anthropometric traits with fat-free and fat mass measures in separate sub-branches.
The fourth branch, labeled as "complex," comprised keratometry measurements, assays on white blood cells and platelets, skin and hair color traits, autoimmune disorders, and cardiovascular diseases, including other cardiovascular-related traits like hand-grip strength and environmental/behavioral factors such as physical activity and diet.
Results were consistent within these branches, with similar traits clustered together across different resolutions.
Arrows between clusters indicated traits moving between groups, mainly within the "complex" branch and from the "large" branch to the "complex" branch.
This behavior is typical for complex diseases due to shared genetic and environmental factors, making them challenging to categorize into a single cluster.
Further details can be found in Supplementary Data 3-7 for all clustering results.


![
**Cluster-specific and general transcriptional processes associated with different diseases.**
The plot shows a submatrix of $\hat{\mathbf{M}}$ for the main trait clusters at $k$=29, considering only LVs (rows) that are well-aligned with at least one pathway.
](images/clustering/global_clustermap-plain.svg "Heatmap with gene modules and traits"){#fig:clustering:heatmap width="100%"}


We identified the key latent variables (LVs) driving clusters of traits by training decision tree classifiers on the input data using each cluster at $k$=29 as labels.
This analysis revealed the top discriminative LVs for each cluster.
Some of these LVs aligned with known pathways, while others were novel but expressed in relevant tissues.
Certain LVs showed specificity to particular traits, while others were associated with a broad range of phenotypes, indicating potential involvement in general biological functions.
Our regression framework confirmed significant associations between specific LVs and traits.
For instance, LVs like LV928 and LV30, linked to early erythrocyte lineage progenitors, were predominantly expressed in early erythropoiesis stages and strongly associated with red blood cell assays.
In contrast, LV730, expressed in thrombocytes from various cancer samples, was associated with platelet assays.
Similarly, LV598, expressed in corneal endothelial cells, was linked to keratometry measurements.


The autoimmune and cardiovascular disease sub-branches converged at $k=10$ (Figure 1), suggesting potential LVs that influence these diseases.
LV57, expressed in T cells (Figure 2 and Table 1), showed strong associations with autoimmune disorders in PhenomeXcan (Table 2) and replicated associations with hypothyroidism in eMERGE (Table 3).
However, LV57 also exhibited significant associations with deep venous thrombosis in both databases.
In contrast, LV844 was more autoimmune-specific, linked to polymyalgia rheumatica, type 1 diabetes, rheumatoid arthritis, and celiac disease in PhenomeXcan (Table 4), but these associations were not replicated in eMERGE.
LV844 was expressed in various cell types, including blood, breast organoids, myeloma cells, lung fibroblasts, and brain cells (Figure 3 and Table 5).


The cardiovascular sub-branch showed 129 significant LV-trait associations in PhenomeXcan and 23 in eMERGE.
LV136, related to collagen formation and muscle contraction pathways, was associated with coronary artery disease and keratometry measurements in PhenomeXcan.
In eMERGE, LV136 was associated with coronary atherosclerosis.
This LV was expressed in various cell types, including fibroblasts, mesenchymal stem cells, osteoblasts, pancreatic stellate cells, cardiomyocytes, and adipocytes.
Within the cardiovascular sub-branch, neuropsychiatric and neurodevelopmental disorders such as Alzheimer's disease, schizophrenia, and ADHD were found to share several risk factors with cardiovascular diseases.
Alzheimer's disease was significantly associated with LV21 in PhenomeXcan, which was strongly expressed in soft tissue sarcomas, monocytes/macrophages (including microglia from cortex samples), and aortic valves.
LV21 was also associated with lipids and high cholesterol in PhenomeXcan and hyperlipidemia in eMERGE.
Macrophages play a crucial role in reverse cholesterol transport and atherogenesis, while lipid metabolism in microglia is important in neurodegenerative diseases.
