## Discussion

This paper presents a novel computational strategy that combines statistical associations from Transcriptome-wide Association Studies (TWAS) with gene modules exhibiting similar expression patterns across various cell types.
The key innovation lies in projecting gene-trait associations through a latent representation derived not only from measures of normal tissue but also from cell types under different stimuli and developmental stages.
This approach enhances interpretation by moving beyond statistical associations to infer cell type-specific characteristics of complex phenotypes.
The method effectively identifies disease-relevant cell types from summary statistics, with several disease-associated gene modules being replicated in eMERGE.
Through a CRISPR screen focused on lipid regulation, the study demonstrates that the gene module-based approach can prioritize causal genes even in the absence of single gene associations.
The findings are interpreted through an omnigenic perspective that distinguishes between "core" and "peripheral" genes, indicating that the approach can pinpoint genes directly influencing the trait without mediated regulation by other genes.
This ability to prioritize alternative and potentially more attractive therapeutic targets highlights the potential impact of this innovative approach in identifying disease etiology and drug mechanisms.


The academic paper 'Projecting genetic associations through gene expression patterns highlights disease etiology and drug mechanisms' explores the use of gene expression patterns to project genetic associations, focusing on functional genomics, gene co-expression, therapeutic targets, drug repurposing, and clustering of complex traits.
The research problem addressed is how to effectively connect diseases, drugs, and cell types using a gene module perspective.
The proposed solution involves integrating drug-induced transcriptional profiles to improve the prediction of drug-disease links.
The study demonstrates that the LV-based drug-repurposing approach outperforms gene-based methods in predicting links between 322 drugs and 53 diseases.
Additionally, the research highlights the application of the approach to cardiovascular traits and the drug niacin, revealing connections between pathophysiological processes and mechanisms of action in adipose tissue, immune cells, and ovarian granulosa cells.
The LV-based approach not only enhances statistical prediction but also aids in generating novel hypotheses for evaluating potential mechanisms of action and adverse effects of known or experimental drugs.


Our study aimed to investigate genetic associations through gene expression patterns to highlight disease etiology and drug mechanisms.
We utilized a cluster analysis approach to identify latent variables (LVs) that grouped diseases and traits based on shared and distinct transcriptional mechanisms in relevant tissues.
By analyzing LVs, we were able to identify novel disease-relevant mechanisms, such as LV57, which were not aligned with known pathways.
Some LVs were associated with specific cell types, providing insights into potentially causal cell types for certain phenotypes.
For example, LV136 was associated with cardiovascular disease and corneal biomechanics and expressed in various cell types like fibroblasts, osteoblasts, and cardiomyocytes.
Additionally, LV844 was associated with autoimmune diseases and expressed in whole blood samples.
Our findings suggest that projecting genetic associations through gene expression patterns learned from different datasets is a promising approach to uncovering cell type and pathway effects on complex phenotypes.


We demonstrated that clustering trees, originally introduced to examine developmental processes in single-cell data, offer a multi-resolution grouping of phenotypes based on latent variable associations.
We utilized hard-partitioning algorithms, where each trait belongs exclusively to one cluster, and the distance between two traits considers all gene modules.
It is possible for two complex diseases to share only a few biological processes rather than being similar across most of them.
Additionally, our TWAS results were obtained from a large set of GWAS with varying sample sizes and qualities.
While we addressed potential issues arising from data heterogeneity before conducting cluster analyses on traits, data preprocessing steps are always challenging and may not completely eliminate bias.
Previous research has shown that considering groups of related diseases is more effective in detecting shared genetic etiology, and clustering trees offer a method to explore such relationships in the context of latent variables.


We developed an LV-based regression framework to determine if gene modules are linked to a trait using TWAS $p$-values.
PhenomeXcan served as the discovery cohort for over four thousand traits, with many LV-trait associations replicating in eMERGE.
In PhenomeXcan, we identified 3,450 significant LV-trait associations (FDR < 0.05), with 686 LVs (out of 987) linked to at least one trait and 1,176 traits associated with at least one LV.
In eMERGE, we discovered 196 significant LV-trait associations, with 116 LVs associated with at least one trait/phecode and 81 traits linked to at least one LV.
While we focused on specific disease types within our trait clusters, the full set of associations across other disease domains can be found in our [Github repository](https://github.com/greenelab/phenoplier) for future research.
It is important to note, as outlined in [Methods](#sec:methods:reg), that one limitation of the regression approach is the approximate accuracy of gene-gene correlations, which could potentially result in false positives if the correlation among the top genes in a module is not precisely captured.
Nevertheless, the regression model is reasonably well-calibrated, and no inflation was observed when applying the method to real data.


Our research aims to address the question of how gene modules with coordinated expression patterns may exhibit coordinated pathological effects.
To achieve this, we have integrated two complementary approaches in our study.
The first approach involves using MultiPLIER, a tool that extracts latent variables from large expression datasets.
These latent variables can represent either real transcriptional processes or technical factors known as "batch effects." However, the model derived from recount2, which we utilized, was originally designed for rare disorders and may not be the optimal representation for the wide range of complex diseases we are considering.
Additionally, the linear combination of variables in the factorization method used by recount2 may overlook important and more complex co-expression patterns.
Despite these limitations, it is worth noting that our models do not impose many assumptions on the latent expression representation, allowing for potential substitution of MultiPLIER with similar approaches like GenomicSuperSignature. 

The second approach we employed in our study is Transcriptome-Wide Association Study (TWAS), where we focus solely on the hypothesis that GWAS loci influence traits through changes in gene expression.
It is important to note that TWAS may not capture other effects, such as coding variants disrupting protein-protein interactions.
Furthermore, TWAS has limitations that can result in false positives.
Similar to GWAS, which typically identifies groups of associated variants in linkage disequilibrium (LD), TWAS often identifies multiple genes within the same locus.
This can be attributed to the sharing of GWAS variants in gene expression models, correlated expression of neighboring genes, or the correlation of predicted expression due to eQTLs in LD, among other factors.
Nevertheless, our LV-based regression framework effectively addresses these gene-gene correlations in TWAS.


Our research paper, titled 'Projecting genetic associations through gene expression patterns highlights disease etiology and drug mechanisms,' aims to address the question of how genetic studies and functional genomics can be used to identify therapeutic targets and potentially repurpose drugs.
Previous studies have shown that drugs supported by genetic evidence are more likely to succeed in the drug development pipeline [@doi:10.1038/ng.3314; @doi:10.1038/nn.4618].
Our proposed solution involves projecting association results through latent variables to prioritize disease-treatment pairs, rather than focusing solely on single-gene effects.
This approach offers the advantage of identifying interpretable genetic features that can shed light on potential mechanisms of action.
By prioritizing drugs for diseases with diverse tissue etiologies, we aim to select the most appropriate tissue model from transcriptome-wide association studies (TWAS) to uncover reversed transcriptome patterns between genes and drug-induced perturbations.


The research problem addressed in this paper is how to project genetic associations through gene expression patterns to highlight disease etiology and drug mechanisms.
The proposed solution involves using a representation derived from a factorization of bulk RNA-seq data.
To improve the quality and interpretability of these representations, detailed perturbation datasets and single-cell profiling of tissues are necessary, along with the annotation of sample metadata.
New approaches for inferring and annotating structured metadata show promise and can be directly applied to existing data.
These advancements pave the way for latent variable projections to be widely utilized in unraveling the genetic basis of complex human phenotypes.
By offering a new perspective on mechanistic understanding of statistical associations from TWAS, our method can generate testable hypotheses for the post-GWAS functional characterization of complex diseases, which is expected to be of great importance in the future.
