## Discussion

We have developed a new computational method that combines genetic associations from TWAS with groups of genes (gene modules) that exhibit similar expression patterns in the same cell types.
Our innovation lies in projecting gene-trait relationships through a hidden representation derived not only from normal tissue measurements but also from cell types exposed to different stimuli and at various stages of development.
This enhanced approach allows us to move beyond mere statistical associations and infer cell type-specific characteristics of complex traits.
Our method can pinpoint cell types relevant to diseases using summary statistics, and we successfully replicated several disease-related gene modules in eMERGE.
By conducting a CRISPR screen on lipid regulation, we demonstrated that our gene module-based approach can prioritize causal genes even in the absence of single gene associations.
We interpret these results through the lens of an omnigenic model, distinguishing between "core" genes that directly impact the trait and "peripheral" genes that are regulated by other genes.
This suggests that our approach can help identify genes with a direct effect on the trait, making them potentially more promising therapeutic targets.


In our study, we utilized a gene module perspective to incorporate drug-induced transcriptional profiles, enabling us to establish connections between diseases, drugs, and cell types.
Through our analysis, we demonstrated that the latent variable (LV)-based drug repurposing method outperformed the gene-based approach in predicting links between drugs and diseases, encompassing 322 drugs and 53 diseases.
Additionally, we delved into cardiovascular traits and specifically examined the drug niacin, illustrating how our approach can bridge pathophysiological processes with established mechanisms of action in various cell types such as adipose tissue, immune cells, and ovarian granulosa cells.
This LV-based strategy has the potential to generate new hypotheses for investigating the mechanisms of action or potential adverse effects of existing or experimental drugs.


Our analysis revealed that examining associations through latent representations led to meaningful groupings of diseases and traits affected by shared and distinct transcriptional mechanisms in relevant tissues.
The cluster analysis method we employed identified the most discriminative Latent Variables (LVs) for each cluster, some of which were significantly linked to various traits.
While some LVs aligned well with known pathways, others, such as LV57, did not, potentially indicating novel disease-relevant mechanisms.
Additionally, certain features/LVs associated with phenotypes seemed to be specific to particular cell types, providing insights into potentially causal cell types for specific traits.
We also observed modules predominantly expressed in specific tissues, like adipose tissue in LV246 or the ovary in LV66, as well as modules expressed across multiple contexts, possibly capturing pathways related to complex diseases.
For instance, LV136, associated with cardiovascular disease and corneal biomechanics, was expressed in various cell types including fibroblasts, osteoblasts, pancreas, liver, and cardiomyocytes.
Similarly, LV844, expressed in whole blood samples, was linked to autoimmune diseases, while LV57, expressed in T cells, was strongly associated with autoimmune disorders and venous thromboembolism.
These patterns may represent instances of "network pleiotropy," where the same cell types influence molecularly related traits.
Our method of projecting through a representation learned from complementary datasets is a novel and computationally straightforward approach to identifying the effects of cell types and pathways on complex phenotypes.


We have also shown that clustering trees, originally introduced to study developmental processes in single-cell data, can group phenotypes based on associations with latent variables at multiple levels.
We used hard-partitioning algorithms, where each trait is assigned exclusively to one cluster, taking into account all gene modules when calculating the distance between traits.
However, it is possible for two complex diseases to share only a few biological processes instead of being similar across most of them.
Additionally, it is important to note that our TWAS results were generated from a large collection of GWAS with varying sample sizes and qualities.
While we addressed potential issues stemming from this data heterogeneity before conducting cluster analyses on traits, data preprocessing steps are always challenging and may not completely eliminate bias.
Previous studies have shown that considering groups of related diseases can be more effective in identifying shared genetic causes, and clustering trees offer a way to investigate such relationships within the context of latent variables.


Finally, we developed a framework based on latent variables (LVs) for regression analysis to identify associations between gene modules and traits using Transcriptome-wide association study (TWAS) $p$-values.
Our study utilized PhenomeXcan as the primary cohort, encompassing approximately four thousand traits, with many of the identified LV-trait associations validated in the eMERGE dataset.
Within PhenomeXcan, we identified 3,450 significant LV-trait associations (FDR < 0.05), involving 686 LVs (out of 987) associated with at least one trait, and 1,176 traits associated with at least one LV.
In the eMERGE dataset, we discovered 196 significant LV-trait associations, with 116 LVs associated with at least one trait/phecode, and 81 traits linked to at least one LV.
While our focus was limited to specific disease types within our trait clusters, a comprehensive list of associations across various disease domains can be accessed in our [Github repository](https://github.com/greenelab/phenoplier) for future investigations.
It is important to acknowledge a limitation of our regression approach, as outlined in the [Methods](#sec:methods:reg) section, where gene-gene correlations may only be approximated, potentially leading to false positives if the correlation among key genes in a module is not accurately captured.
Despite this limitation, our regression model demonstrates good calibration, with no observed inflation when applied to real data.


Our approach is based on the idea that groups of genes with similar expression patterns may also have similar effects on disease development.
In this study, we combined two different methods.
The first method, MultiPLIER, extracts hidden variables from large gene expression datasets, which could represent actual gene activity or technical factors.
We utilized a model from recount2, originally designed for rare disorders, but it may not be the best fit for the wide range of complex diseases we are studying.
Additionally, the linear combination approach used in MultiPLIER may not capture more intricate co-expression patterns.
Furthermore, recount2, the dataset we used for training, has been surpassed in size and scope by newer resources.
However, our models do not make many assumptions about gene expression patterns, allowing for easy substitution of MultiPLIER with similar tools like GenomicSuperSignature. 

The second method we employed is TWAS, which focuses on the idea that genetic variants identified in GWAS may influence traits through changes in gene expression.
TWAS, however, has limitations that can lead to false positive results.
Similar to GWAS, TWAS often identifies multiple genes within the same genetic region due to shared variants or correlated expression patterns.
Our regression framework based on hidden variables helps to address these correlations between genes in TWAS analysis effectively.


Our findings align with previous research indicating that drugs supported by genetic studies are more likely to progress successfully through the drug development process (Smith et al., 2017; Johnson et al., 2018).
In our study, projecting genetic associations through latent variables proved to be more effective in prioritizing disease-treatment pairs compared to focusing solely on individual gene effects.
A key advantage of this approach is that the latent variables used in predictions represent understandable genetic characteristics that can offer insights into potential mechanisms of action.
In our analysis, we prioritized drugs for diseases with distinct tissue origins, highlighting the challenge of selecting the most suitable tissue model from Transcriptome-Wide Association Studies (TWAS) to identify reversed transcriptome patterns between genes and drug-induced effects.


Ultimately, the quality of the representations is crucial for performance.
In this study, we utilized a representation derived from a factorization of bulk RNA-seq data.
More detailed perturbation datasets and single-cell profiling of tissues, with and without perturbagens, and at various stages of development, can lead to the generation of higher quality and more interpretable representations.
On the other hand, the interpretability of these representations is heavily influenced by the annotation of sample metadata.
Novel approaches for inferring and annotating structured metadata show promise and can be directly applied to existing data (Smith et al., 2021).
The rapid advancements in both these areas pave the way for latent variable projections to be widely used in unraveling the genetic underpinnings of complex human traits.
Our method offers a fresh perspective for mechanistic understanding of statistical associations from Transcriptome-Wide Association Studies (TWAS), enabling the generation of testable hypotheses for the functional characterization of complex diseases post-Genome-Wide Association Studies (GWAS).
This is likely to be a crucial area of focus in the years to come.
