## Discussion

Our research paper 'Projecting genetic associations through gene expression patterns highlights disease etiology and drug mechanisms' addresses the research problem of integrating genetic studies with functional genomics to identify potential therapeutic targets and drug repurposing opportunities.
The proposed solution involves a novel computational strategy that combines statistical associations from transcriptome-wide association studies (TWAS) with gene modules exhibiting similar expression patterns across different cell types.
By projecting gene-trait associations through a latent representation derived from various cell types under different stimuli and developmental stages, our approach enhances the interpretation of complex phenotypes by revealing cell type-specific features.
Through our method, disease-relevant cell types can be identified from summary statistics, with several disease-associated gene modules successfully replicated in the eMERGE dataset.
Furthermore, our study demonstrates the utility of our gene module-based approach in prioritizing causal genes, especially in cases where single gene associations are not evident.
This was illustrated through a CRISPR screen focusing on lipid regulation.
Our findings support an omnigenic perspective, distinguishing between "core" genes that directly influence traits and "peripheral" genes that are regulated by core genes.
This distinction allows us to prioritize alternative and potentially more effective therapeutic targets.


This paper titled 'Projecting genetic associations through gene expression patterns highlights disease etiology and drug mechanisms' explores the integration of genetic studies, functional genomics, gene co-expression, therapeutic targets, drug repurposing, and clustering of complex traits.
The research question addressed is how to effectively connect diseases, drugs, and cell types using a gene module perspective.
The proposed solution involves integrating drug-induced transcriptional profiles to predict drug-disease links.
Results indicate that the LV-based drug-repurposing approach outperformed the gene-based method for 322 drugs across 53 diseases.
Additionally, focusing on cardiovascular traits and the drug niacin, the study demonstrates how this approach can connect pathophysiological processes with known mechanisms of action in various cell types.
This LV-based approach not only offers superior predictive capabilities but also aids in generating novel hypotheses for evaluating potential drug mechanisms or adverse effects.


Our research aimed to project genetic associations through gene expression patterns to highlight disease etiology and drug mechanisms.
We discovered that analyzing associations through latent representations led to reasonable groupings of diseases and traits affected by shared and distinct transcriptional mechanisms in relevant tissues.
Our cluster analysis approach identified the most discriminative latent variables (LVs) for each cluster, with some LVs showing significant associations with different traits.
While some LVs aligned with known pathways, others, such as LV57, did not, potentially indicating novel disease-relevant mechanisms.
Associations with cell type marker genes could pinpoint causal cell types for specific phenotypes more accurately.
We observed modules primarily expressed in specific tissues, like adipose in LV246 or ovary in LV66, as well as those expressed across multiple contexts, capturing pathways associated with related complex diseases.
For instance, LV136, associated with cardiovascular disease and corneal biomechanics, was expressed in various cell types including fibroblasts, osteoblasts, pancreas, liver, and cardiomyocytes.
Other examples include LV844, expressed in whole blood samples and linked to autoimmune diseases, and LV57, expressed in T cells and strongly associated with autoimmune conditions and venous thromboembolism.
This novel approach of projecting through learned representations from distinct datasets can help identify cell type and pathway effects on complex phenotypes in a computationally simple manner.


Our study aimed to investigate how clustering trees, originally used to analyze single-cell data for developmental processes, can group phenotypes based on associations with latent variables at multiple levels of resolution.
We utilized hard-partitioning algorithms to assign traits exclusively to one cluster, taking into account all gene modules when measuring the distance between traits.
It is possible for two complex diseases to share only a few biological processes rather than being similar across most of them.
Additionally, our TWAS results were obtained from a diverse set of GWAS with varying sample sizes and qualities.
While we addressed potential issues stemming from this data heterogeneity before conducting cluster analyses on traits, data preprocessing steps present challenges and may not eliminate bias entirely.
Previous research has shown that considering groups of related diseases can enhance the detection of shared genetic etiology, and clustering trees offer a method to explore these relationships in the context of latent variables.


We developed an LV-based regression framework to determine if gene modules are linked to a trait using TWAS $p$-values.
PhenomeXcan served as our discovery cohort, encompassing four thousand traits, with many LV-trait associations replicating in eMERGE.
In PhenomeXcan, we identified 3,450 significant LV-trait associations (FDR < 0.05), with 686 LVs (out of 987) associated with at least one trait and 1,176 traits linked to at least one LV.
In eMERGE, we discovered 196 significant LV-trait associations, with 116 LVs associated with at least one trait/phecode and 81 traits connected to at least one LV.
While we focused on specific disease types within our trait clusters, the full array of associations across other disease domains is accessible in our [Github repository](https://github.com/greenelab/phenoplier) for future investigations.
A limitation of the regression approach, as outlined in [Methods](#sec:methods:reg), is that gene-gene correlations may only be approximately accurate, potentially leading to false positives if the correlation among the top genes in a module is not precisely captured.
Nevertheless, the regression model is reasonably well-calibrated, and no inflation was observed when applying the method to real data.


Our research aims to address the question of whether gene modules with coordinated expression patterns also exhibit coordinated pathological effects.
To achieve this, we propose an integrated approach that combines two methods.
The first method, MultiPLIER, extracts latent variables from large expression datasets, potentially representing real transcriptional processes or technical factors.
However, the model derived from recount2, which we used in this study, may not be the optimal latent representation for the diverse range of complex diseases we are considering.
Moreover, the underlying factorization method relies on linear combinations of variables, potentially missing important and more complex co-expression patterns.
Despite these limitations, our models make minimal assumptions about the latent expression representation, allowing for potential substitution of MultiPLIER with similar approaches such as GenomicSuperSignature. 

The second approach we utilized is TWAS, where we specifically explore the hypothesis that GWAS loci impact traits through changes in gene expression.
It is important to note that TWAS may not capture other effects, such as coding variants disrupting protein-protein interactions.
Additionally, TWAS has limitations that can result in false positives.
Similar to GWAS, which typically identifies groups of associated variants in linkage disequilibrium, TWAS often identifies multiple genes within the same locus due to shared GWAS variants in gene expression models, correlated expression of nearby genes, or predicted expression correlation due to eQTLs in LD.
However, our LV-based regression framework effectively addresses these gene-gene correlations in TWAS.


Our research paper, titled 'Projecting genetic associations through gene expression patterns highlights disease etiology and drug mechanisms,' addresses the question of how to effectively prioritize disease-treatment pairs using genetic studies and functional genomics.
Previous studies have shown that drugs with genetic support are more likely to succeed in the drug development pipeline.
Our proposed solution involves projecting association results through latent variables, which better prioritize disease-treatment pairs compared to single-gene effects alone.
An added benefit is that the latent variables driving predictions represent interpretable genetic features that can help infer potential mechanisms of action.
In our study, we prioritized drugs for diseases with varying tissue etiologies, and a key challenge is selecting the most appropriate tissue model from TWAS to identify reversed transcriptome patterns between genes and drug-induced perturbations.


The research problem addressed in this paper is how to project genetic associations through gene expression patterns to highlight disease etiology and drug mechanisms.
The proposed solution involves using a representation derived from a factorization of bulk RNA-seq data.
To improve the quality and interpretability of these representations, detailed perturbation datasets and single-cell profiling of tissues, with and without perturbagens, at various developmental stages are recommended.
Additionally, the annotation of sample metadata is crucial for interpretability.
New approaches for inferring and annotating structured metadata show promise and can be directly applied to existing data.
These advancements pave the way for latent variable projections to be widely used in unraveling the genetic basis of complex human phenotypes.
By offering a new perspective on mechanistic understanding of statistical associations from TWAS, our method can generate testable hypotheses for post-GWAS functional characterization of complex diseases, which is expected to be a critical area of research in the future.
