## Discussion

Our research paper, titled 'Projecting genetic associations through gene expression patterns highlights disease etiology and drug mechanisms,' addresses the question of how to effectively integrate genetic studies and functional genomics to identify therapeutic targets and potential drug repurposing opportunities.
We propose a novel computational strategy that combines statistical associations from Transcriptome-Wide Association Studies (TWAS) with gene modules exhibiting similar expression patterns in various cell types.
The key innovation of our approach lies in projecting gene-trait associations through a latent representation derived from diverse cell types under different stimuli and developmental stages, surpassing traditional measures of normal tissue.
This method enhances interpretation by uncovering cell type-specific characteristics of complex phenotypes.
By applying our approach, we can pinpoint disease-relevant cell types from summary statistics, with several disease-associated gene modules validated in the eMERGE dataset.
Through a CRISPR screen focusing on lipid regulation, we demonstrate that our gene module-based strategy effectively prioritizes causal genes even in the absence of single gene associations.
Our interpretation of these results aligns with the omnigenic model, distinguishing between "core" and "peripheral" genes and suggesting the potential to identify genes directly impacting traits without mediated regulation by other genes.
This enables the prioritization of alternative and potentially more promising therapeutic targets.


The academic paper titled 'Projecting genetic associations through gene expression patterns highlights disease etiology and drug mechanisms' explores the integration of genetic studies, functional genomics, gene co-expression, therapeutic targets, drug repurposing, and clustering of complex traits.
The research problem addressed is how to effectively connect diseases, drugs, and cell types using a gene module perspective.
The proposed solution involves integrating drug-induced transcriptional profiles to predict drug-disease links.
The study demonstrates that the LV-based drug-repurposing approach is more accurate than the gene-based approach for predicting links between 322 drugs and 53 diseases.
Additionally, the research focuses on cardiovascular traits and the drug niacin to illustrate how the approach can connect pathophysiological processes with known mechanisms of action in adipose tissue, immune cells, and ovarian granulosa cells.
The LV-based approach is suggested as a valuable tool for generating new hypotheses to investigate potential mechanisms of action or adverse effects of existing or experimental drugs.


Our study aimed to project genetic associations through gene expression patterns to highlight disease etiology and drug mechanisms.
We discovered that analyzing associations through latent representations allowed for reasonable groupings of diseases and traits affected by shared and distinct transcriptional mechanisms in relevant tissues.
Our cluster analysis approach successfully identified the most discriminative latent variables (LVs) for each cluster, some of which were significantly associated with different traits.
While some LVs aligned with known pathways, others, like LV57, may represent novel disease-relevant mechanisms.
In some instances, the features/LVs linked to phenotypes were associated with specific cell types, providing more precise insight into potentially causal cell types for a phenotype.
Modules expressed primarily in one tissue, such as adipose in LV246 or ovary in LV66, were observed, while others appeared to be expressed in multiple contexts, capturing pathways associated with related complex diseases.
For example, LV136 was associated with cardiovascular disease and corneal biomechanics, expressed in various cell types including fibroblasts, osteoblasts, pancreas, liver, and cardiomyocytes.
Other examples include LV844, expressed in whole blood samples and associated with autoimmune diseases, and LV57, clearly expressed in T cells and strongly associated with autoimmune diseases and venous thromboembolism.
These patterns may represent cases of "network pleiotropy," where the same cell types mediate molecularly related traits.
Our novel approach of projecting through a representation learned on complementary but distinct datasets offers a computationally simple method to identify cell type and pathway effects on complex phenotypes.


We demonstrated that clustering trees, originally introduced to examine developmental processes in single-cell data, offer a multi-resolution grouping of phenotypes based on latent variable associations.
We utilized hard-partitioning algorithms, where one trait exclusively belongs to one cluster, taking into account all gene modules when calculating distances between traits.
It is plausible for two complex diseases to share only a few biological processes rather than being similar across most of them.
Additionally, our TWAS results were obtained from a large set of GWAS with varying sample sizes and qualities.
While we addressed potential issues stemming from data heterogeneity before conducting cluster analyses on traits, data preprocessing steps are always challenging and may not completely eliminate bias.
Previous research has shown that considering groups of related diseases is more effective in detecting shared genetic etiology, and clustering trees offer a way to explore such relationships in the context of latent variables.


We developed an LV-based regression framework to identify associations between gene modules and traits using TWAS $p$-values.
The PhenomeXcan cohort was used as a discovery cohort, examining over four thousand traits, with many LV-trait associations replicating in eMERGE.
In PhenomeXcan, we identified 3,450 significant LV-trait associations (FDR < 0.05), with 686 LVs (out of 987) linked to at least one trait, and 1,176 traits associated with at least one LV.
In eMERGE, 196 significant LV-trait associations were found, with 116 LVs associated with at least one trait/phecode and 81 traits associated with at least one LV.
While we focused on specific disease types within our trait clusters, the complete set of associations across other disease domains can be accessed in our [Github repository](https://github.com/greenelab/phenoplier) for future research.
It is important to note, as detailed in the [Methods](#sec:methods:reg) section, that one limitation of the regression approach is the approximate accuracy of gene-gene correlations, potentially leading to false positives if the correlation among the top genes in a module is not precisely captured.
Despite this limitation, the regression model is well-calibrated, and no inflation was observed when applying the method to real data.


Our research aims to address the question of how gene expression patterns can be used to project genetic associations, highlighting disease etiology and drug mechanisms.
To achieve this, we propose an integrated approach that combines two methods.
The first method, MultiPLIER, extracts latent variables from large expression datasets, potentially representing transcriptional processes or technical factors.
However, the linear combinations used in this method may overlook complex co-expression patterns.
While the training dataset, recount2, has been surpassed in size by other resources, our models are flexible and can easily incorporate alternative approaches like GenomicSuperSignature.
The second method we employ is TWAS, focusing on the hypothesis that GWAS loci influence traits through changes in gene expression.
Although TWAS has limitations that can lead to false positives, our LV-based regression framework effectively accounts for gene-gene correlations within TWAS.
By integrating these two approaches, we aim to provide insights into disease mechanisms and potential therapeutic targets through the clustering of complex traits.


Our research paper, titled 'Projecting genetic associations through gene expression patterns highlights disease etiology and drug mechanisms', addresses the question of how to effectively prioritize disease-treatment pairs by projecting association results through latent variables.
Previous studies have shown that drugs with genetic support are more likely to succeed in the drug development pipeline.
Our proposed solution involves utilizing latent variables to better prioritize disease-treatment pairs, rather than considering single-gene effects alone.
This approach not only improves prioritization but also provides interpretable genetic features that can offer insights into potential mechanisms of action.
We specifically focused on prioritizing drugs for diseases with varying tissue etiologies, which presented a challenge in selecting the most appropriate tissue model from TWAS to identify reversed transcriptome patterns between genes and drug-induced perturbations.


The research problem addressed in this paper is how to project genetic associations through gene expression patterns to highlight disease etiology and drug mechanisms.
The proposed solution involves using a representation derived from a factorization of bulk RNA-seq data.
To improve the quality and interpretability of these representations, detailed perturbation datasets and single-cell profiling of tissues, with and without perturbagens, at various stages of development are essential.
Additionally, the interpretability of the representations relies on the annotation of sample metadata.
New approaches to infer and annotate structured metadata show promise and can be directly applied to existing data.
These advancements in representation quality and interpretability set the stage for the widespread application of latent variable projections to unravel the genetic basis of complex human phenotypes.
By offering a new perspective for mechanistic understanding of statistical associations from TWAS, our method can generate testable hypotheses for the post-GWAS functional characterization of complex diseases, which will likely be crucial in the future.
