## Methods {#sec:methods}

PhenoPLIER is a framework that integrates gene-trait associations and drug-induced transcriptional responses with groups of functionally-related genes, known as gene modules or latent variables (LVs).
Gene-trait associations are determined using the PrediXcan family of methods, while latent variables are identified through the MultiPLIER models applied on large gene expression compendia.
PhenoPLIER offers: 

1) A regression model to calculate an LV-trait association.
2) A consensus clustering approach implemented in the latent space to uncover shared and distinct transcriptomic characteristics among traits.
3) An interpretable, LV-based drug repurposing framework.

The specific details of these methods are outlined below.

*equation_definition
LV: Latent Variable
LV-trait association: Association between a latent variable and a trait.


### The PrediXcan family of methods for gene-based associations {#sec:methods:predixcan}

We utilized Summary-PrediXcan (S-PrediXcan) (Barbeira et al., 2018) and Summary-MultiXcan (S-MultiXcan) (Barbeira et al., 2019) as the gene-based statistical approaches, which are part of the PrediXcan family of methods (Barbeira et al., 2019).
These approaches are collectively known as TWAS (transcription-wide association studies).
S-PrediXcan, the summary-based version of PrediXcan, calculates the univariate association between a trait and a gene's predicted expression in a single tissue.
Conversely, S-MultiXcan, the summary-based version of MultiXcan, calculates the joint association between a gene's predicted expression in all tissues and a trait.
S-PrediXcan and S-MultiXcan only require GWAS summary statistics instead of individual-level genotype and phenotype data.

*equation_definition*
- S-PrediXcan: Univariate association between a trait and a gene's predicted expression in a single tissue.
- S-MultiXcan: Joint association between a gene's predicted expression in all tissues and a trait.

Here we briefly provide the details about the Transcriptome-Wide Association Study (TWAS) methods that are necessary to explain our regression framework later (see the referenced articles for more information).
In the following, we refer to $\mathbf{y}$ as a vector of traits for $n$ individuals that is centered for convenience (so that no intercept is necessary).

*equation_definition*
$\tilde{\mathbf{t}}_l = \sum_{a \in \mathrm{model}_l} w_{a}^{l} X_{a}$

This equation represents the gene's predicted expression for all individuals in tissue $l$, where $X_a$ is the genotype of SNP $a$, $w_{a}$ is its weight in the tissue prediction model $l$, and $\mathbf{t}_l$ is the standardized version of $\tilde{\mathbf{t}}_l$ with mean equal to zero and standard deviation equal to one.

S-PrediXcan [@doi:10.1038/s41467-018-03621-1] is the condensed form of PrediXcan [@doi:10.1038/ng.3367], which models a trait as a linear function of a gene's expression in a single tissue using the univariate model:

*equation_definition

$$
\mathbf{y} = \mathbf{t}_l \gamma_l + \bm{\epsilon}_l,
$$ {#eq:predixcan}

Here, $\hat{\gamma}_l$ represents the estimated effect size or regression coefficient, and $\bm{\epsilon}_l$ denotes the error terms with variance $\sigma_{\epsilon}^{2}$.
The significance of the association is determined by calculating the $z$-score $\hat{z}_{l}=\hat{\gamma}_l / \mathrm{se}(\hat{\gamma}_l)$ for a gene's tissue model $l$.
While PrediXcan requires individual-level data for model fitting, S-PrediXcan approximates PrediXcan $z$-scores using only GWAS summary statistics with the following expression:

*equation_definition

$$
\hat{z}_{l} \approx \sum_{a \in model_{l}} w_a^l \frac{\hat{\sigma}_a}{\hat{\sigma}_l} \frac{\hat{\beta}_a}{\mathrm{se}(\hat{\beta}_a)},
$$ {#eq:spredixcan}

Here, $\hat{\sigma}_a$ represents the variance of SNP $a$, $\hat{\sigma}_l$ is the variance of the predicted expression of a gene in tissue $l$, and $\hat{\beta}_a$ denotes the estimated effect size of SNP $a$ from the GWAS.
These Transcriptome-Wide Association Study (TWAS) methods rely on estimating genotype variances and covariances using the Genotype-Tissue Expression project (GTEx v8) [@doi:10.1126/science.aaz1776] as the reference panel.
Given that S-PrediXcan offers tissue-specific direction of effects (indicating whether higher or lower predicted gene expression correlates with increased disease risk), we utilized the $z$-scores in our drug repurposing strategy (explained later).

S-MultiXcan (Barbeira et al., 2019) is the summary version of MultiXcan, a tool more powerful than PrediXcan in detecting gene-trait associations, although it does not provide the direction of effects.
The primary output of MultiXcan is the p-value, obtained with an F-test, of the multiple tissue model:

$$
\begin{split}
\mathbf{y} & = \sum_{l=1}^{p} \mathbf{t}_l g_l + \mathbf{e} \\
& = \mathbf{T} \mathbf{g} + \mathbf{e},
\end{split}
$$

where $\mathbf{T}$ is a matrix with $p$ columns $\mathbf{t}_l$, $\hat{g}_l$ is the estimated effect size for the predicted gene expression in tissue $l$ (and thus $\hat{\mathbf{g}}$ is a vector with $p$ estimated effect sizes $\hat{g}_l$), and $\mathbf{e}$ are the error terms with variance $\sigma_{e}^{2}$.
Due to the high correlation between predicted expression values across different tissues, MultiXcan uses the principal components (PCs) of $\mathbf{T}$ to address collinearity issues.
S-MultiXcan derives the joint regression estimates (effect sizes and their variances) in Equation (1) using the marginal estimates from S-PrediXcan (Barbeira et al., 2019).
Under the null hypothesis of no association, $\hat{\mathbf{g}}^{\top} \frac{\mathbf{T}^{\top}\mathbf{T}}{\sigma_{e}^{2}} \hat{\mathbf{g}} \sim \chi_{p}^{2}$, and therefore the significance of the association in S-MultiXcan is estimated with:

$$
\begin{split}
\frac{\hat{\mathbf{g}}^{\top} (\mathbf{T}^{\top}\mathbf{T}) \hat{\mathbf{g}}}{\sigma_{e}^{2}} & \approx \bm{\hat{\gamma}}^{\top} \frac{\sqrt{n-1}}{\sigma_{\epsilon}} \left(\frac{\mathbf{T}^{\top} \mathbf{T}}{n-1}\right)^{-1} \frac{\sqrt{n-1}}{\sigma_{\epsilon}} \bm{\hat{\gamma}} \\
& = \hat{\mathbf{z}}^{\top} Cor(\mathbf{T})^{-1} \hat{\mathbf{z}},
\end{split}
$$

where $\hat{\mathbf{z}}$ is a vector with $p$ z-scores for each tissue available for the gene, and $Cor(\mathbf{T})$ is the autocorrelation matrix of $\mathbf{T}$.
Since $\mathbf{T}^{\top}\mathbf{T}$ is singular for many genes, S-MultiXcan computes the pseudo-inverse $Cor(\mathbf{T})^{+}$ using the $k$ top PCs, and thus $\hat{\mathbf{z}}^{\top} Cor(\mathbf{T})^{+} \hat{\mathbf{z}} \sim \chi_k^2$.
To arrive at this expression, S-MultiXcan uses the conservative approximation $\sigma_{e}^{2} \approx \sigma_{\epsilon}^{2}$, meaning the variance of the error terms in the joint regression is approximately equal to the residual variance of the marginal regressions.
Additionally, $Cor(\mathbf{T})$ is estimated using a global genotype covariance matrix, while marginal $\hat{z}_l$ in S-PrediXcan are approximated using tissue-specific genotype covariances.
Although S-MultiXcan yields highly concordant estimates compared with MultiXcan, results are not perfectly correlated across genes (Barbeira et al., 2019).
These differences are crucial for our LV-based regression model when computing the gene-gene correlation matrix.
We utilized S-MultiXcan results for our LV-based regression model and cluster analyses of traits.


### TWAS resources {#sec:methods:twas}

We used two large TWAS resources from different cohorts for discovery and replication, all obtained from European ancestries.
PhenomeXcan [@doi:10.1126/sciadv.aba2083], our discovery cohort, provides results on 4,091 traits across different categories.
Supplementary Data 1 has all the details about the included GWAS, sample size and disease/trait categories.
In PhenomeXcan, these publicly available GWAS summary statistics were used to compute
1) gene-based associations with the PrediXcan family of methods (described before), and
2) a posterior probability of colocalization between GWAS loci and *cis*-eQTL with fastENLOC [@doi:10.1126/sciadv.aba2083; @doi:10.1016/j.ajhg.2020.11.012].
We refer to the matrix of $z$-scores from S-PrediXcan (Equation (@eq:spredixcan)) across $q$ traits and $m$ genes in tissue $t$ as $\mathbf{M}^{t} \in \mathbb{R}^{q \times m}$.
As explained later, matrices $\mathbf{M}^{t}$ were used in our LV-based drug repurposing framework since they provide direction of effects.
The S-MultiXcan results (22,515 gene associations across 4,091 traits) were used in our LV-based regression framework and our cluster analyses of traits.
For the cluster analyses, we used the $p$-values converted to $z$-scores: $\mathbf{M}=\Phi^{-1}(1 - p/2)$, where $\Phi^{-1}$ is the probit function.
Higher $z$-scores correspond to stronger associations.

Our discovery cohort was eMERGE [@doi:10.1038/gim.2013.72], where the same TWAS methods were run on 309 phecodes [@doi:10.1101/2021.10.21.21265225] across different categories (more information about traits are available in [@doi:10.1101/2021.10.21.21265225]).
We used these results to replicate the associations found with our LV-based regression framework in PhenomeXcan.


### MultiPLIER and Pathway-level information extractor (PLIER) {#sec:methods:multiplier}

MultiPLIER (Gligorijevic et al., 2019) extracts patterns of co-expressed genes from recount2 (Collado-Torres et al., 2017) (without including GTEx samples), a large gene expression dataset.
The approach applies the pathway-level information extractor method (PLIER) (Mao et al., 2019), which performs unsupervised learning using prior knowledge (canonical pathways) to reduce technical noise.
PLIER uses a matrix factorization approach that deconvolutes gene expression data into a set of latent variables (LV), where each LV represents a gene module.

*equation_definition: LV = Latent Variable

The MultiPLIER models reduced the dimensionality in recount2 to 987 LVs.

*equation_definition: LVs = Latent Variables

Given a gene expression dataset $\mathbf{Y}^{m \times c}$ with $m$ genes and $c$ experimental conditions and a prior knowledge matrix $\mathbf{C} \in \{0,1\}^{m \times p}$ for $p$ MSigDB pathways (so that $\mathbf{C}_{ij} = 1$ if gene $i$ belongs to pathway $j$), PLIER finds $\mathbf{U}$, $\mathbf{Z}$, and $\mathbf{B}$ minimizing

$$
||\mathbf{Y} - \mathbf{Z}\mathbf{B}||^{2}_{F} + \lambda_1 ||\mathbf{Z} - \mathbf{C}\mathbf{U}||^{2}_{F} + \lambda_2 ||\mathbf{B}||^{2}_{F} + \lambda_3 ||\mathbf{U}||_{L^1}
$$ 

{#eq:met:plier_func}

subject to $\mathbf{U}>0, \mathbf{Z}>0$; $\mathbf{Z}^{m \times l}$ are the gene loadings with $l$ latent variables, $\mathbf{B}^{l \times c}$ is the latent space for $c$ conditions, $\mathbf{U}^{p \times l}$ specifies which of the $p$ prior-information pathways in $\mathbf{C}$ are represented for each latent variable, and $\lambda_i$ are different regularization parameters used in the training step.
$\mathbf{Z}$ is a low-dimensional representation of the gene space where each latent variable aligns as much as possible to prior knowledge, and it might represent either a known or novel gene module (i.e., a meaningful biological pattern) or noise. 

*equation_definition

- $\mathbf{Y}$: Gene expression dataset
- $\mathbf{C}$: Prior knowledge matrix
- $\mathbf{U}$: Latent variables matrix
- $\mathbf{Z}$: Gene loadings matrix
- $\mathbf{B}$: Latent space matrix
- $\lambda_i$: Regularization parameters
- $l$: Number of latent variables
- $p$: Number of pathways
- $c$: Number of conditions
- $m$: Number of genes

[@doi:10.1016/j.cels.2015.12.004]

For our drug repurposing and cluster analyses, we utilized a model to project gene-trait associations (obtained from TWAS) and gene-drug associations (obtained from LINCS L1000) into a low-dimensional gene module space.
Specifically, TWAS associations $\mathbf{M}$ (derived from either S-PrediXcan or S-MultiXcan) were projected using the following equation:

*equation_definition*
$$
\hat{\mathbf{M}} = (\mathbf{Z}^{\top} \mathbf{Z} + \lambda_{2} \mathbf{I})^{-1} \mathbf{Z}^{\top} \mathbf{M},
$$ {#eq:proj}
*equation_definition*

Here, $\hat{\mathbf{M}}^{l \times q}$ is a matrix where traits are represented by gene modules instead of single genes.
Later on, we applied the same methodology to project drug-induced transcriptional profiles in LINCS L1000, enabling us to represent drugs using gene modules.


### Regression model for LV-trait associations {#sec:methods:reg}

We adapted the gene-set analysis framework from MAGMA [@doi:10.1371/journal.pcbi.1004219] to TWAS.
We used a competitive test to predict gene-trait associations from TWAS using gene weights from an LV, testing whether top-weighted genes for an LV are more strongly associated with the phenotype than other genes with relatively small or zero weights.
Thus, we fit the model

$$
\mathbf{m}=\beta_{0} + \mathbf{s} \beta_{s} + \sum_{i} \mathbf{x}_{i} \beta_{i} + \bm{\epsilon},
$$ {#eq:reg:model}

In our study, we utilized a vector $\mathbf{m}$ to represent S-MultiXcan gene $p$-values for a particular trait, which underwent a $-log_{10}$ transformation.
Additionally, we employed a binary indicator vector $\mathbf{s}$, where $s_{\ell}$ equals 1 for the top 1% of genes with the largest loadings for latent variable (LV) $\ell$ from vector $\mathbf{Z}_{\ell}$, and 0 otherwise.
The gene property $\mathbf{x}_{i}$ was included as a covariate, and effect sizes were denoted by $\beta$ (with $\beta_{0}$ serving as the intercept).
Furthermore, the error terms were represented by vector $\bm{\epsilon}$, which followed a multivariate normal distribution (MVN) with a mean of 0 and variance $\sigma^{2}$, and a correlation matrix $\mathbf{R}$ accounting for gene correlations.

*equation_definition: $\mathbf{m}$ - vector of S-MultiXcan gene $p$-values for a trait
*equation_definition: $\mathbf{s}$ - binary indicator vector with $s_{\ell}=1$ for top 1% of genes with largest loadings for LV $\ell$
*equation_definition: $\mathbf{x}_{i}$ - gene property used as a covariate
*equation_definition: $\beta

The model tests the null hypothesis $\beta_{s} = 0$ against the one-sided hypothesis $\beta_{s} > 0$.
Therefore, $\beta_{s}$ reflects the difference in trait associations between genes that are part of latent variable (LV) $\ell$ and genes outside of it.
Following the MAGMA framework, we used two gene properties as covariates:

*equation_definition*
1) *gene size*: defined as the number of principal components (PCs) retained in S-MultiXcan,
2) *gene density*: defined as the ratio of the number of PCs to the number of tissues available.

Since the error terms $\bm{\epsilon}$ could be correlated, we cannot assume they have independent normal distributions as in a standard linear regression model.
In the PrediXcan family of methods, the predicted expression of a pair of genes could be correlated if they share eQTLs or if these are in LD.
Therefore, we used a generalized least squares approach to account for these correlations.
The gene-gene correlation matrix $\mathbf{R}$ was approximated by computing the correlations between the model sum of squares (SSM) for each pair of genes under the null hypothesis of no association.
These correlations are derived from the individual-level MultiXcan model (Equation (@eq:multixcan)), where the predicted expression matrix $\mathbf{T}_{i} \in \mathbb{R}^{n \times p_i}$ of a gene $i$ across $p_i$ tissues is projected into its top $k_i$ principal components (PCs), resulting in matrix $\mathbf{P}_{i} \in \mathbb{R}^{n \times k_i}$.
From the MAGMA framework, we know that the SSM for each gene is proportional to $\mathbf{y}^{\top} \mathbf{P}_{i} \mathbf{P}_{i}^{\top} \mathbf{y$.
Under the null hypothesis of no association, the covariances between the SSM of genes $i$ and $j$ are therefore given by $2 \times \mathrm{Trace}(\mathbf{P}_{i}^{\top} \mathbf{P}_{j} \mathbf{P}_{j}^{\top} \mathbf{P}_{i})$.
The standard deviations of each SSM are given by $\sqrt{2 \times k_{i}} \times (n - 1)$.
Therefore, the correlation between the SSMs for genes $i$ and $j$ can be written as follows:

*equation_definition*

$$
\begin{split}
\mathbf{R}_{ij} & = \frac{2 \times \mathrm{Tr}(\mathbf{P}_{i}^{\top} \mathbf{P}_{j} \mathbf{P}_{j}^{\top} \mathbf{P}_{i})}{\sqrt{2 \times k_{i}} \times \sqrt{2 \times k_{j}} \times (n - 1)^2} \\
& = \frac{2 \times \mathrm{Tr}(Cor(\mathbf{P}_{i}, \mathbf{P}_{j}) \times Cor(\mathbf{P}_{j}, \mathbf{P}_{i}))}{\sqrt{2 \times k_{i}} \times \sqrt{2 \times k_{j}}},
\end{split}
$$ {#eq:reg:r}

where columns $\mathbf{P}$ are standardized, $\mathrm{Tr}$ is the trace of a matrix, and the cross-correlation matrix between PCs $Cor(\mathbf{P}_{i}, \mathbf{P}_{j}) \in \mathbb{R}^{k_i \times k_j}$ is given by

*equation_definition*

$$
\begin{split}
Cor(\mathbf{P}_{i}, \mathbf{P}_{j}) & = Cor(\mathbf{T}_{i} \mathbf{V}_{i}^{\top} \mathrm{diag}(\lambda_i)^{-1/2}, \mathbf{T}_{j} \mathbf{V}_{j}^{\top} \mathrm{diag}(\lambda_j)^{-1/2}) \\
& = \mathrm{diag}(\lambda_i)^{-1/2} \mathbf{V}_{i} (\frac{\mathbf{T}_{i}^{\top} \mathbf{T}_{j}}{n-1}) \mathbf{V}_{j}^{\top} \mathrm{diag}(\lambda_j)^{-1/2},
\end{split}
$$ {#eq:reg:cor_pp}

where $\frac{\mathbf{T}_{i}^{\top} \mathbf{T}_{j}}{n-1} \in \mathbb{R}^{p_i \times p_j}$ is the cross-correlation matrix between the predicted expression levels of genes $i$ and $j$, and columns of $\mathbf{V}_{i}$ and scalars $\lambda_i$ are the eigenvectors and eigenvalues of $\mathbf{T}_{i}$, respectively.
S-MultiXcan keeps only the top eigenvectors using a condition number threshold of $\frac{\max(\lambda_i)}{\lambda_i} < 30$.
To estimate the correlation of predicted expression levels for genes $i$ in tissue $k$ and gene $j$ in tissue $l$, $(\mathbf{t}_k^i, \mathbf{t}_l^j)$ ($\mathbf{t}_k^i$ is the $k$th column of $\mathbf{T}_{i}$), we used.

*equation_definition*

$$
\begin{split}
\frac{(\mathbf{T}_{i}^{\top} \mathbf{T}_{j})_{kl}}{n-1} & = Cor(\mathbf{t}_k^i, \mathbf{t}_l^j) \\
& = \frac{ Cov(\mathbf{t}_k, \mathbf{t}_l) } { \sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)} } \\
& = \frac{ Cov(\sum_{a \in \mathrm{model}_k} w_a^k X_a, \sum_{b \in \mathrm{model}_l} w_b^l X_b) }  {\sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)} } \\
& = \frac{ \sum_{\substack{a \in \mathrm{model}_k \\ b \in \mathrm{model}_l}} w_a^k w_b^l Cov(X_a, X_b)} {\sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)} } \\
& = \frac{ \sum_{\substack{a \in \mathrm{model}_k \\ b \in \mathrm{model}_l}} w_a^k w_b^l \Gamma_{ab}} {\sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)} },
\end{split}
$$ {#eq:reg:corr_genes}

where $X_a$ is the genotype of SNP $a$, $w_a^k$ is the weight of SNP $a$ for gene expression prediction in the tissue model $k$, and $\Gamma = \widehat{\mathrm{var}}(\mathbf{X}) = (\mathbf{X} - \bar{\mathbf{X}})^{\top} (\mathbf{X} - \bar{\mathbf{X}}) / (n-1)$ is the genotype covariance matrix using GTEx v8 as the reference panel, which is the same used in all TWAS methods described here.
The variance of the predicted expression values of gene $i$ in tissue $k$ is estimated as.

*equation_definition*

$$
\begin{split}
\widehat{\mathrm{var}}(\mathbf{t}_k^i) & = (\mathbf{W}^k)^\top \Gamma^k \mathbf{W}^k \\
& = \sum_{\substack{a \in \mathrm{model}_k \\ b \in \mathrm{model}_k}} w_a^k w_b^k \Gamma_{ab}^k.
\end{split}
$$ {#eq:reg:var_gene}

Since we utilized the MultiXcan regression model (Equation (1)), $\mathbf{R}$ serves as an approximation of gene correlations in S-MultiXcan.
As previously described, S-MultiXcan approximates the joint regression parameters in MultiXcan by utilizing the marginal regression estimates from S-PrediXcan in Equation (2) with certain simplifying assumptions and different genotype covariance matrices.
This presents challenges in deriving an S-MultiXcan-specific solution to calculate $\mathbf{R}$.
To address this issue, we opted to use a submatrix $\mathbf{R}_{\ell}$ that corresponds to genes belonging to LV $\ell$ exclusively (the top 1% of genes), rather than the entire matrix $\mathbf{R}$.
This simplification is considered conservative as it takes into account correlations for only the top genes.
Our simulations (see Supplementary Note 1) demonstrate that the model is reasonably well-calibrated and can rectify LVs with closely related and highly correlated genes at the top (e.g., Figure 1).
However, the simulations also identified 127 LVs where the model lacked calibration (e.g., Figure 2).
This discrepancy can be attributed to challenges in accurately computing a gene correlation matrix, prompting us to exclude these LVs from our primary analyses. 

*equation_definition
- $\mathbf{R}$: Gene correlation matrix
- S-MultiXcan: Extension of MultiXcan regression model
- S-PrediXcan: Method for predicting gene expression levels

In Equation (@eq:reg:corr_genes), we only considered tissue models present in S-PrediXcan results and SNPs present in GWAS used as input for the TWAS approaches for each gene to obtain more accurate correlation estimates [@doi:10.1371/journal.pgen.1007889].
This step was essential for the computation of different correlation matrices for PhenomeXcan and eMERGE datasets.
In PhenomeXcan, where most GWAS (4,049) were sourced from the UK Biobank using a consistent pipeline and SNPs, a single correlation matrix sufficed.
Conversely, for other datasets, we employed a distinct correlation matrix for each group of traits that shared the same or most SNPs. 

*equation_definition
Equation (@eq:reg:corr_genes): Calculation of correlation between genes based on tissue models and SNPs present in GWAS.

*symbol_definition
GWAS: Genome-Wide Association Study
TWAS: Transcriptome-Wide Association Study
SNPs: Single Nucleotide Polymorphisms

We ran our regression model for all 987 LVs across the 4,091 traits in PhenomeXcan.
For replication, we ran the model in the 309 phecodes in eMERGE.
We adjusted the $p$-values using the Benjamini-Hochberg procedure.


### LV-based drug repurposing approach {#sec:methods:drug}

For the drug-disease prediction, we developed an LV-based method within a drug repositioning framework previously utilized for psychiatry traits (Gandal et al., 2018), where individual genes associated with a trait are anticorrelated with drug expression profiles.
Our LV-based method was compared to the single-gene approach described in a study by Subramanian et al.
(2017).
In the single-gene method, we calculated a drug-disease score by multiplying each S-PrediXcan set of signed z-scores in tissue $t$, denoted as $\mathbf{M}^t$, with another set of signed z-scores from transcriptional responses profiled in LINCS L1000 (Subramanian et al., 2017), represented as $\mathbf{L}^{c \times m}$ (for $c$ compounds).
Here, $\mathbf{M}^t$ indicates whether a higher or lower predicted expression of a gene is linked to disease risk, while $\mathbf{L}$ reflects whether a drug increases or decreases the expression of a gene.
These two matrices can be multiplied to compute a score for a drug-disease pair.
The resulting product is defined as:

*equation_definition*
$\mathbf{D}^{t,k}=-1 \cdot \mathbf{M}^{t,k} \mathbf{L}^\top$

where $k$ denotes the number of most significant gene associations in $\mathbf{M}^t$ for each trait.
As proposed by Gandal et al.
(2018), $k$ could encompass all genes or be limited to the top 50, 100, 250, or 500 genes.
Subsequently, we averaged score ranks across all $k$ and obtained $\mathbf{D}^t$.
Finally, for each drug-disease pair, we selected the maximum prediction score across all tissues:

*equation_definition*
$\mathbf{D}_{ij} = \max \{ \mathbf{D}_{ij}^t \mid \forall t \}$


The same procedure was utilized for the latent variable (LV)-based approach, in which we projected the matrices $\mathbf{M}^{t}$ and $\mathbf{L}$ into the gene module latent space using the following equation:

\begin{equation}
\hat{\mathbf{M}}^t = \mathbf{M}^{t} \mathbf{W}^{t}, \quad \hat{\mathbf{L}} = \mathbf{L} \mathbf{W}^{l}
\end{equation}

This resulted in $\hat{\mathbf{M}}^t$ and $\hat{\mathbf{L}}^{l \times c}$, respectively.
Finally, we calculated the matrix $\mathbf{D}^{t,k}$ as:

\begin{equation}
\mathbf{D}^{t,k} = -1 \cdot \hat{\mathbf{L}}^{\top} \hat{\mathbf{M}}^{t,k}
\end{equation}

In this case, the parameter $k$ could encompass all LVs or the


Since the gold standard of drug-disease medical indications is described with Disease Ontology IDs (DOID) [@doi:10.1093/nar/gky1032], we mapped PhenomeXcan traits to the Experimental Factor Ontology [@doi:10.1093/bioinformatics/btq099] using [@url:https://github.com/EBISPOT/EFO-UKB-mappings], and then to DOID.


### Consensus clustering of traits {#sec:methods:clustering}

We performed two preprocessing steps on the S-MultiXcan results before the cluster analysis.
First, we combined results in matrix $\mathbf{M}$ (with $p$-values converted to $z$-scores, as described before) for traits that mapped to the same Experimental Factor Ontology (EFO) (Welter et al., 2014) term using the Stouffer's method:

\[
\sum w_i M_{ij} / \sqrt{\sum w_i^2}
\]

where $w_i$ is a weight based on the GWAS sample size for trait $i$, and $M_{ij}$ is the $z$-score for gene $j$.
Second, we divided all $z$-scores for each trait $i$ by their sum to reduce the effect of highly polygenic traits:

\[
M_{ij} / \sum M_{ij}
\]

Finally, we projected this data matrix using Equation (1), obtaining $\hat{\mathbf{M}}$ with $n$ = 3,752 traits and $l$ = 987 LVs as the input of our clustering pipeline.


A partitioning of the estimated matrix, denoted as $\hat{\mathbf{M}}$, with $n$ traits into $k$ clusters is represented as a label vector $\pi \in \mathbb{N}^n$.
Consensus clustering approaches consist of two steps: 1) the generation of an ensemble $\Pi$ with $r$ partitions of the dataset: $\Pi=\{\pi_1, \pi_2, \ldots, \pi_r\}$, and 2) the combination of the ensemble into a consolidated solution defined as:

$$
\pi^* = \mathrm{arg}\,\underset{\hat{\pi}}{\max} Q(\{ \lvert \mathcal{L}^i \lvert \phi(\hat{\pi}_{\mathcal{L}^i}, \pi_{i \mathcal{L}^i}) \mid i \in \{1,\ldots,r\} \}),
$$

where $\mathcal{L}^i$ is a set of data indices with known cluster labels for partition $i$, $\phi\colon \mathbb{N}^n \times \mathbb{N}^n \to \mathbb{R}$ is a function that measures the similarity between two partitions, and $Q$ is a measure of central tendency, such as the mean or median.
The adjusted Rand index (ARI) was used for $\phi$ [@doi:10.1007/BF01908075], and the median was used for $Q$.
To obtain $\pi^*$, a consensus function $\Gamma\colon \mathbb{N}^{n \times r} \to \mathbb{N}^n$ with $\Pi$ as the input is defined.
Consensus functions based on the evidence accumulation clustering (EAC) paradigm were utilized [@doi:10.1109/TPAMI.2005.113], where $\Pi$ is first transformed into a distance matrix $\mathbf{D}_{ij} = d_{ij} / r$, with $d_{ij}$ representing the number of times traits $i$ and $j$ were grouped in different clusters across all $r$ partitions in $\Pi$.
Subsequently, any similarity-based clustering algorithm, denoted as $\Gamma$, is applied on $\mathbf{D}$ to derive the final partition $\pi^*$.


For the ensemble generation step, we utilized various algorithms to create a highly diverse set of partitions, as diversity plays a crucial role in ensembles (refer to Figure 1).
This approach was inspired by previous studies [@doi:10.1016/j.ins.2016.04.027; @doi:10.1109/TPAMI.2011.84; @doi:10.1016/j.patcog.2014.04.005].
Three different data representations were considered: the raw dataset, its projection into the top 50 principal components, and the embedding learned by UMAP [@arxiv:1802.03426] using 50 components.
Each of these representations was subjected to five clustering algorithms that encompass a broad spectrum of assumptions regarding the data structure: $k$-means [@Arthur2007], spectral clustering [@Ng2001], a Gaussian mixture model (GMM), hierarchical clustering, and DBSCAN [@Ester1996].

*equation_definition*

For $k$-means, spectral clustering, and GMM, a range of $k$ values between 2 and $\sqrt{n} \approx 60$ was specified, with five partitions generated for each $k$ using different random seeds.
In the case of hierarchical clustering, four partitions were created for each $k$ utilizing common linkage criteria such as ward, complete, average, and single.
For DBSCAN, various ranges of parameters $\epsilon$ (maximum distance between two data points to be considered part of the same neighborhood) and *minPts* (minimum number of data points in a neighborhood for a data point to be considered a core point) were explored based on a method outlined in a previous study [@doi:10.1088/1755-1315/31/1/012012].
Specifically, *minPts* values ranging from 2 to 125 were considered.

*equation_definition*

The plausible range of $\epsilon$ values for each data representation (raw, PCA, and UMAP) was determined by analyzing the distribution of the mean distance of the *minPts*-nearest neighbors across all data points.
To address situations where certain combinations of *minPts* and $\epsilon$ did not yield meaningful partitions, such as detecting all points as noisy or identifying only one cluster, a resampling strategy was implemented for partitions generated by DBSCAN to ensure an equal representation of this algorithm in the ensemble.
Overall, this process resulted in a final ensemble comprising 4,428 partitions of 3,752 traits.


Finally, spectral clustering was utilized on matrix $\mathbf{D}$ to obtain the ultimate consensus partitions.
Matrix $\mathbf{D}$ was initially converted into a similarity matrix by applying a Radial Basis Function (RBF) kernel $\mathrm{exp}(-\gamma \mathbf{D}^2)$ using four distinct values for $\gamma$ that were determined empirically to be the most effective.
Consequently, for each value of $k$ ranging from 2 to 60, four consensus partitions were derived and the one maximizing Equation (@eq:consensus:obj_func) was selected.
Subsequently, the set of 59 solutions was filtered to retain only those with an ensemble agreement exceeding the 75th percentile (refer to Figure @fig:sup:clustering:agreement), resulting in a total of 15 final consensus partitions as illustrated in Figure @fig:clustering:tree. 

*equation_definition: $\mathrm{exp}(-\gamma \mathbf{D}^2)$ - Radial Basis Function (RBF) kernel
*equation_definition: Equation (@eq:consensus:obj_func) - Objective function for consensus partition maximization

The input data in our clustering pipeline undergoes several linear and nonlinear transformations, including Principal Component Analysis (PCA), Uniform Manifold Approximation and Projection (UMAP), and the ensemble transformation using the Ensemble Average Clustering (EAC) paradigm (distance matrix $\mathbf{D}$).

*equation_definition
- PCA: Principal Component Analysis
- UMAP: Uniform Manifold Approximation and Projection
- EAC: Ensemble Average Clustering

Although consensus clustering has clear advantages for biological data [1], this set of data transformations complicates the interpretation of results.
To circumvent this, we used a supervised learning approach to detect which gene modules/Latent Variables (LVs) are the most important for each cluster of traits (Figure 1b).
Note that we did not use this supervised model for prediction but only to learn which features (LVs) were most discriminative for each cluster.

For this, we used the highest resolution partition ($k=29$, although any could be used) to train a decision tree model using each of the clusters as labels and the projected data $\hat{\mathbf{M}}$ as the training samples.
For each $k$, we built a set of binary labels with the current cluster's traits as the positive class and the rest of the traits as the negative class.
Then, we selected the LV in the root node of the trained model only if its threshold was positive and larger than one standard deviation.

Next, we removed this LV from $\hat{\mathbf{M}}$ (regardless of being previously selected or not) and trained the model again.
We repeated this procedure 20 times to extract the top 20 LVs that better discriminate traits in a cluster from the rest.

In [Supplementary Note 2](#sm:clustering:null_sim), we performed several analyses under a null hypothesis of no structure in the data to verify that the clustering results detected by this pipeline were real.


### CRISPR-Cas9 screening {#sec:methods:crispr}

**Cell culture.** HepG2 cells were obtained from ATCC (ATCC® HB-8065™) and were cultured in Eagle's Minimum Essential Medium with L-Glutamine (EMEM, Cat.
112-018-101, Quality Biology) supplemented with 10% Fetal Bovine Serum (FBS, Gibco, Cat.16000-044) and 1% Penicillin-Streptomycin (Pen/Strep, Gibco, Cat.15140-122).
The cells were maintained at 37°C in a humidity-controlled incubator with 5% CO2 and were kept at a density not exceeding 80% confluency in Collagen-I coated flasks.

*equation_definition*
- EMEM: Eagle's Minimum Essential Medium
- FBS: Fetal Bovine Serum
- Pen/Strep: Penicillin-Streptomycin

*equation_definition*

**Genome-wide lentiviral pooled CRISPR-Cas9 library.** The 3rd generation Broad GPP genome-wide Human Brunello CRISPR knockout Pooled library, obtained from Addgene (Cat.
73179-LV) courtesy of David Root and John Doench, was utilized for HepG2 cell transduction.
This library comprises 76,441 sgRNAs targeting 19,114 genes in the human genome, with an average of 4 sgRNAs per gene.
Each 20 nucleotide sgRNA sequence was integrated into the lentiCRIS-PRv2 backbone between the U6 promoter and gRNA scaffold.
Lentiviral vectors encoding Cas9 were employed to introduce the sgRNA cassette-containing plasmids into cells during replication.
Cells that did not successfully transduce were eliminated through puromycin selection.

*equation_definition*
- sgRNA: single guide RNA
- U6: RNA polymerase III promoter

Equation (1):
\[ sgRNA = 20nt \]

Equation (2):
\[ sgRNA \, cassette = U6 \, promoter + gRNA \, scaffold \]

**Lentiviral titer determination.** No-spin lentiviral transduction was utilized for the screen.
Approximately 2.5 million cells were seeded in each well of a Collagen-I coated 6-well plate in the presence of 8 µg/ml polybrene (Millipore Sigma, Cat.
TR-1003 G).
Different titrated virus volumes (e.g., 0, 50, 100, 200, 250, and 400 µl) were assigned to each well.
EMEM complete media was added to make the final volume 1.24 ml.
Sixteen to eighteen hours post-transduction, virus/polybrene-containing media was removed from each well.
Cells were washed twice with 1x DPBS and replaced with fresh EMEM.
At 24 hours, cells in each well were trypsinized, diluted (e.g., 1:10), and seeded in pairs of wells of 6-well plates.
At 60 hours post-transduction, cell media in each well was replaced with fresh EMEM.
Two µg/ml of puromycin (Gibco, Cat.
A1113803) was added to one well out of the pair.
Two to five days after puromycin selection, or if the 0 virus well treated with puromycin had no surviving cells, cells in both wells with and without puromycin were collected and counted for viability.

*equation_definition*
PI% = (Number of cells with puromycin selection / Number of cells without puromycin selection) x 100

Percentage of Infection (PI%) was obtained by comparing the cell numbers with and without puromycin selection within each pair.
By means of Poisson's distribution theory, when the transduction efficiency (PI%) is between 30-50%, which corresponds to a Multiplicity of Infection (MOI) of approximately 0.35-0.70.
At an MOI close to 0.3, around 25% of cells are infected, and the majority of those infected cells are predicted to have only one copy of the virus.
Therefore, a virus volume of 120 µl, yielding 30-40% transduction efficiency, was chosen for further large-scale viral transduction.

Lentiviral transduction was performed in HepG2 cells using the Brunello CRISPR knockout pooled library to ensure a coverage of at least 500 cells per single-guide RNA (sgRNA).
An MOI between 0.3-0.4 was maintained to guarantee that 95% of infected cells received only one viral particle per cell.
Approximately 200 million cells were used for the screen.
The transduction process followed a similar protocol as previously described.
Specifically, 2.5 million cells were seeded in each well of 14 6-well plates with 8 μg/ml of polybrene.
Subsequently, 120 μl of the virus was added to each experimental well.
After 18 hours post-transduction, the virus/polybrene mix medium was removed, and cells from each well were collected, counted, and combined into T175 flasks.
At 60 hours post-transduction, 2 μg/ml of puromycin was introduced to each flask.
The mediums were refreshed every two days with fresh EMEM supplemented with 2 μg/ml puromycin.
Seven days after puromycin selection, the cells were harvested, pooled, counted, and replated.

*equation_definition*
MOI = (Number of viral particles)/(Number of target cells)

*equation_definition*
Coverage = (Number of cells per sgRNA)/(Total number of cells)

Fluorescent dye staining was conducted 9 days after puromycin selection, with cells divided into two groups.
The Unsorted Control group consisted of 20-30 million cells, which were collected and centrifuged at 500 x g for 5 minutes at 4°C.
The resulting pellet was stored at -80°C for subsequent genomic DNA isolation.
The remaining cells, approximately 200 million, were plated in 100mm dishes and subjected to staining with a fluorescent dye (LipidSpotTM 488, Biotium, Cat.
70065-T).
Specifically, LipidSpot 488 was diluted to a 1:100 ratio with DPBS, and 4ml of the staining solution was added to each dish, followed by an incubation at 37°C for 30 minutes.
Cell images were acquired using a fluorescent microscope EVOS to detect the GFP signal (Figure 1). 

*equation_definition*

Equation (@id): Symbol1 = Definition of Symbol1

Equation (@id): Symbol2 = Definition of Symbol2

**Fluorescence-activated cell sorting (FACS).** Cells were immediately collected into 50 mL tubes and spun at 500 x g for 5 minutes at 4°C.
After a DPBS wash, cell pellets were resuspended with FACS Sorting Buffer (1x DPBS without Ca2+/Mg2+, 2.5 mM EDTA, 25 mM HEPES, 1% BSA).
The solution was filter sterilized and kept at 4°C, with cells kept cold from this point on.
Single cells were obtained by gentle pipetting.
The cell solution was then filtered through a cell strainer (Falcon, Cat.
352235) and kept on ice, protected from light.
Collected cells were sorted on FACSJazz using a 100 µm nozzle.
Approximately 20% of each GFP-High and GFP-Low population (Figure 2) were collected into 15 mL tubes.
After sorting, cells were immediately spun down, and the pellets were kept at -80°C for further genomic DNA isolation.

*equation_definition*
- FACS Sorting Buffer = 1x DPBS without Ca2+/Mg2+, 2.5 mM EDTA, 25 mM HEPES, 1% BSA

*equation_definition*
- GFP-High = High green fluorescent protein expression
- GFP-Low = Low green fluorescent protein expression

**Genomic DNA isolation and verification.** Three conditions of genomic DNA (Un-Sorted Control, lentiV2 GFP-High, and lentiV2 GFP-Low) were extracted using the QIAamp DNA Blood Mini Kit (Qiagen, Cat.51104), followed by UV spectroscopy (Nanodrop) to assess the quality and quantity of the genomic DNA.
A total of 80-160 µg of genomic DNA was isolated for each condition.
The sgRNA cassette and lentiviral-specific transgene in the isolated genomic DNA were verified through polymerase chain reaction (PCR) (Figure 3).

*equation_definition

Equation 1: \( y = mx + b \)

In Equation 1, \( y \) represents the dependent variable, \( x \) represents the independent variable, \( m \) represents the slope of the line, and \( b \) represents the y-intercept.

*equation_definition

**Illumina libraries generation and sequencing.** The fragment containing the sgRNA cassette was amplified using P5/P7 primers, as outlined in previous work by Smith et al.
[1], and the primer sequences were adapted from the Broad Institute protocol (Figure 1).
A stagger sequence of 0-8 nucleotides was included in P5, with an 8-base pair uniquely barcoded sequence in P7.
The primers were synthesized by Integrated DNA Technologies (IDT) and each primer was PAGE purified.
For each condition, 32 PCR reactions were set up.
Each 100 μL PCR reaction contained approximately 5 μg of genomic DNA, 5 μL of each 10 μM P5 and P7 primer.
ExTaq DNA Polymerase (TaKaRa, Cat.
RR001A) was used for amplification.
The PCR thermal cycler parameters were set as follows: initial denaturation at 95°C for 1 minute; followed by 24 cycles of denaturation at 94°C for 30 seconds, annealing at 52.5°C for 30 seconds, extension at 72°C for 30 seconds, and a final elongation at 72°C for 10 minutes.
The expected PCR products ranged from 285 to 293 base pairs (Figure 2A).
PCR products from the same condition were pooled and purified using SPRIselect beads (Beckman Coulter, Cat.
B23318).
The purified Illumina libraries were quantitated using Qubit, and the library quality was assessed on a Bioanalyzer using a High Sensitivity DNA Chip.
A single peak at approximately 285 base pairs was expected (Figure 2B).
The final Illumina library samples were sequenced on a NovaSeq 6000.
The samples were pooled and loaded onto an SP flow cell, along with a 20% PhiX control v3 library spike-in.

*equation_definition*
- P5: Primer 5
- P7: Primer 7
- gDNA: Genomic DNA
- PCR: Polymerase Chain Reaction
- PAGE: Polyacrylamide Gel Electrophoresis
- IDT: Integrated DNA Technologies
- ExTaq: Thermus aquaticus DNA polymerase
- SPRI: Solid Phase Reversible Immobilization


## Data availability

All the main datasets generated in this study are available at [https://doi.org/10.5281/zenodo.8071382](https://doi.org/10.5281/zenodo.8071382) [@doi:10.5281/zenodo.8071382] and the GitHub repository [https://github.com/greenelab/phenoplier](https://github.com/greenelab/phenoplier).

The main input datasets used are TWAS from PhenomeXcan [@doi:10.1126/sciadv.aba2083] for 4,091 traits and from the Electronic Medical Records and Genomics (eMERGE) network phase III [@doi:10.1101/2021.10.21.21265225] for 309 traits; transcriptional responses to small molecule perturbations from LINCS L1000 [@doi:10.1016/j.cell.2017.10.049] that were further preprocessed and mapped to DrugBank IDs from [@doi:10.5281/zenodo.47223]; latent space/gene module models from MultiPLIER [@doi:10.1016/j.cels.2019.04.003].

The data utilized in this study were obtained from publicly available sources including PhenomeXcan, LINCS L1000, and MultiPLIER.
All significant findings from the phenome-wide TWAS conducted on eMERGE and Penn Medicine BioBank (PMBB) datasets are detailed in a previous publication [@doi:10.1101/2021.10.21.21265225].
The raw individual-level datasets from PMBB cannot be publicly shared due to institutional privacy policies.
Researchers seeking access to these datasets should reach out to the Penn Medicine Biobank ([https://pmbb.med.upenn.edu/pmbb/](https://pmbb.med.upenn.edu/pmbb/)).
The eMERGE network phase III data can be accessed on dbGAP under the accession number phs001584.v2.p2. 

*equation_definition*

Equation (@id): Symbol represents Definition.

*equation_definition*

Equation (@doi:10.1101/2021.10.21.21265225): Symbol represents Definition.


## Code availability

The code necessary to reproduce all the analyses in this work is available at [https://doi.org/10.5281/zenodo.8071382](https://doi.org/10.5281/zenodo.8071382) [@doi:10.5281/zenodo.8071382] and the GitHub repository [https://github.com/greenelab/phenoplier](https://github.com/greenelab/phenoplier).

For the CRISPR screening, we utilized FlowJo v10.7 and FACS Jazz Software v1.1.
Data analysis was conducted using Python 3.8 and R 3.6, along with various computational packages.
The key Python packages included Jupyter Lab (2.2), pandas (1.1), matplotlib (3.3), seaborn (0.11), numpy (1.19), scipy (1.5), scikit-learn (0.23), and umap-learn (0.4).
In R, we employed Bioconductor (3.10), clusterProfiler (3.14), clustree (0.4), and fgsea (1.17).
Additionally, we developed multiple scripts and notebooks, which are openly available under a permissive license.
Comprehensive documentation outlining all analysis steps was provided.
To ensure reproducibility, we offer a Docker image containing the same runtime environment and a demo for quick validation of the methods on authentic data.

*equation_definition:
- FlowJo v10.7: Flow cytometry data analysis software.
- FACS Jazz Software v1.1: Software for analyzing data from fluorescence-activated cell sorting (FACS) instruments.
- Python 3.8: Programming language for data analysis and scientific computing.
- R 3.6: Statistical computing and graphics software.
- Jupyter Lab (2.2): Interactive development environment for Python.
- pandas (1.1): Data manipulation and analysis library for Python.
- matplotlib (3.3): Data visualization library for Python.
- seaborn (0.11): Data visualization library based on matplotlib.
- numpy (1.19): Numerical computing library for Python.
- scipy (1.5): Scientific computing library for Python.
- scikit-
