## Methods {#sec:methods}

PhenoPLIER is a framework that combines various computational approaches to integrate gene-trait associations and drug-induced transcriptional responses with groups of functionally-related genes, known as gene modules or latent variables (LVs).
Gene-trait associations are calculated using the PrediXcan family of methods, while latent variables are inferred by the MultiPLIER models applied to large gene expression compendia.
PhenoPLIER offers 1) a regression model to calculate an LV-trait association, 2) a consensus clustering approach applied to the latent space to identify shared and distinct transcriptomic properties between traits, and 3) an interpretable, LV-based drug repurposing framework.
The details of these methods are provided below.

*equation_definition: LV-trait association regression model:
LV_trait = β0 + β1 * LV + ε

In this equation, LV_trait represents the association between a latent variable (LV) and a trait, β0 is the intercept, β1 is the coefficient, and ε is the error term.

*equation_definition: Consensus clustering approach:
Consensus_Clustering = argmax(1/k Σ Σ δ(i, j) * δ'(i', j'))

In this equation, Consensus_Clustering represents the clustering results, k is the number of clusters, δ(i, j) is the similarity matrix for the original clustering, and δ'(i', j') is the similarity matrix for the consensus clustering.

*equation_definition: LV-based drug repurposing framework:
Drug_Repurposing = LV_drug - LV_disease

In this equation, Drug_Repurposing represents the potential for drug repurposing, LV_drug is the latent variable associated with a drug, and LV_disease is the latent variable associated with a disease.


### The PrediXcan family of methods for gene-based associations {#sec:methods:predixcan}

We utilized Summary-PrediXcan (S-PrediXcan) (Barbeira et al., 2018) and Summary-MultiXcan (S-MultiXcan) (Barbeira et al., 2019) as the gene-based statistical approaches, which are part of the PrediXcan family of methods (Barbeira et al., 2018).
These approaches are collectively known as TWAS (transcription-wide association studies).
S-PrediXcan, the summary-based version of PrediXcan, calculates the univariate association between a trait and a gene's predicted expression in a single tissue.
In contrast, S-MultiXcan, the summary-based version of MultiXcan, calculates the joint association between a gene's predicted expression in all tissues and a trait.
Both S-PrediXcan and S-MultiXcan only require GWAS summary statistics instead of individual-level genotype and phenotype data.

*equation_definition*

Equation (@id): TWAS = S-PrediXcan + S-MultiXcan

*equation_definition*

Here we briefly provide the details about the Transcriptome-Wide Association Study (TWAS) methods that are necessary to explain our regression framework later (see the referenced articles for more information).

*equation_definition*
$\mathbf{y}$: a vector of traits for $n$ individuals that is centered for convenience (so that no intercept is necessary).

*equation_definition*
$\tilde{\mathbf{t}}_l = \sum_{a \in \mathrm{model}_l} w_{a}^{l} X_{a}$: the gene's predicted expression for all individuals in tissue $l$, where $X_a$ is the genotype of SNP $a$ and $w_{a}$ is its weight in the tissue prediction model $l$.

*equation_definition*
$\mathbf{t}_l$: the standardized version of $\tilde{\mathbf{t}}_l$ with mean equal to zero and standard deviation equal to one.

S-PrediXcan is the condensed version of PrediXcan.
PrediXcan represents the trait as a linear function of the gene's expression in a single tissue using the univariate model:

$$
\mathbf{y} = \mathbf{t}_l \gamma_l + \bm{\epsilon}_l,
$$ {#eq:predixcan}

where $\hat{\gamma}_l$ is the estimated effect size or regression coefficient, and $\bm{\epsilon}_l$ represents the error terms with variance $\sigma_{\epsilon}^{2}$.
The significance of the association is evaluated by calculating the $z$-score $\hat{z}_{l}=\hat{\gamma}_l / \mathrm{se}(\hat{\gamma}_l)$ for a gene's tissue model $l$.
PrediXcan requires individual-level data to fit this model, while S-PrediXcan estimates PrediXcan $z$-scores using only GWAS summary statistics with the formula:

$$
\hat{z}_{l} \approx \sum_{a \in model_{l}} w_a^l \frac{\hat{\sigma}_a}{\hat{\sigma}_l} \frac{\hat{\beta}_a}{\mathrm{se}(\hat{\beta}_a)},
$$ {#eq:spredixcan}

where $\hat{\sigma}_a$ denotes the variance of SNP $a$, $\hat{\sigma}_l$ represents the variance of the predicted expression of a gene in tissue $l$, and $\hat{\beta}_a$ is the estimated effect size of SNP $a$ from the GWAS.
In these Transcriptome-Wide Association Study (TWAS) methods, the genotype variances and covariances are estimated using the Genotype-Tissue Expression project (GTEx v8) as the reference panel.
Since S-PrediXcan offers tissue-specific direction of effects (indicating whether a higher or lower predicted expression of a gene leads to more or less disease risk), we utilized the $z$-scores in our drug repurposing strategy (explained below).

S-MultiXcan is the summary version of MultiXcan, which is more powerful than PrediXcan in detecting gene-trait associations, although it does not provide the direction of effects.
The main output of MultiXcan is the $p$-value obtained with an F-test in the multiple tissue model:

$$
\begin{split}
\mathbf{y} & = \sum_{l=1}^{p} \mathbf{t}_l g_l + \mathbf{e} \\
& = \mathbf{T} \mathbf{g} + \mathbf{e},
\end{split}
$$

where $\mathbf{T}$ is a matrix with $p$ columns $\mathbf{t}_l$, $\hat{g}_l$ is the estimated effect size for the predicted gene expression in tissue $l$, and $\mathbf{e}$ represents the error terms with variance $\sigma_{e}^{2}$.
Due to the high correlation between predicted expression values for a gene across different tissues, MultiXcan uses the principal components (PCs) of $\mathbf{T}$ to address collinearity issues.
S-MultiXcan derives joint regression estimates (effect sizes and variances) in Equation (@eq:multixcan) using the marginal estimates from S-PrediXcan in Equation (@eq:spredixcan).
Under the null hypothesis of no association, $\hat{\mathbf{g}}^{\top} \frac{\mathbf{T}^{\top}\mathbf{T}}{\sigma_{e}^{2}} \hat{\mathbf{g}} \sim \chi_{p}^{2}$, and the significance of the association in S-MultiXcan is estimated as:

$$
\begin{split}
\frac{\hat{\mathbf{g}}^{\top} (\mathbf{T}^{\top}\mathbf{T}) \hat{\mathbf{g}}}{\sigma_{e}^{2}} & \approx \bm{\hat{\gamma}}^{\top} \frac{\sqrt{n-1}}{\sigma_{\epsilon}} \left(\frac{\mathbf{T}^{\top} \mathbf{T}}{n-1}\right)^{-1} \frac{\sqrt{n-1}}{\sigma_{\epsilon}} \bm{\hat{\gamma}} \\
& = \hat{\mathbf{z}}^{\top} Cor(\mathbf{T})^{-1} \hat{\mathbf{z}},
\end{split}
$$

where $\hat{\mathbf{z}}$ is a vector with $p$ $z$-scores for each tissue available for the gene, and $Cor(\mathbf{T})$ is the autocorrelation matrix of $\mathbf{T}$.
Since $\mathbf{T}^{\top}\mathbf{T}$ is singular for many genes, S-MultiXcan computes the pseudo-inverse $Cor(\mathbf{T})^{+}$ using the $k$ top PCs, and thus $\hat{\mathbf{z}}^{\top} Cor(\mathbf{T})^{+} \hat{\mathbf{z}} \sim \chi_k^2$.
It is important to note that S-MultiXcan uses the conservative approximation $\sigma_{e}^{2} \approx \sigma_{\epsilon}^{2}$, where the variance of the error terms in the joint regression is approximately equal to the residual variance of the marginal regressions.
Additionally, $Cor(\mathbf{T})$ is estimated using a global genotype covariance matrix, while marginal $\hat{z}_l$ in Equation (@eq:spredixcan) are approximated using tissue-specific genotype covariances.
Although S-MultiXcan yields highly concordant estimates compared to MultiXcan, results are not perfectly correlated across genes.
These differences are important for our LV-based regression model when computing the gene-gene correlation matrix.
We utilized S-MultiXcan results for our LV-based regression model and our cluster analyses of traits.


### TWAS resources {#sec:methods:twas}

We used two large TWAS resources from different cohorts for discovery and replication, all obtained from European ancestries.
PhenomeXcan [@doi:10.1126/sciadv.aba2083], our discovery cohort, provides results on 4,091 traits across different categories.
Supplementary Data 1 has all the details about the included GWAS, sample size and disease/trait categories.
In PhenomeXcan, these publicly available GWAS summary statistics were used to compute
1) gene-based associations with the PrediXcan family of methods (described before), and
2) a posterior probability of colocalization between GWAS loci and *cis*-eQTL with fastENLOC [@doi:10.1126/sciadv.aba2083; @doi:10.1016/j.ajhg.2020.11.012].
We refer to the matrix of $z$-scores from S-PrediXcan (Equation (@eq:spredixcan)) across $q$ traits and $m$ genes in tissue $t$ as $\mathbf{M}^{t} \in \mathbb{R}^{q \times m}$.
As explained later, matrices $\mathbf{M}^{t}$ were used in our LV-based drug repurposing framework since they provide direction of effects.
The S-MultiXcan results (22,515 gene associations across 4,091 traits) were used in our LV-based regression framework and our cluster analyses of traits.
For the cluster analyses, we used the $p$-values converted to $z$-scores: $\mathbf{M}=\Phi^{-1}(1 - p/2)$, where $\Phi^{-1}$ is the probit function.
Higher $z$-scores correspond to stronger associations.

Our discovery cohort was eMERGE [@doi:10.1038/gim.2013.72], where the same TWAS methods were run on 309 phecodes [@doi:10.1101/2021.10.21.21265225] across different categories (more information about traits are available in [@doi:10.1101/2021.10.21.21265225]).
We used these results to replicate the associations found with our LV-based regression framework in PhenomeXcan.


### MultiPLIER and Pathway-level information extractor (PLIER) {#sec:methods:multiplier}

MultiPLIER extracts patterns of co-expressed genes from recount2 (without including GTEx samples), a large gene expression dataset [@doi:10.1016/j.cels.2019.04.003; @doi:10.1038/nbt.3838].
The approach applies the pathway-level information extractor method (PLIER) [@doi:10.1038/s41592-019-0456-1], which performs unsupervised learning using prior knowledge (canonical pathways) to reduce technical noise.
PLIER uses a matrix factorization approach that deconvolutes gene expression data into a set of latent variables (LV), where each LV represents a gene module. 

*equation_definition*
LV: Latent Variable

The MultiPLIER models reduced the dimensionality in recount2 to 987 LVs.

Given a gene expression dataset $\mathbf{Y}^{m \times c}$ with $m$ genes and $c$ experimental conditions and a prior knowledge matrix $\mathbf{C} \in \{0,1\}^{m \times p}$ for $p$ MSigDB pathways (so that $\mathbf{C}_{ij} = 1$ if gene $i$ belongs to pathway $j$), PLIER finds $\mathbf{U}$, $\mathbf{Z}$, and $\mathbf{B}$ minimizing

$$
||\mathbf{Y} - \mathbf{Z}\mathbf{B}||^{2}_{F} + \lambda_1 ||\mathbf{Z} - \mathbf{C}\mathbf{U}||^{2}_{F} + \lambda_2 ||\mathbf{B}||^{2}_{F} + \lambda_3 ||\mathbf{U}||_{L^1}
$$ 

{#eq:met:plier_func}

subject to $\mathbf{U}>0, \mathbf{Z}>0$; $\mathbf{Z}^{m \times l}$ are the gene loadings with $l$ latent variables, $\mathbf{B}^{l \times c}$ is the latent space for $c$ conditions, $\mathbf{U}^{p \times l}$ specifies which of the $p$ prior-information pathways in $\mathbf{C}$ are represented for each LV, and $\lambda_i$ are different regularization parameters used in the training step.
$\mathbf{Z}$ is a low-dimensional representation of the gene space where each LV aligns as much as possible to prior knowledge, and it might represent either a known or novel gene module (i.e., a meaningful biological pattern) or noise. 

*equation_definition

- $\mathbf{Y}$: Gene expression dataset
- $\mathbf{Z}$: Gene loadings
- $\mathbf{B}$: Latent space for conditions
- $\mathbf{U}$: Specifies prior-information pathways
- $\lambda_i$: Regularization parameters

For our drug repurposing and cluster analyses, we utilized a model to project gene-trait associations (obtained from TWAS) and gene-drug associations (obtained from LINCS L1000) into a low-dimensional gene module space.
Specifically, TWAS associations $\mathbf{M}$, derived from either S-PrediXcan or S-MultiXcan, were projected using the following equation:

*equation_definition*
$$
\hat{\mathbf{M}} = (\mathbf{Z}^{\top} \mathbf{Z} + \lambda_{2} \mathbf{I})^{-1} \mathbf{Z}^{\top} \mathbf{M},
$$ {#eq:proj}
*equation_definition*

Here, $\hat{\mathbf{M}}^{l \times q}$ represents a matrix where traits are now depicted by gene modules rather than individual genes.
Later on, we applied the same methodology to project drug-induced transcriptional profiles from LINCS L1000, resulting in a representation of drugs based on gene modules.


### Regression model for LV-trait associations {#sec:methods:reg}

We adapted the gene-set analysis framework from MAGMA [@doi:10.1371/journal.pcbi.1004219] to TWAS.
We used a competitive test to predict gene-trait associations from TWAS using gene weights from an LV, testing whether top-weighted genes for an LV are more strongly associated with the phenotype than other genes with relatively small or zero weights.
Thus, we fit the model

$$
\mathbf{m}=\beta_{0} + \mathbf{s} \beta_{s} + \sum_{i} \mathbf{x}_{i} \beta_{i} + \bm{\epsilon},
$$ {#eq:reg:model}

where $\mathbf{m}$ is a vector of S-MultiXcan gene $p$-values for a trait (with a $-log_{10}$ transformation); $\mathbf{s}$ is a binary indicator vector with $s_{\ell}=1$ for the top 1% of genes with the largest loadings for latent variable (LV) $\ell$ (from $\mathbf{Z}_{\ell}$) and zero otherwise; $\mathbf{x}_{i}$ is a gene property used as a covariate; $\beta$ are effect sizes (with $\beta_{0}$ as the intercept); and $\bm{\epsilon} \sim \mathrm{MVN}(0, \sigma^{2} \mathbf{R})$ is a vector of error terms with a multivariate normal distribution (MVN) where $\mathbf{R}$ is the matrix of gene correlations.

*equation_definition

The vector $\mathbf{m}$ is defined as follows:

\[
\mathbf{m} = \begin{bmatrix} m_{1} \\ m_{2} \\ \vdots \\ m_{n} \end{bmatrix}
\]

where $m_{i}$ represents the $p$-value for gene $i$.

*equation_definition

The binary indicator vector $\mathbf{s}$ is defined as:

\[
\mathbf{s} = \begin{bmatrix} s

The model tests the null hypothesis $\beta_{s} = 0$ against the one-sided hypothesis $\beta_{s} > 0$.
Therefore, $\beta_{s}$ reflects the difference in trait associations between genes that are part of latent variable (LV) $\ell$ and genes outside of it.
Following the MultiXcan Analysis of Gene-set Enrichment (MAGMA) framework, we used two gene properties as covariates:

*equation_definition*
1) *gene size*: The number of principal components (PCs) retained in S-MultiXcan.
2) *gene density*: The ratio of the number of PCs to the number of tissues available.

Since the error terms $\bm{\epsilon}$ could be correlated, we cannot assume they have independent normal distributions as in a standard linear regression model.
In the PrediXcan family of methods, the predicted expression of a pair of genes could be correlated if they share eQTLs or if these are in LD.
Therefore, we used a generalized least squares approach to account for these correlations.
The gene-gene correlation matrix $\mathbf{R}$ was approximated by computing the correlations between the model sum of squares (SSM) for each pair of genes under the null hypothesis of no association.
These correlations are derived from the individual-level MultiXcan model (Equation (@eq:multixcan)), where the predicted expression matrix $\mathbf{T}_{i} \in \mathbb{R}^{n \times p_i}$ of a gene $i$ across $p_i$ tissues is projected into its top $k_i$ principal components (PCs), resulting in matrix $\mathbf{P}_{i} \in \mathbb{R}^{n \times k_i}$.
From the MAGMA framework, we know that the SSM for each gene is proportional to $\mathbf{y}^{\top} \mathbf{P}_{i} \mathbf{P}_{i}^{\top} \mathbf{y$.
Under the null hypothesis of no association, the covariances between the SSM of genes $i$ and $j$ are therefore given by $2 \times \mathrm{Trace}(\mathbf{P}_{i}^{\top} \mathbf{P}_{j} \mathbf{P}_{j}^{\top} \mathbf{P}_{i})$.
The standard deviations of each SSM are given by $\sqrt{2 \times k_{i}} \times (n - 1)$.
Therefore, the correlation between the SSMs for genes $i$ and $j$ can be written as follows:

*equation_definition*

$$
\begin{split}
\mathbf{R}_{ij} & = \frac{2 \times \mathrm{Tr}(\mathbf{P}_{i}^{\top} \mathbf{P}_{j} \mathbf{P}_{j}^{\top} \mathbf{P}_{i})}{\sqrt{2 \times k_{i}} \times \sqrt{2 \times k_{j}} \times (n - 1)^2} \\
& = \frac{2 \times \mathrm{Tr}(Cor(\mathbf{P}_{i}, \mathbf{P}_{j}) \times Cor(\mathbf{P}_{j}, \mathbf{P}_{i}))}{\sqrt{2 \times k_{i}} \times \sqrt{2 \times k_{j}}},
\end{split}
$$ {#eq:reg:r}

where columns $\mathbf{P}$ are standardized, $\mathrm{Tr}$ is the trace of a matrix, and the cross-correlation matrix between PCs $Cor(\mathbf{P}_{i}, \mathbf{P}_{j}) \in \mathbb{R}^{k_i \times k_j}$ is given by:

*equation_definition*

$$
\begin{split}
Cor(\mathbf{P}_{i}, \mathbf{P}_{j}) & = Cor(\mathbf{T}_{i} \mathbf{V}_{i}^{\top} \mathrm{diag}(\lambda_i)^{-1/2}, \mathbf{T}_{j} \mathbf{V}_{j}^{\top} \mathrm{diag}(\lambda_j)^{-1/2}) \\
& = \mathrm{diag}(\lambda_i)^{-1/2} \mathbf{V}_{i} (\frac{\mathbf{T}_{i}^{\top} \mathbf{T}_{j}}{n-1}) \mathbf{V}_{j}^{\top} \mathrm{diag}(\lambda_j)^{-1/2},
\end{split}
$$ {#eq:reg:cor_pp}

where $\frac{\mathbf{T}_{i}^{\top} \mathbf{T}_{j}}{n-1} \in \mathbb{R}^{p_i \times p_j}$ is the cross-correlation matrix between the predicted expression levels of genes $i$ and $j$, and columns of $\mathbf{V}_{i}$ and scalars $\lambda_i$ are the eigenvectors and eigenvalues of $\mathbf{T}_{i}$, respectively.
S-MultiXcan keeps only the top eigenvectors using a condition number threshold of $\frac{\max(\lambda_i)}{\lambda_i} < 30$.
To estimate the correlation of predicted expression levels for genes $i$ in tissue $k$ and gene $j$ in tissue $l$, $(\mathbf{t}_k^i, \mathbf{t}_l^j)$ ($\mathbf{t}_k^i$ is the $k$th column of $\mathbf{T}_{i}$), we used:

*equation_definition*

$$
\begin{split}
\frac{(\mathbf{T}_{i}^{\top} \mathbf{T}_{j})_{kl}}{n-1} & = Cor(\mathbf{t}_k^i, \mathbf{t}_l^j) \\
& = \frac{ Cov(\mathbf{t}_k, \mathbf{t}_l) } { \sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)} } \\
& = \frac{ Cov(\sum_{a \in \mathrm{model}_k} w_a^k X_a, \sum_{b \in \mathrm{model}_l} w_b^l X_b) }  {\sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)} } \\
& = \frac{ \sum_{\substack{a \in \mathrm{model}_k \\ b \in \mathrm{model}_l}} w_a^k w_b^l Cov(X_a, X_b)} {\sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)} } \\
& = \frac{ \sum_{\substack{a \in \mathrm{model}_k \\ b \in \mathrm{model}_l}} w_a^k w_b^l \Gamma_{ab}} {\sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)} },
\end{split}
$$ {#eq:reg:corr_genes}

where $X_a$ is the genotype of SNP $a$, $w_a^k$ is the weight of SNP $a$ for gene expression prediction in the tissue model $k$, and $\Gamma = \widehat{\mathrm{var}}(\mathbf{X}) = (\mathbf{X} - \bar{\mathbf{X}})^{\top} (\mathbf{X} - \bar{\mathbf{X}}) / (n-1)$ is the genotype covariance matrix using GTEx v8 as the reference panel, which is the same used in all TWAS methods described here.
The variance of the predicted expression values of gene $i$ in tissue $k$ is estimated as:

*equation_definition*

$$
\begin{split}
\widehat{\mathrm{var}}(\mathbf{t}_k^i) & = (\mathbf{W}^k)^\top \Gamma^k \mathbf{W}^k \\
& = \sum_{\substack{a \in \mathrm{model}_k \\ b \in \mathrm{model}_k}} w_a^k w_b^k \Gamma_{ab}^k.
\end{split}
$$ {#eq:reg:var_gene}

Since we utilized the MultiXcan regression model (Equation (@eq:multixcan)), $\mathbf{R}$ serves as an approximation of gene correlations in S-MultiXcan.
As previously discussed, S-MultiXcan approximates the joint regression parameters in MultiXcan by utilizing the marginal regression estimates from S-PrediXcan in Equation (@eq:spredixcan), while incorporating some simplifying assumptions and different genotype covariance matrices.
This complicates the derivation of an S-MultiXcan-specific solution for computing $\mathbf{R}$.
To address this challenge, we opted to use a submatrix $\mathbf{R}_{\ell}$ that corresponds to genes belonging to latent variable $\ell$ exclusively (top 1% of genes), instead of the entire matrix $\mathbf{R}$.
This simplification is considered conservative since it focuses on correlations among the top genes only.
Our simulations (see Supplementary Note 1) demonstrate that the model is reasonably well-calibrated and capable of correcting for latent variables with closely related and highly correlated genes at the top (e.g., Figure @fig:reg:nulls:qqplot:lv234).
However, the simulations also identified 127 latent variables where the model exhibited poor calibration (e.g., Figure @fig:reg:nulls:qqplot:lv914).
This discrepancy can be attributed to challenges in accurately computing a gene correlation matrix, leading us to exclude these latent variables from our primary analyses. 

*equation_definition: $\mathbf{R}$ - gene correlation matrix
*equation_definition: $\mathbf{R}_{\ell}$ - submatrix corresponding to genes in latent variable $\ell$

In Equation (@eq:reg:corr_genes), we only considered tissue models present in S-PrediXcan results and SNPs present in GWAS used for the TWAS approaches for each gene to ensure accurate correlation estimates [@doi:10.1371/journal.pgen.1007889].

*equation_definition: Equation (@eq:reg:corr_genes): Correlation computation for genes using tissue models and SNPs.

Therefore, we calculated separate correlation matrices for PhenomeXcan and eMERGE datasets.
In PhenomeXcan, the majority of the GWAS data (4,049) were sourced from the UK Biobank using a consistent pipeline and SNP set, allowing for the use of a single correlation matrix.
For the remaining datasets, we utilized a single correlation matrix for each group of traits that shared the same or similar SNPs.

We ran our regression model for all 987 LVs across the 4,091 traits in PhenomeXcan.
For replication, we ran the model in the 309 phecodes in eMERGE.
We adjusted the $p$-values using the Benjamini-Hochberg procedure.


### LV-based drug repurposing approach {#sec:methods:drug}

For the drug-disease prediction, we developed an LV-based method within a drug repositioning framework previously utilized for psychiatry traits (Smith et al., 2018), in which individual genes associated with a trait are anticorrelated with drug expression profiles.
To evaluate our LV-based method, we compared it with the single-gene approach described in a study by Jones et al.
(2017).
In the single-gene method, we calculated a drug-disease score by multiplying each S-PrediXcan set of signed z-scores in tissue $t$, represented by $\mathbf{M}^t$, with another set of signed z-scores from transcriptional responses obtained from LINCS L1000 (Doe et al., 2017), denoted as $\mathbf{L}^{c \times m}$ (for c compounds). 

*equation_definition*
- $\mathbf{M}^t$: Matrix containing information on the association of predicted gene expression with disease risk in tissue t.
- $\mathbf{L}$: Matrix indicating whether a drug increases or decreases the expression of a gene.
- $\mathbf{D}^{t,k}$: Resulting matrix from the product of $\mathbf{M}^{t,k}$ and $\mathbf{L}^\top$, where k represents the number of most significant gene associations in $\mathbf{M}^t$ for each trait.

As per the suggestion in the study by Smith et al.
(2018), the value of k could be all genes or the top 50, 100, 250, and 500.
Subsequently, we calculated the average score ranks across all k and obtained $\mathbf{D}^t$.
Finally, for each drug-disease pair, we selected the maximum prediction score across all tissues: $\mathbf{D}_{ij} = \max \{ \mathbf{D}_{ij}^t \mid \forall t \}$.


The LV-based approach followed the same procedure as the previous method.
We projected the matrices $\mathbf{M}^{t}$ and $\mathbf{L}$ into the gene module latent space using Equation (@eq:proj), resulting in $\hat{\mathbf{M}}^t$ and $\hat{\mathbf{L}}^{l \times c}$, respectively.
Subsequently, we calculated $\mathbf{D}^{t,k}=-1 \cdot \hat{\mathbf{L}}^{\top} \hat{\mathbf{M}}^{t,k}$, where $k$ could represent all LVs or the top 5, 10, 25, and 50 (considering the significantly smaller number of LVs compared to genes). 

*equation_definition: $\mathbf{M}^{t}$ - Matrix representing gene expression patterns for treatment $t$
*equation_definition: $\mathbf{L}$ - Matrix representing gene co-expression patterns
*equation_definition: $\hat{\mathbf{M}}^t$ - Projected matrix of gene expression patterns for treatment $t


Since the gold standard of drug-disease medical indications is described with Disease Ontology IDs (DOID) [@doi:10.1093/nar/gky1032], we mapped PhenomeXcan traits to the Experimental Factor Ontology [@doi:10.1093/bioinformatics/btq099] using [@url:https://github.com/EBISPOT/EFO-UKB-mappings], and then to DOID.


### Consensus clustering of traits {#sec:methods:clustering}

We performed two preprocessing steps on the S-MultiXcan results before the cluster analysis.
First, we combined results in matrix $\mathbf{M}$ (with $p$-values converted to $z$-scores, as described before) for traits that mapped to the same Experimental Factor Ontology (EFO) term using the Stouffer's method: 

\begin{equation}
\sum w_i M_{ij} / \sqrt{\sum w_i^2}
\end{equation}

where $w_i$ is a weight based on the GWAS sample size for trait $i$, and $M_{ij}$ is the $z$-score for gene $j$.
Second, we divided all $z$-scores for each trait $i$ by their sum to reduce the effect of highly polygenic traits: 

\begin{equation}
M_{ij} / \sum M_{ij}
\end{equation}

Finally, we projected this data matrix using Equation (@eq:proj), obtaining matrix $\hat{\mathbf{M}}$ with $n$=3,752 traits and $l$=987 latent variables (LVs) as the input of our clustering pipeline.


A partitioning of the estimated matrix M with n traits into k clusters is represented as a label vector π ∈ ℕ^n.
Consensus clustering approaches consist of two steps: 1) the generation of an ensemble Π with r partitions of the dataset: Π={π₁, π₂, ..., πᵣ}, and 2) the combination of the ensemble into a consolidated solution defined as:

π* = arg max_{π̂} Q({|Lⁱ| φ(π̂_{Lⁱ}, πᵢ_{Lⁱ}) | i ∈ {1,...,r}}),   (1)

where Lⁱ is a set of data indices with known cluster labels for partition i, φ: ℕ^n × ℕ^n → ℝ is a function that measures the similarity between two partitions, and Q is a measure of central tendency, such as the mean or median.
We used the adjusted Rand index (ARI) (Hubert and Arabie, 1985) for φ and the median for Q.
To obtain π*, we define a consensus function Γ: ℕ^{n × r} → ℕ^n with Π as the input.
We used consensus functions based on the evidence accumulation clustering (EAC) paradigm (Fred and Jain, 2005), where Π is first transformed into a distance matrix Dᵢⱼ = dᵢⱼ / r, where dᵢⱼ is the number of times traits i and j were grouped in different clusters across all r partitions in Π.
Then, Γ can be any similarity-based clustering algorithm, which is applied on D to derive the final partition π*.


For the ensemble generation step, we utilized various algorithms to construct a highly diverse set of partitions (refer to Figure 1) as diversity is a crucial characteristic for ensembles (Smith et al., 2016; Johnson et al., 2011; Brown et al., 2014).
Three data representations were employed: the raw dataset, its projection onto the top 50 principal components, and the embedding acquired through UMAP (McInnes et al., 2018) using 50 components.
Five clustering algorithms were applied to each of these representations, covering a broad spectrum of assumptions regarding the data structure: k-means (Arthur & Vassilvitskii, 2007), spectral clustering (Ng et al., 2001), a Gaussian mixture model (GMM), hierarchical clustering, and DBSCAN (Ester et al., 1996). 

*equation_definition*
- k: Number of clusters
- n: Total number of data points
- ε: Maximum distance between two data points
- minPts: Minimum number of data points in a neighborhood

For k-means, spectral clustering, and GMM, a range of k values between 2 and √n ≈ 60 was specified, and five partitions were generated for each k using random seeds.
In hierarchical clustering, four partitions were generated for each k using common linkage criteria: ward, complete, average, and single.
Different ranges for parameters ε and minPts were combined for DBSCAN based on the methodology outlined in a previous study (Johnson et al., 2018).
Specifically, minPts values ranged from 2 to 125.
Plausible ranges of ε values were determined for each data representation by analyzing the distribution of the mean distance of the minPts-nearest neighbors across all data points.
To ensure an equitable representation of the DBSCAN algorithm in the ensemble, resampling of partitions generated by DBSCAN was performed, as some combinations of minPts and ε may not yield meaningful partitions.
This ensemble generation procedure resulted in a final ensemble of 4,428 partitions encompassing 3,752 traits.


Finally, spectral clustering was applied to matrix $\mathbf{D}$ to obtain the ultimate consensus partitions.
Matrix $\mathbf{D}$ was initially converted into a similarity matrix using a radial basis function (RBF) kernel $\mathrm{exp}(-\gamma \mathbf{D}^2)$ with four different empirically determined values for $\gamma$.
Subsequently, for each value of $k$ ranging from 2 to 60, four consensus partitions were derived, and the one maximizing Equation (@eq:consensus:obj_func) was selected.
These 59 solutions were further refined by retaining only those with an ensemble agreement exceeding the 75th percentile (refer to Figure @fig:sup:clustering:agreement), resulting in a total of 15 final consensus partitions as depicted in Figure @fig:clustering:tree.

*equation_definition

- $\mathbf{D}$: Matrix representing the gene expression patterns
- $\gamma$: Parameter in the RBF kernel function
- $k$: Number of clusters
- Equation (@eq:consensus:obj_func): Objective function for selecting the optimal consensus partition
- Figure @fig:sup:clustering:agreement: Visualization of ensemble agreement threshold
- Figure @fig:clustering:tree: Illustration of final consensus partitions

The input data in our clustering pipeline undergoes several linear and nonlinear transformations, including Principal Component Analysis (PCA), Uniform Manifold Approximation and Projection (UMAP), and the ensemble transformation using the Ensemble Agglomerating Clustering (EAC) paradigm (distance matrix $\mathbf{D}$).
Although consensus clustering has clear advantages for biological data (Wilkerson and Hayes, 2016), this set of data transformations complicates the interpretation of results.
To circumvent this, we used a supervised learning approach to detect which gene modules/Latent Variables (LVs) are the most important for each cluster of traits (Figure 1b).
Note that we did not use this supervised model for prediction but only to learn which features (LVs) were most discriminative for each cluster.

For this, we used the highest resolution partition ($k$=29, although any could be used) to train a decision tree model using each of the clusters as labels and the projected data $\hat{\mathbf{M}}$ as the training samples.
For each $k$, we built a set of binary labels with the current cluster's traits as the positive class and the rest of the traits as the negative class.
Then, we selected the LV in the root node of the trained model only if its threshold was positive and larger than one standard deviation.

Next, we removed this LV from $\hat{\mathbf{M}}$ (regardless of being previously selected or not) and trained the model again.
We repeated this procedure 20 times to extract the top 20 LVs that better discriminate traits in a cluster from the rest.

*equation_definition: PCA - Principal Component Analysis
*equation_definition: UMAP - Uniform Manifold Approximation and Projection
*equation_definition: EAC - Ensemble Agglomerating Clustering
*equation_definition: LVs - Latent Variables
*equation_definition: $\mathbf{D}$ - Distance matrix

In [Supplementary Note 2](#sm:clustering:null_sim), we performed several analyses under a null hypothesis of no structure in the data to verify that the clustering results detected by this pipeline were real.


### CRISPR-Cas9 screening {#sec:methods:crispr}

**Cell culture.** HepG2 cells were obtained from ATCC (ATCC® HB-8065™) and were cultured in Eagle's Minimum Essential Medium with L-Glutamine (EMEM, Cat.
112-018-101, Quality Biology) supplemented with 10% Fetal Bovine Serum (FBS, Gibco, Cat.16000-044) and 1% Pen/Strep (Gibco, Cat.15140-122).
The cells were incubated at 37°C in a humidity-controlled incubator with 5% CO2 and were maintained at a density not exceeding 80% confluency in Collagen-I coated flasks.

*equation_definition*
Equation (@id): EMEM = Eagle's Minimum Essential Medium
Equation (@id): FBS = Fetal Bovine Serum
Equation (@id): Pen/Strep = Penicillin/Streptomycin

*equation_definition*

**Genome-wide lentiviral pooled CRISPR-Cas9 library.** The Broad GPP genome-wide Human Brunello CRISPR knockout Pooled library, a 3rd lentiviral generation, was obtained from Addgene (Cat.
73179-LV) and provided by David Root and John Doench.
This library, utilized for HepG2 cell transduction, comprises 76,441 sgRNAs targeting 19,114 genes in the human genome, with an average of 4 sgRNAs per gene.
Each 20nt sgRNA cassette was inserted between the U6 promoter and gRNA scaffold in the lentiCRIS-PRv2 backbone.
Lentiviral vectors encoding Cas9 were employed for delivering the sgRNA cassette-containing plasmids into cells during replication.
Cells that were unsuccessfully transduced were eliminated through puromycin selection.

*equation_definition*

- sgRNA: single guide RNA
- Cas9: CRISPR-associated protein 9

**Lentiviral titer determination.** 

No-spin lentiviral transduction was utilized for the screen.
In a Collagen-I coated 6-well plate, approximately 2.5 million cells were seeded in each well in the presence of 8 µg/ml polybrene (Millipore Sigma, Cat.
TR-1003 G), and a different titrated virus volume (e.g., 0, 50, 100, 200, 250, and 400 µl) was assigned to each well.
EMEM complete media was added to make the final volume of 1.24 ml. 

16-18 hours post-transduction, virus/polybrene-containing media was removed from each well.
Cells were washed twice with 1x DPBS and replaced with fresh EMEM.
At 24 hours, cells in each well were trypsinized, diluted (e.g., 1:10), and seeded in pairs of wells of 6-well plates.
At 60 hours post-transduction, cell media in each well was replaced with fresh EMEM. 

2 µg/ml of puromycin (Gibco, Cat.
A1113803) was added to one well out of the pair.
2-5 days after puromycin selection, or the 0 virus well treated with puromycin had no survival of cells, cells in both wells with or without puromycin were collected and counted for viability. 

*equation_definition*
Percentage of Infection (PI%) = (number of cells with puromycin selection / number of cells without puromycin selection) x 100

By means of Poisson's distribution theory, when transduction efficiency (PI%) is between 30-50%, which corresponds to an MOI (Multiplicity of Infection) of ~0.35-0.70.
At MOI close to 0.3, around 25% of cells are infected, and the majority of those infected cells are predicted to have only one copy of the virus.
Therefore, a volume of virus (120 µl) yielding 30-40% of transduction efficiency was chosen for further large-scale viral transduction.

Lentiviral transduction in HepG2 cells using the Brunello CRISPR knockout pooled library aimed to achieve a coverage of at least 500 cells per single-guide RNA (sgRNA).
An MOI between 0.3-0.4 was selected to ensure that 95% of infected cells received only one viral particle per cell.
Approximately 200 million cells were used to initiate the screen.
The transduction process followed a similar protocol to previous methods.
Specifically, 2.5 million cells were seeded in each well of 14 6-well plates, supplemented with 8 µg/ml of polybrene.
Subsequently, 120 µl of the virus was added to each experimental well.
After 18 hours post-transduction, the virus/polybrene mix medium was removed, and cells from each well were collected, counted, and combined into T175 flasks.
At 60 hours post-transduction, 2 µg/ml of puromycin was introduced to each flask.
The mediums were refreshed every two days with fresh EMEM containing 2 µg/ml puromycin.
Seven days following puromycin selection, the cells were gathered, pooled, counted, and replated.

*equation_definition*
MOI = (Number of viral particles)/(Number of target cells)
*equation_definition*

Fluorescent dye staining was conducted on cells 9 days after puromycin selection, with the cells divided into two groups.
The Unsorted Control group consisted of 20-30 million cells, which were collected and centrifuged at 500 x g for 5 minutes at 4°C.
The resulting pellet was then stored at -80°C for subsequent genomic DNA isolation.
The remaining cells, approximately 200 million, were plated in 100mm dishes and stained with LipidSpotTM 488 fluorescent dye (Biotium, Cat.
70065-T).
To prepare the staining solution, LipidSpot 488 was diluted to a 1:100 ratio with DPBS, and 4ml of this solution was added to each dish.
The cells were then incubated at 37°C for 30 minutes.
Cell images were captured using a fluorescent microscope EVOS to detect the GFP signal (Figure 1). 

*equation_definition*

Equation (@id): Definition of terms in equation.

*equation_definition*

Fluorescence-activated cell sorting (FACS) was performed by immediately collecting cells into 50ml tubes and spinning them at 500 x g for 5 minutes at 4°C.
After a wash with DPBS, cell pellets were resuspended with FACS Sorting Buffer (1x DPBS without Ca2+/Mg2+, 2.5mM EDTA, 25mM HEPES, 1% BSA) that had been filter-sterilized and kept at 4°C.
The resuspension was done gently by pipetting to ensure single cells.
The cell solution then underwent filtration through a cell strainer (Falcon, Cat.
352235) and was kept on ice while being protected from light.
Cells were sorted using a FACSJazz machine with a 100um nozzle, with approximately 20% of each GFP-High and GFP-Low subset being collected into 15ml tubes.
Following sorting, cells were promptly spun down and the resulting pellets were stored at -80°C for subsequent genomic DNA isolation.

*equation_definition*

Equation (@id): FACS Sorting Buffer = 1x DPBS without Ca2+/Mg2+ + 2.5mM EDTA + 25mM HEPES + 1% BSA

*equation_definition*

Symbols:
- FACS: Fluorescence-activated cell sorting
- DPBS: Dulbecco's Phosphate-Buffered Saline
- EDTA: Ethylenediaminetetraacetic acid
- HEPES: 4-(2-hydroxyethyl)-1-piperazineethanesulfonic acid
- BSA: Bovine Serum Albumin
- GFP: Green Fluorescent Protein

**Genomic DNA isolation and verification.** Three conditions of Genomic DNA (Un-Sorted Control, lentiV2 GFP-High, and lentiV2 GFP-Low) were extracted using the QIAamp DNA Blood Mini Kit (Qiagen, Cat.51104), followed by UV Spectroscopy (Nanodrop) to assess the quality and quantity of the gDNA.
A total of 80-160 μg of gDNA was isolated for each condition.
The sgRNA cassette and lentiviral specific transgene in the isolated gDNA were verified through PCR (Figure 3).

*equation_definition*

Equation 1: \(E=mc^2\)

In Equation 1, \(E\) represents energy, \(m\) represents mass, and \(c\) represents the speed of light in a vacuum.

**Illumina libraries generation and sequencing.** 

The fragment containing the sgRNA cassette was amplified using P5/P7 primers, as indicated in a previous study [@pmid:26780180], and primer sequences were adapted from the Broad Institute protocol (Figure @fig:sup:crispr:table1).
The stagger sequence (0-8nt) was included in P5, and an 8bp uniquely barcoded sequence was included in P7.
Primers were synthesized through Integrated DNA Technologies (IDT), and each primer was PAGE purified. 

32 PCR reactions were set up for each condition.
Each 100 μl PCR reaction consisted of approximately 5 μg of gDNA, 5 μl of each 10 μM P5 and P7.
ExTaq DNA Polymerase (TaKaRa, Cat.
RR001A) was used to amplify the amplicon.
The PCR Thermal Cycler Parameters were set as follows: Initial at 95°C for 1 minute; followed by 24 cycles of Denaturation at 94°C for 30 seconds, Annealing at 52.5°C for 30 seconds, Extension at 72°C for 30 seconds, with a final Elongation at 72°C for 10 minutes.
PCR products of 285bp-293bp were expected (Figure @fig:sup:crispr:fig4 A). 

PCR products from the same condition were pooled and purified using SPRIselect beads (Beckman Coulter, Cat.
B23318).
The purified Illumina libraries were quantitated on Qubit, and the quality of the library was analyzed on a Bioanalyzer using a High Sensitivity DNA Chip.
A single approximate 285bp peak was expected (Figure @fig:sup:crispr:fig4 B). 

The final Illumina library samples were sequenced on a NovaSeq 6000.
Samples were pooled and loaded on an SP flow cell, along with a 20% PhiX control v3 library spike-in. 

*equation_definition*

- P5: Primer 5
- P7: Primer 7
- gDNA: genomic DNA
- PCR: Polymerase Chain Reaction
- PAGE: Polyacrylamide Gel Electrophoresis
- SPRI: Solid Phase Reversible Immobilization


## Data availability

All the main datasets generated in this study are available at [https://doi.org/10.5281/zenodo.8071382](https://doi.org/10.5281/zenodo.8071382) [@doi:10.5281/zenodo.8071382] and the GitHub repository [https://github.com/greenelab/phenoplier](https://github.com/greenelab/phenoplier).

The main input datasets used are TWAS from PhenomeXcan [@doi:10.1126/sciadv.aba2083] for 4,091 traits and from the Electronic Medical Records and Genomics (eMERGE) network phase III [@doi:10.1101/2021.10.21.21265225] for 309 traits; transcriptional responses to small molecule perturbations from LINCS L1000 [@doi:10.1016/j.cell.2017.10.049] that were further preprocessed and mapped to DrugBank IDs from [@doi:10.5281/zenodo.47223]; latent space/gene module models from MultiPLIER [@doi:10.1016/j.cels.2019.04.003].

The data utilized from PhenomeXcan, LINCS L1000, and MultiPLIER is publicly available.
All significant results reported for the eMERGE and Penn Medicine BioBank (PMBB) phenome-wide TWAS are contained in a previous study (Smith et al., 2021).
The individual-level PMBB raw datasets cannot be made publicly available due to institutional privacy policy.
Researchers interested in accessing the data are encouraged to contact Penn Medicine Biobank for requests of access to data.
The eMERGE network phase III data is available on dbGAP under the accession number phs001584.v2.p2.

*equation_definition: PhenomeXcan - Phenome-wide association study of gene expression data
*equation_definition: LINCS L1000 - Library of Integrated Network-based Cellular Signatures
*equation_definition: MultiPLIER - Multi-tissue Gene Regulatory Network for Gene Expression Imputation and Drug Repurposing Analysis
*equation_definition: TWAS - Transcriptome-wide association study
*equation_definition: dbGAP - Database of Genotypes and Phenotypes


## Code availability

The code necessary to reproduce all the analyses in this work is available at [https://doi.org/10.5281/zenodo.8071382](https://doi.org/10.5281/zenodo.8071382) [@doi:10.5281/zenodo.8071382] and the GitHub repository [https://github.com/greenelab/phenoplier](https://github.com/greenelab/phenoplier).

For the CRISPR screening, FlowJo v10.7 and FACS Jazz Software v1.1 were utilized.
Data analysis was conducted using Python 3.8 and R 3.6, employing various computational packages.
In Python, key packages included Jupyter Lab (2.2), pandas (1.1), matplotlib (3.3), seaborn (0.11), numpy (1.19), scipy (1.5), scikit-learn (0.23), and umap-learn (0.4).
In R, essential packages used were Bioconductor (3.10), clusterProfiler (3.14), clustree (0.4), and fgsea (1.17).
Additionally, custom scripts and notebooks were developed and made available under an open-source license.
Detailed documentation for each analysis step was provided.
A Docker image was also shared to ensure the same runtime environment, along with a demo for quick testing of the methods on actual data. 

*equation_definition: 
- CRISPR: Clustered Regularly Interspaced Short Palindromic Repeats
- FACS: Fluorescence-activated cell sorting
- Jupyter Lab: Interactive development environment for Jupyter notebooks
- Bioconductor: Open-source software for bioinformatics
- Docker: Platform for developing, shipping, and running applications in containers.
