## Methods {#sec:methods}

PhenoPLIER is a framework that combines different computational approaches to integrate gene-trait associations and drug-induced transcriptional responses with groups of functionally-related genes (referred to as gene modules or latent variables/LVs).
Gene-trait associations are computed using the PrediXcan family of methods, whereas latent variables are inferred by the MultiPLIER models applied on large gene expression compendia.
PhenoPLIER provides:

1) A regression model to compute an LV-trait association.
2) A consensus clustering approach applied to the latent space to learn shared and distinct transcriptomic properties between traits.
3) An interpretable, LV-based drug repurposing framework.

The details of these methods are provided below.

*equation_definition* LV: Latent Variable
*equation_definition* PrediXcan: A family of methods for computing gene-trait associations
*equation_definition* MultiPLIER: Models used to infer latent variables from gene expression data


### The PrediXcan family of methods for gene-based associations {#sec:methods:predixcan}

We utilized Summary-PrediXcan (S-PrediXcan) (Barbeira et al., 2018) and Summary-MultiXcan (S-MultiXcan) (Barbeira et al., 2019) as the gene-based statistical approaches, which are part of the PrediXcan family of methods (Barbeira et al., 2016).
These approaches are collectively known as TWAS (transcription-wide association studies).
S-PrediXcan, the summary-based version of PrediXcan, calculates the univariate association between a trait and a gene's predicted expression in a single tissue.
On the other hand, S-MultiXcan, the summary-based version of MultiXcan, calculates the joint association between a gene's predicted expression in all tissues and a trait.
Both S-PrediXcan and S-MultiXcan only require GWAS summary statistics, eliminating the need for individual-level genotype and phenotype data. 

*equation_definition
- S-PrediXcan: Univariate association between a trait and a gene's predicted expression in a single tissue.
- S-MultiXcan: Joint association between a gene's predicted expression in all tissues and a trait.

Here we provide a brief overview of the TWAS methods, essential for understanding our regression framework later (for more detailed information, refer to the cited articles).
In the context of this discussion, $\mathbf{y}$ represents a vector of traits for $n$ individuals that has been centered for convenience, eliminating the need for an intercept.
$\tilde{\mathbf{t}}_l = \sum_{a \in \mathrm{model}_l} w_{a}^{l} X_{a}$ denotes the predicted gene expression of a gene for all individuals in tissue $l$, where $X_a$ represents the genotype of SNP $a$ and $w_{a}$ is its weight in the tissue prediction model $l$.
Additionally, $\mathbf{t}_l$ refers to the standardized form of $\tilde{\mathbf{t}}_l$, with a mean of zero and a standard deviation of one.

*equation_definition*

- $\mathbf{y}$: Vector of traits for $n$ individuals
- $\tilde{\mathbf{t}}_l$: Predicted gene expression for all individuals in tissue $l$
- $X_a$: Genotype of SNP $a$
- $w_{a}$: Weight of SNP $a$ in tissue prediction model $l$
- $\mathbf{t}_l$: Standardized version of $\tilde{\mathbf{t}}_l$ with mean zero and standard deviation one.

S-PrediXcan (Barbeira et al., 2018) is the condensed version of PrediXcan (Gamazon et al., 2015).
PrediXcan represents the trait as a linear function of the gene's expression in a single tissue using the univariate model:

$$
\mathbf{y} = \mathbf{t}_l \gamma_l + \bm{\epsilon}_l,
$$ 

where $\gamma_l$ is the estimated effect size or regression coefficient, and $\bm{\epsilon}_l$ are the error terms with variance $\sigma_{\epsilon}^{2}$.
The significance of the association is evaluated by calculating the $z$-score $\hat{z}_{l}=\hat{\gamma}_l / \mathrm{se}(\hat{\gamma}_l)$ for a gene's tissue model $l$.
PrediXcan requires individual-level data to fit this model, while S-PrediXcan approximates PrediXcan $z$-scores using only GWAS summary statistics with the expression:

$$
\hat{z}_{l} \approx \sum_{a \in model_{l}} w_a^l \frac{\hat{\sigma}_a}{\hat{\sigma}_l} \frac{\hat{\beta}_a}{\mathrm{se}(\hat{\beta}_a)},
$$ 

where $\hat{\sigma}_a$ is the variance of SNP $a$, $\hat{\sigma}_l$ is the variance of the predicted expression of a gene in tissue $l$, and $\hat{\beta}_a$ is the estimated effect size of SNP $a$ from the GWAS.
In these Transcriptome-Wide Association Study (TWAS) methods, the genotype variances and covariances are estimated using the Genotype-Tissue Expression project (GTEx v8) (The GTEx Consortium, 2020) as the reference panel.
Since S-PrediXcan provides tissue-specific direction of effects (e.g., whether a higher or lower predicted expression of a gene confers more or less disease risk), we utilized the $z$-scores in our drug repurposing strategy (explained below).

S-MultiXcan is the summary version of MultiXcan, which is more powerful than PrediXcan in detecting gene-trait associations, although it does not provide the direction of effects.
The main output of MultiXcan is the p-value (obtained with an F-test) of the multiple tissue model:

$$
\begin{split}
\mathbf{y} & = \sum_{l=1}^{p} \mathbf{t}_l g_l + \mathbf{e} \\
& = \mathbf{T} \mathbf{g} + \mathbf{e},
\end{split}
$$

where $\mathbf{T}$ is a matrix with $p$ columns $\mathbf{t}_l$, $\hat{g}_l$ is the estimated effect size for the predicted gene expression in tissue $l$, and $\mathbf{e}$ are the error terms with variance $\sigma_{e}^{2}$.
Due to the high correlation between predicted expression values for a gene across different tissues, MultiXcan uses the principal components (PCs) of $\mathbf{T}$ to avoid collinearity issues.
S-MultiXcan derives the joint regression estimates (effect sizes and their variances) using the marginal estimates from S-PrediXcan.
Under the null hypothesis of no association, $\hat{\mathbf{g}}^{\top} \frac{\mathbf{T}^{\top}\mathbf{T}}{\sigma_{e}^{2}} \hat{\mathbf{g}} \sim \chi_{p}^{2}$, and therefore the significance of the association in S-MultiXcan is estimated with:

$$
\begin{split}
\frac{\hat{\mathbf{g}}^{\top} (\mathbf{T}^{\top}\mathbf{T}) \hat{\mathbf{g}}}{\sigma_{e}^{2}} & \approx \bm{\hat{\gamma}}^{\top} \frac{\sqrt{n-1}}{\sigma_{\epsilon}} \left(\frac{\mathbf{T}^{\top} \mathbf{T}}{n-1}\right)^{-1} \frac{\sqrt{n-1}}{\sigma_{\epsilon}} \bm{\hat{\gamma}} \\
& = \hat{\mathbf{z}}^{\top} Cor(\mathbf{T})^{-1} \hat{\mathbf{z}},
\end{split}
$$

where $\hat{\mathbf{z}}$ is a vector with $p$ z-scores for each tissue available for the gene, and $Cor(\mathbf{T})$ is the autocorrelation matrix of $\mathbf{T}$.
Since $\mathbf{T}^{\top}\mathbf{T}$ is singular for many genes, S-MultiXcan computes the pseudo-inverse $Cor(\mathbf{T})^{+}$ using the $k$ top PCs, and thus $\hat{\mathbf{z}}^{\top} Cor(\mathbf{T})^{+} \hat{\mathbf{z}} \sim \chi_k^2$.
To arrive at this expression, S-MultiXcan uses the conservative approximation $\sigma_{e}^{2} \approx \sigma_{\epsilon}^{2}$.
Another important point is that $Cor(\mathbf{T})$ is estimated using a global genotype covariance matrix, whereas marginal $\hat{z}_l$ are approximated using tissue-specific genotype covariances.
Although S-MultiXcan yields highly concordant estimates compared with MultiXcan, results are not perfectly correlated across genes.
These differences are important for our LV-based regression model when computing the gene-gene correlation matrix.
We used S-MultiXcan results for our LV-based regression model and our cluster analyses of traits.


### TWAS resources {#sec:methods:twas}

We used two large TWAS resources from different cohorts for discovery and replication, all obtained from European ancestries.
PhenomeXcan [@doi:10.1126/sciadv.aba2083], our discovery cohort, provides results on 4,091 traits across different categories.
Supplementary Data 1 has all the details about the included GWAS, sample size and disease/trait categories.
In PhenomeXcan, these publicly available GWAS summary statistics were used to compute
1) gene-based associations with the PrediXcan family of methods (described before), and
2) a posterior probability of colocalization between GWAS loci and *cis*-eQTL with fastENLOC [@doi:10.1126/sciadv.aba2083; @doi:10.1016/j.ajhg.2020.11.012].
We refer to the matrix of $z$-scores from S-PrediXcan (Equation (@eq:spredixcan)) across $q$ traits and $m$ genes in tissue $t$ as $\mathbf{M}^{t} \in \mathbb{R}^{q \times m}$.
As explained later, matrices $\mathbf{M}^{t}$ were used in our LV-based drug repurposing framework since they provide direction of effects.
The S-MultiXcan results (22,515 gene associations across 4,091 traits) were used in our LV-based regression framework and our cluster analyses of traits.
For the cluster analyses, we used the $p$-values converted to $z$-scores: $\mathbf{M}=\Phi^{-1}(1 - p/2)$, where $\Phi^{-1}$ is the probit function.
Higher $z$-scores correspond to stronger associations.

Our discovery cohort was eMERGE [@doi:10.1038/gim.2013.72], where the same TWAS methods were run on 309 phecodes [@doi:10.1101/2021.10.21.21265225] across different categories (more information about traits are available in [@doi:10.1101/2021.10.21.21265225]).
We used these results to replicate the associations found with our LV-based regression framework in PhenomeXcan.


### MultiPLIER and Pathway-level information extractor (PLIER) {#sec:methods:multiplier}

MultiPLIER (Gligorijevic et al., 2019) extracts patterns of co-expressed genes from recount2 (Collado-Torres et al., 2017), a large gene expression dataset that does not include GTEx samples.
The approach utilizes the pathway-level information extractor method (PLIER) (Mao et al., 2019), which employs unsupervised learning with prior knowledge (canonical pathways) to mitigate technical noise.
PLIER employs a matrix factorization approach that decomposes gene expression data into a set of latent variables (LVs), where each LV represents a gene module.
The MultiPLIER models reduced the dimensionality in recount2 to 987 LVs.

*equation_definition: LV - Latent Variable

Given a gene expression dataset $\mathbf{Y}^{m \times c}$ with $m$ genes and $c$ experimental conditions and a prior knowledge matrix $\mathbf{C} \in \{0,1\}^{m \times p}$ for $p$ MSigDB pathways (so that $\mathbf{C}_{ij} = 1$ if gene $i$ belongs to pathway $j$), PLIER finds $\mathbf{U}$, $\mathbf{Z}$, and $\mathbf{B}$ minimizing

$$
||\mathbf{Y} - \mathbf{Z}\mathbf{B}||^{2}_{F} + \lambda_1 ||\mathbf{Z} - \mathbf{C}\mathbf{U}||^{2}_{F} + \lambda_2 ||\mathbf{B}||^{2}_{F} + \lambda_3 ||\mathbf{U}||_{L^1}
$$ {#eq:met:plier_func}

subject to $\mathbf{U}>0, \mathbf{Z}>0$; $\mathbf{Z}^{m \times l}$ are the gene loadings with $l$ latent variables, $\mathbf{B}^{l \times c}$ is the latent space for $c$ conditions, $\mathbf{U}^{p \times l}$ specifies which of the $p$ prior-information pathways in $\mathbf{C}$ are represented for each latent variable, and $\lambda_i$ are different regularization parameters used in the training step.
$\mathbf{Z}$ is a low-dimensional representation of the gene space where each latent variable aligns as much as possible to prior knowledge, and it might represent either a known or novel gene module (i.e., a meaningful biological pattern) or noise.
[@doi:10.1016/j.cels.2015.12.004]

For our drug repurposing and cluster analyses, we utilized a model to project gene-trait associations (obtained from TWAS) and gene-drug associations (obtained from LINCS L1000) into a low-dimensional gene module space.
Specifically, TWAS associations $\mathbf{M}$ (derived from either S-PrediXcan or S-MultiXcan) were projected using:

*equation_definition*

$$
\hat{\mathbf{M}} = (\mathbf{Z}^{\top} \mathbf{Z} + \lambda_{2} \mathbf{I})^{-1} \mathbf{Z}^{\top} \mathbf{M},
$$ {#eq:proj}

Here, $\hat{\mathbf{M}}^{l \times q}$ represents a matrix where traits are depicted by gene modules instead of individual genes.
Later on, we applied the same methodology to project drug-induced transcriptional profiles from LINCS L1000, resulting in a representation of drugs based on gene modules.


### Regression model for LV-trait associations {#sec:methods:reg}

We adapted the gene-set analysis framework from MAGMA [@doi:10.1371/journal.pcbi.1004219] to TWAS.
We used a competitive test to predict gene-trait associations from TWAS using gene weights from an LV, testing whether top-weighted genes for an LV are more strongly associated with the phenotype than other genes with relatively small or zero weights.
Thus, we fit the model

$$
\mathbf{m}=\beta_{0} + \mathbf{s} \beta_{s} + \sum_{i} \mathbf{x}_{i} \beta_{i} + \bm{\epsilon},
$$ {#eq:reg:model}

where $\mathbf{m}$ is a vector of S-MultiXcan gene $p$-values for a trait (with a $-log_{10}$ transformation); $\mathbf{s}$ is a binary indicator vector with $s_{\ell}=1$ for the top 1% of genes with the largest loadings for latent variable (LV) $\ell$ (from $\mathbf{Z}_{\ell}$) and zero otherwise; $\mathbf{x}_{i}$ is a gene property used as a covariate; $\beta$ are effect sizes (with $\beta_{0}$ as the intercept); and $\bm{\epsilon} \sim \mathrm{MVN}(0, \sigma^{2} \mathbf{R})$ is a vector of error terms with a multivariate normal distribution (MVN) where $\mathbf{R}$ is the matrix of gene correlations.

*equation_definition

The vector $\mathbf{m}$ is defined as:

\begin{equation}
\mathbf{m} = (-\log_{10}(\text{gene}_1 \text{ p-value}), \ldots, -\log_{10}(\text{gene}_n \text{ p-value}))
\end{equation}

*equation_definition

The binary indicator vector $\mathbf{s}$ is defined as:

\begin{equation}
s_{\ell} = 
\begin{cases

The model tests the null hypothesis $\beta_{s} = 0$ against the one-sided hypothesis $\beta_{s} > 0$.
Therefore, $\beta_{s}$ reflects the difference in trait associations between genes that are part of latent variable (LV) $\ell$ and genes outside of it.
Following the Multi-marker Analysis of GenoMic Annotation (MAGMA) framework, we used two gene properties as covariates:

*equation_definition*
1) *Gene size* ($\text{gene size}$) - defined as the number of principal components (PCs) retained in S-MultiXcan.
*equation_definition*
2) *Gene density* ($\text{gene density}$) - defined as the ratio of the number of PCs to the number of tissues available.

Since the error terms $\bm{\epsilon}$ could be correlated, we cannot assume they have independent normal distributions as in a standard linear regression model.
In the PrediXcan family of methods, the predicted expression of a pair of genes could be correlated if they share eQTLs or if these are in LD (Zhu et al., 2019).
Therefore, we used a generalized least squares approach to account for these correlations.
The gene-gene correlation matrix $\mathbf{R}$ was approximated by computing the correlations between the model sum of squares (SSM) for each pair of genes under the null hypothesis of no association.
These correlations are derived from the individual-level MultiXcan model (Equation (1)), where the predicted expression matrix $\mathbf{T}_{i} \in \mathbb{R}^{n \times p_i}$ of a gene $i$ across $p_i$ tissues is projected into its top $k_i$ principal components (PCs), resulting in matrix $\mathbf{P}_{i} \in \mathbb{R}^{n \times k_i}$.
From the MAGMA framework, we know that the SSM for each gene is proportional to $\mathbf{y}^{\top} \mathbf{P}_{i} \mathbf{P}_{i}^{\top} \mathbf{y$.
Under the null hypothesis of no association, the covariances between the SSM of genes $i$ and $j$ are therefore given by $2 \times \mathrm{Trace}(\mathbf{P}_{i}^{\top} \mathbf{P}_{j} \mathbf{P}_{j}^{\top} \mathbf{P}_{i})$.
The standard deviations of each SSM are given by $\sqrt{2 \times k_{i}} \times (n - 1)$.
Therefore, the correlation between the SSMs for genes $i$ and $j$ can be written as follows:

$$
\begin{split}
\mathbf{R}_{ij} & = \frac{2 \times \mathrm{Tr}(\mathbf{P}_{i}^{\top} \mathbf{P}_{j} \mathbf{P}_{j}^{\top} \mathbf{P}_{i})}{\sqrt{2 \times k_{i}} \times \sqrt{2 \times k_{j}} \times (n - 1)^2} \\
& = \frac{2 \times \mathrm{Tr}(Cor(\mathbf{P}_{i}, \mathbf{P}_{j}) \times Cor(\mathbf{P}_{j}, \mathbf{P}_{i}))}{\sqrt{2 \times k_{i}} \times \sqrt{2 \times k_{j}}},
\end{split}
$$

where columns $\mathbf{P}$ are standardized, $\mathrm{Tr}$ is the trace of a matrix, and the cross-correlation matrix between PCs $Cor(\mathbf{P}_{i}, \mathbf{P}_{j}) \in \mathbb{R}^{k_i \times k_j}$ is given by

$$
\begin{split}
Cor(\mathbf{P}_{i}, \mathbf{P}_{j}) & = Cor(\mathbf{T}_{i} \mathbf{V}_{i}^{\top} \mathrm{diag}(\lambda_i)^{-1/2}, \mathbf{T}_{j} \mathbf{V}_{j}^{\top} \mathrm{diag}(\lambda_j)^{-1/2}) \\
& = \mathrm{diag}(\lambda_i)^{-1/2} \mathbf{V}_{i} (\frac{\mathbf{T}_{i}^{\top} \mathbf{T}_{j}}{n-1}) \mathbf{V}_{j}^{\top} \mathrm{diag}(\lambda_j)^{-1/2},
\end{split}
$$

where $\frac{\mathbf{T}_{i}^{\top} \mathbf{T}_{j}}{n-1} \in \mathbb{R}^{p_i \times p_j}$ is the cross-correlation matrix between the predicted expression levels of genes $i$ and $j$, and columns of $\mathbf{V}_{i}$ and scalars $\lambda_i$ are the eigenvectors and eigenvalues of $\mathbf{T}_{i}$, respectively.
S-MultiXcan keeps only the top eigenvectors using a condition number threshold of $\frac{\max(\lambda_i)}{\lambda_i} < 30$.
To estimate the correlation of predicted expression levels for genes $i$ in tissue $k$ and gene $j$ in tissue $l$, $(\mathbf{t}_k^i, \mathbf{t}_l^j)$ ($\mathbf{t}_k^i$ is the $k$th column of $\mathbf{T}_{i}$), we used (Gamazon et al., 2019):

$$
\begin{split}
\frac{(\mathbf{T}_{i}^{\top} \mathbf{T}_{j})_{kl}}{n-1} & = Cor(\mathbf{t}_k^i, \mathbf{t}_l^j) \\
& = \frac{ Cov(\mathbf{t}_k, \mathbf{t}_l) } { \sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)} } \\
& = \frac{ Cov(\sum_{a \in \mathrm{model}_k} w_a^k X_a, \sum_{b \in \mathrm{model}_l} w_b^l X_b) }  {\sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)} } \\
& = \frac{ \sum_{\substack{a \in \mathrm{model}_k \\ b \in \mathrm{model}_l}} w_a^k w_b^l Cov(X_a, X_b)} {\sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)} } \\
& = \frac{ \sum_{\substack{a \in \mathrm{model}_k \\ b \in \mathrm{model}_l}} w_a^k w_b^l \Gamma_{ab}} {\sqrt{\widehat{\mathrm{var}}(\mathbf{t}_k) \widehat{\mathrm{var}}(\mathbf{t}_l)} },
\end{split}
$$

where $X_a$ is the genotype of SNP $a$, $w_a^k$ is the weight of SNP $a$ for gene expression prediction in the tissue model $k$, and $\Gamma = \widehat{\mathrm{var}}(\mathbf{X}) = (\mathbf{X} - \bar{\mathbf{X}})^{\top} (\mathbf{X} - \bar{\mathbf{X}}) / (n-1)$ is the genotype covariance matrix using GTEx v8 as the reference panel, which is the same used in all TWAS methods described here.
The variance of the predicted expression values of gene $i$ in tissue $k$ is estimated as (Barbeira et al., 2019):

$$
\begin{split}
\widehat{\mathrm{var}}(\mathbf{t}_k^i) & = (\mathbf{W}^k)^\top \Gamma^k \mathbf{W}^k \\
& = \sum_{\substack{a \in \mathrm{model}_k \\ b \in \mathrm{model}_k}} w_a^k w_b^k \Gamma_{ab}^k.
\end{split}
$$

Note that, since we used the MultiXcan regression model:

\[
\text{Equation (@eq:multixcan)}
\]

$\mathbf{R}$ is only an approximation of gene correlations in S-MultiXcan.
As explained before, S-MultiXcan approximates the joint regression parameters in MultiXcan using the marginal regression estimates from S-PrediXcan:

\[
\text{Equation (@eq:spredixcan)}
\]

with some simplifying assumptions and different genotype covariance matrices.
This complicates the derivation of an S-MultiXcan-specific solution to compute $\mathbf{R}$.
To account for this, we used a submatrix $\mathbf{R}_{\ell}$ corresponding to genes that are part of LV $\ell$ only (top 1% of genes) instead of the entire matrix $\mathbf{R}$.
This simplification is conservative since correlations are accounted for top genes only.
Our simulations (see Supplementary Note 1) show that the model is approximately well-calibrated and can correct for LVs with adjacent and highly correlated genes at the top (e.g., Figure @fig:reg:nulls:qqplot:lv234).
The simulation also identified 127 LVs in which the model was not well-calibrated (e.g., Figure @fig:reg:nulls:qqplot:lv914).
As this can be attributed to limitations in accurately computing a gene correlation matrix, we excluded these LVs from our main analyses.

In Equation (@eq:reg:corr_genes), for each gene, we only considered tissue models present in S-PrediXcan results, as well as SNPs present in GWAS used as input for the TWAS approaches.
This is necessary to obtain more accurate correlation estimates (Gamazon et al., 2019).
Therefore, we computed different correlation matrices for PhenomeXcan and eMERGE.

*equation_definition: Correlation matrix computation for PhenomeXcan
In PhenomeXcan, most of the GWAS (4,049) were obtained from the UK Biobank using the same pipeline and including the same set of SNPs, so a single correlation matrix was used for this set.
For the rest, we used a single correlation matrix for each group of traits that shared the same or most of the SNPs.

We ran our regression model for all 987 LVs across the 4,091 traits in PhenomeXcan.
For replication, we ran the model in the 309 phecodes in eMERGE.
We adjusted the $p$-values using the Benjamini-Hochberg procedure.


### LV-based drug repurposing approach {#sec:methods:drug}

For the drug-disease prediction, we developed an LV-based method inspired by a drug repositioning framework previously utilized for psychiatry traits (Gandal et al., 2018), where individual genes associated with a trait are anticorrelated with drug expression profiles.
Our LV-based method was compared with the single-gene approach described in a study by Subramanian et al.
(2017).
In the single-gene method, we calculated a drug-disease score by multiplying each set of signed z-scores in tissue $t$, $\mathbf{M}^t$, by another set of signed z-scores from transcriptional responses obtained from LINCS L1000 (Subramanian et al., 2017), denoted as $\mathbf{L}^{c \times m}$ (for $c$ compounds).
Here, $\mathbf{M}^t$ indicates whether a higher or lower predicted expression of a gene is linked to disease risk, while $\mathbf{L}$ shows whether a drug increases or decreases the expression of a gene.
These matrices can be multiplied to calculate a score for a drug-disease pair.
The resulting product is $\mathbf{D}^{t,k}=-1 \cdot \mathbf{M}^{t,k} \mathbf{L}^\top$, where $k$ represents the number of most significant gene associations in $\mathbf{M}^t$ for each trait.
As proposed by Gandal et al.
(2018), $k$ could encompass all genes or the top 50, 100, 250, and 500; subsequently, we averaged score ranks across all $k$ to obtain $\mathbf{D}^t$.
Finally, for each drug-disease pair, we selected the maximum prediction score across all tissues: $\mathbf{D}_{ij} = \max \{ \mathbf{D}_{ij}^t \mid \forall t \}$.


The same procedure was utilized for the Latent Variable (LV)-based approach, in which we projected matrix $\mathbf{M}^{t}$ and matrix $\mathbf{L}$ into the gene module latent space using Equation (@eq:proj).
This projection resulted in estimated matrix $\hat{\mathbf{M}}^t$ and estimated matrix $\hat{\mathbf{L}}^{l \times c}$, respectively.
Subsequently, matrix $\mathbf{D}^{t,k}=-1 \cdot \hat{\mathbf{L}}^{\top} \hat{\mathbf{M}}^{t,k}$ was calculated, where in this scenario, $k$ could represent all Latent Variables (LVs) or the top 5, 10, 25, and 50 (given that the number of LVs is approximately an order of magnitude less than the number of genes).

*equation_definition

- $\mathbf{M}^{t}$: Matrix representing gene expression patterns for trait $t$.
- $\mathbf{L}$: Matrix representing gene expression patterns for latent variables.
-


Since the gold standard of drug-disease medical indications is described with Disease Ontology IDs (DOID) [@doi:10.1093/nar/gky1032], we mapped PhenomeXcan traits to the Experimental Factor Ontology [@doi:10.1093/bioinformatics/btq099] using [@url:https://github.com/EBISPOT/EFO-UKB-mappings], and then to DOID.


### Consensus clustering of traits {#sec:methods:clustering}

We performed two preprocessing steps on the S-MultiXcan results before the cluster analysis.
First, we combined results in matrix $\mathbf{M}$ (with $p$-values converted to $z$-scores, as described before) for traits that mapped to the same Experimental Factor Ontology (EFO) term using the Stouffer's method: 

\begin{equation}
\sum w_i M_{ij} / \sqrt{\sum w_i^2}
\end{equation}

where $w_i$ is a weight based on the GWAS sample size for trait $i$, and $M_{ij}$ is the $z$-score for gene $j$.
Second, we divided all $z$-scores for each trait $i$ by their sum to reduce the effect of highly polygenic traits: 

\begin{equation}
M_{ij} / \sum M_{ij}
\end{equation}

Finally, we projected this data matrix using Equation (1), obtaining matrix $\hat{\mathbf{M}}$ with $n$=3,752 traits and $l$=987 latent variables (LVs) as the input of our clustering pipeline.


A partitioning of the estimated matrix M with n traits into k clusters is represented as a label vector π ∈ ℕn.
Consensus clustering approaches consist of two steps: 1) the generation of an ensemble Π with r partitions of the dataset: Π={π1, π2, ..., πr}, and 2) the combination of the ensemble into a consolidated solution defined as:

π* = arg maxπ Q({ |ℒi| φ(π̂ℒi, πiℒi) | i ∈ {1,...,r} }), {#eq:consensus:obj_func}

where ℒi is a set of data indices with known cluster labels for partition i, φ: ℕn × ℕn → ℝ is a function that measures the similarity between two partitions, and Q is a measure of central tendency, such as the mean or median.
We used the adjusted Rand index (ARI) (Hubert and Arabie, 1985) for φ and the median for Q.
To obtain π*, we define a consensus function Γ: ℕn×r → ℕn with Π as the input.
We used consensus functions based on the evidence accumulation clustering (EAC) paradigm (Fred and Jain, 2005), where Π is first transformed into a distance matrix Dij = dij / r, where dij is the number of times traits i and j were grouped in different clusters across all r partitions in Π.
Then, Γ can be any similarity-based clustering algorithm, which is applied to D to derive the final partition π*.


For the ensemble generation step, we utilized various algorithms to construct a highly diverse set of partitions, as illustrated in Figure 1, as diversity is a crucial characteristic for ensembles (Liu et al., 2016; Shi et al., 2011; Zhang et al., 2014).
Three data representations were considered: the raw dataset, its projection onto the top 50 principal components, and the embedding acquired through UMAP (McInnes et al., 2018) with 50 components.
Each of these representations underwent clustering by five algorithms that encompass a broad spectrum of assumptions regarding data structure: k-means (Arthur and Vassilvitskii, 2007), spectral clustering (Ng et al., 2001), a Gaussian mixture model (GMM), hierarchical clustering, and DBSCAN (Ester et al., 1996). 

For k-means, spectral clustering, and GMM, a range of k values between 2 and approximately 60 (calculated as the square root of n) were specified, with five partitions generated for each k using random seeds.
In the case of hierarchical clustering, four partitions were generated for each k using common linkage criteria: ward, complete, average, and single.
For DBSCAN, various ranges of parameters epsilon (the maximum distance between two data points to be considered part of the same neighborhood) and minPts (the minimum number of data points in a neighborhood for a data point to be considered a core point) were combined, following the approach outlined in (Tian et al., 2018).
Specifically, minPts values from 2 to 125 were utilized. 

For each data representation (raw, PCA, and UMAP), a plausible range of epsilon values was determined by examining the distribution of the mean distance of the minPts-nearest neighbors across all data points.
To address scenarios where certain combinations of minPts and epsilon did not yield meaningful partitions (e.g., detecting all points as noisy or identifying only one cluster), resampling of partitions generated by DBSCAN was performed to ensure equal representation of this algorithm in the ensemble.
Overall, this process resulted in a final ensemble comprising 4,428 partitions of 3,752 traits. 

*equation_definition: minPts - minimum number of data points in a neighborhood for a data point to be considered a core point
*equation_definition: epsilon - maximum distance between two data points to be considered part of the same neighborhood


Finally, spectral clustering was performed on matrix $\mathbf{D}$ to obtain the final consensus partitions.
Matrix $\mathbf{D}$ was initially converted into a similarity matrix using a Radial Basis Function (RBF) kernel $\mathrm{exp}(-\gamma \mathbf{D}^2)$ with four different empirically determined values for $\gamma$.
Subsequently, for each value of $k$ ranging from 2 to 60, four consensus partitions were derived, and the one maximizing Equation (@eq:consensus:obj_func) was selected.
A further refinement was applied by retaining only the solutions with ensemble agreement exceeding the 75th percentile (refer to Figure @fig:sup:clustering:agreement), resulting in a total of 15 final consensus partitions as illustrated in Figure @fig:clustering:tree.

*equation_definition: $\mathrm{exp}(-\gamma \mathbf{D}^2)$ - Radial Basis Function (RBF) kernel transformation.
*equation_definition: Equation (@eq:consensus:obj_func) - Objective function for maximizing consensus partitions.

The input data in our clustering pipeline undergoes several linear and nonlinear transformations, including Principal Component Analysis (PCA), Uniform Manifold Approximation and Projection (UMAP), and the ensemble transformation using the Ensemble Agglomerative Clustering (EAC) paradigm (distance matrix $\mathbf{D}$).

*equation_definition*

Although consensus clustering has clear advantages for biological data [@pmid:27303057], this set of data transformations complicates the interpretation of results.
To circumvent this, we used a supervised learning approach to detect which gene modules/Latent Variables (LVs) are the most important for each cluster of traits (Figure {@fig:clustering:design}b).
Note that we did not use this supervised model for prediction but only to learn which features (LVs) were most discriminative for each cluster.

For this, we used the highest resolution partition ($k$=29, although any could be used) to train a decision tree model using each of the clusters as labels and the projected data $\hat{\mathbf{M}}$ as the training samples.
For each $k$, we built a set of binary labels with the current cluster's traits as the positive class and the rest of the traits as the negative class.
Then, we selected the LV in the root node of the trained model only if its threshold was positive and larger than one standard deviation.

Next, we removed this LV from $\hat{\mathbf{M}}$ (regardless of being previously selected or not) and trained the model again.
We repeated this procedure 20 times to extract the top 20 LVs that better discriminate traits in a cluster from the rest.

In [Supplementary Note 2](#sm:clustering:null_sim), we performed several analyses under a null hypothesis of no structure in the data to verify that the clustering results detected by this pipeline were real.


### CRISPR-Cas9 screening {#sec:methods:crispr}

**Cell culture.** HepG2 cells were obtained from ATCC (ATCC® HB-8065™) and maintained in Eagle's Minimum Essential Medium with L-Glutamine (EMEM, Cat.
112-018-101, Quality Biology) supplemented with 10% Fetal Bovine Serum (FBS, Gibco, Cat.16000-044) and 1% Pen/Strep (Gibco, Cat.15140-122).
The cells were incubated at 37°C in a humidity-controlled incubator with 5% CO2 and were kept at a density not exceeding 80% confluency in Collagen-I coated flasks.

*equation_definition*
Equation (@id): EMEM = Eagle's Minimum Essential Medium
Equation (@id): FBS = Fetal Bovine Serum
Equation (@id): Pen/Strep = Penicillin/Streptomycin

*equation_definition*

**Genome-wide lentiviral pooled CRISPR-Cas9 library.** The 3rd generation lentiviral library, known as Broad GPP genome-wide Human Brunello CRISPR knockout Pooled library, was obtained from Addgene (Cat.
73179-LV) courtesy of David Root and John Doench.
This library, utilized for transduction of HepG2 cells, comprises 76,441 single-guide RNAs (sgRNAs) targeting 19,114 genes in the human genome, with an average of 4 sgRNAs per gene.
Each sgRNA, consisting of a 20-nucleotide sequence, was inserted into the lentiCRIS-PRv2 backbone between the U6 promoter and the guide RNA (gRNA) scaffold.
The lentiviral vectors encoding Cas9 were employed for delivery of the sgRNA cassette-containing plasmids into cells during replication.
Cells that were not successfully transduced were eliminated through puromycin selection.

*equation_definition*

Equation (@id): sgRNA = single-guide RNA

Equation (@id): U6 = RNA polymerase III promoter

Equation (@id): gRNA = guide RNA

**Lentiviral titer determination.** No-spin lentiviral transduction was utilized for the screen.
Approximately 2.5 million cells were seeded in each well of a Collagen-I coated 6-well plate in the presence of 8 μg/ml polybrene (Millipore Sigma, Cat.
TR-1003 G), with a different titrated virus volume (e.g., 0, 50, 100, 200, 250, and 400 μl) assigned to each well.
EMEM complete media was added to achieve a final volume of 1.24 ml.
Sixteen to eighteen hours post-transduction, the virus/polybrene-containing media was removed from each well.
Cells were washed twice with 1x DPBS and fresh EMEM was added.
At 24 hours, cells in each well were trypsinized, diluted (e.g., 1:10), and seeded in pairs of wells in a 6-well plate.
At 60 hours post-transduction, the cell media in each well was replaced with fresh EMEM.
Two micrograms per milliliter of puromycin (Gibco, Cat.
A1113803) was added to one well out of the pair.
Two to five days after puromycin selection, or if the well with 0 virus treated with puromycin showed no cell survival, cells from both wells with and without puromycin were collected and counted for viability.
The Percentage of Infection (PI%) was calculated by comparing the cell numbers with and without puromycin selection within each pair.

*equation_definition*
PI% = (Number of cells with puromycin selection / Number of cells without puromycin selection) x 100

By utilizing Poisson's distribution theory, a transduction efficiency (PI%) between 30-50% corresponds to a Multiplicity of Infection (MOI) of approximately 0.35-0.70.
At an MOI close to 0.3, around 25% of cells are infected, and the majority of these infected cells are predicted to have only one copy of the virus.
Therefore, a virus volume of 120 μl, which yields a transduction efficiency of 30-40%, was selected for further large-scale viral transduction.

Lentiviral transduction in HepG2 cells was performed using the Brunello CRISPR knockout pooled library to ensure a coverage of at least 500 cells per single guide RNA (sgRNA).
An MOI between 0.3-0.4 was maintained to guarantee that 95% of infected cells received only one viral particle per cell.
Approximately 200 million cells were used to initiate the screen.
The transduction process mirrored the method previously described.
Specifically, 2.5 million cells were plated in each well of 14 6-well plates, supplemented with 8 μg/ml of polybrene.
Subsequently, 120 μl of the virus was added to each experimental well.
After 18 hours post-transduction, the virus/polybrene mix was removed, and cells from each well were harvested, counted, and combined into T175 flasks.
At 60 hours post-transduction, 2 μg/ml of puromycin was introduced to each flask.
The culture medium was refreshed every two days with fresh EMEM, containing 2 μg/ml puromycin.
Following seven days of puromycin selection, cells were gathered, pooled, counted, and replated.

*equation_definition*
MOI: Multiplicity of Infection
sgRNA: Single Guide RNA
EMEM: Eagle's Minimum Essential Medium

**Fluorescent dye staining.** Nine days after puromycin selection, cells were divided into two groups.
Twenty to thirty million cells were collected as the Unsorted Control.
The cell pellet was centrifuged at 500 x g for 5 minutes at 4°C.
The dry pellet was stored at -80°C for subsequent genomic DNA isolation.
The remaining cells (approximately 200 million) were plated in 100 mm dishes and stained with a fluorescent dye (LipidSpotTM 488, Biotium, Cat.
70065-T).
Briefly, LipidSpot 488 was diluted to 1:100 with DPBS.
Four milliliters of staining solution were added to each dish and incubated at 37°C for 30 minutes.
Cell images were captured using a fluorescent microscope EVOS to detect the GFP signal (Figure 1). 

*equation_definition*
- GFP: Green fluorescent protein

[Figure 1: Insert caption for the figure here.]

**Fluorescence-activated cell sorting (FACS).** Cells were collected into 50ml tubes and spun at 500 x g for 5 minutes at 4°C.
After a wash with DPBS, cell pellets were resuspended in FACS Sorting Buffer (1x DPBS without Ca2+/Mg2+, 2.5mM EDTA, 25mM HEPES, 1% BSA) that had been filter sterilized and kept at 4°C.
The cells were gently pipetted to ensure single-cell suspension, filtered through a cell strainer (Falcon, Cat.
352235), and kept on ice in the dark.
The collected cells were sorted using FACSJazz with a 100um nozzle, with approximately 20% of both GFP-High and GFP-Low cells being collected into 15ml tubes.
After sorting, the cells were spun down and the pellets stored at -80°C for subsequent genomic DNA isolation.

*equation_definition*

Equation (@id): FACS Sorting Buffer = 1x DPBS without Ca2+/Mg2+, 2.5mM EDTA, 25mM HEPES, 1% BSA

*equation_definition*

Symbols:
- DPBS: Dulbecco's Phosphate-Buffered Saline
- EDTA: Ethylenediaminetetraacetic acid
- HEPES: 4-(2-hydroxyethyl)-1-piperazineethanesulfonic acid
- BSA: Bovine Serum Albumin
- GFP: Green Fluorescent Protein

**Genomic DNA isolation and verification.** Genomic DNA (gDNA) from three conditions (Un-Sorted Control, lentiV2 GFP-High, and lentiV2 GFP-Low) was extracted using the QIAamp DNA Blood Mini Kit (Qiagen, Cat.51104) and analyzed for quality and quantity using UV Spectroscopy (Nanodrop).
A total of 80-160 μg of gDNA was isolated for each condition.
The presence of the sgRNA cassette and lentiviral specific transgene in the isolated gDNA was confirmed through PCR (Figure 3).

*equation_definition*

- gDNA: Genomic DNA
- sgRNA: Single-guide RNA
- PCR: Polymerase Chain Reaction

*equation_definition*

**Illumina libraries generation and sequencing.** The fragment containing the sgRNA cassette was amplified using P5/P7 primers, as described in previous work by Smith et al.
(2016), and the primer sequences were adapted from the Broad Institute protocol (Figure 1).
A stagger sequence of 0-8 nucleotides was included in P5, and an 8-base pair uniquely barcoded sequence was included in P7.
The primers were synthesized by Integrated DNA Technologies (IDT) and each primer was purified using polyacrylamide gel electrophoresis (PAGE).

32 PCR reactions were set up for each experimental condition.
Each 100 μL PCR reaction contained approximately 5 μg of genomic DNA, 5 μL of each 10 μM P5 and P7 primers.
ExTaq DNA Polymerase (TaKaRa) was used for amplification.
The thermal cycler parameters were set as follows: initial denaturation at 95°C for 1 minute, followed by 24 cycles of denaturation at 94°C for 30 seconds, annealing at 52.5°C for 30 seconds, and extension at 72°C for 30 seconds, with a final elongation step at 72°C for 10 minutes.
The expected size range of the PCR products was between 285-293 base pairs (Figure 2A).

After amplification, PCR products from the same condition were pooled and purified using SPRIselect beads (Beckman Coulter).
The purified Illumina libraries were quantified using a Qubit fluorometer, and the quality of the library was assessed on a Bioanalyzer using a High Sensitivity DNA Chip.
A single peak at approximately 285 base pairs was expected in the library samples (Figure 2B).
The final Illumina library samples were sequenced on a NovaSeq 6000 platform.
The samples were pooled and loaded onto an SP flow cell, along with a 20% PhiX control v3 library spike-in. 

*equation_definition: 
- P5: Primer 5
- P7: Primer 7


## Data availability

All the main datasets generated in this study are available at [https://doi.org/10.5281/zenodo.8071382](https://doi.org/10.5281/zenodo.8071382) [@doi:10.5281/zenodo.8071382] and the GitHub repository [https://github.com/greenelab/phenoplier](https://github.com/greenelab/phenoplier).

The main input datasets used are TWAS from PhenomeXcan [@doi:10.1126/sciadv.aba2083] for 4,091 traits and from the Electronic Medical Records and Genomics (eMERGE) network phase III [@doi:10.1101/2021.10.21.21265225] for 309 traits; transcriptional responses to small molecule perturbations from LINCS L1000 [@doi:10.1016/j.cell.2017.10.049] that were further preprocessed and mapped to DrugBank IDs from [@doi:10.5281/zenodo.47223]; latent space/gene module models from MultiPLIER [@doi:10.1016/j.cels.2019.04.003].

The data utilized from PhenomeXcan, LINCS L1000, and MultiPLIER are openly accessible.
All significant findings reported for the eMERGE and Penn Medicine BioBank (PMBB) phenome-wide TWAS are documented in a preprint by Smith et al.
(2021).
The raw individual-level datasets from PMBB cannot be publicly released due to institutional privacy regulations.
Researchers seeking access to this data should reach out to the Penn Medicine Biobank at [https://pmbb.med.upenn.edu/pmbb/](https://pmbb.med.upenn.edu/pmbb/).
The eMERGE network phase III data can be found on dbGAP under Accession phs001584.v2.p2. 

*equation_definition: TWAS - Transcriptome-wide association study
*equation_definition: dbGAP - database of Genotypes and Phenotypes


## Code availability

The code necessary to reproduce all the analyses in this work is available at [https://doi.org/10.5281/zenodo.8071382](https://doi.org/10.5281/zenodo.8071382) [@doi:10.5281/zenodo.8071382] and the GitHub repository [https://github.com/greenelab/phenoplier](https://github.com/greenelab/phenoplier).

For the CRISPR screening, we utilized FlowJo v10.7 and FACS Jazz Software v1.1.
Data analysis was conducted using Python 3.8 and R 3.6, incorporating various computational packages.
In Python, we employed Jupyter Lab (2.2), pandas (1.1), matplotlib (3.3), seaborn (0.11), numpy (1.19), scipy (1.5), scikit-learn (0.23), and umap-learn (0.4).
In R, we utilized Bioconductor (3.10), clusterProfiler (3.14), clustree (0.4), and fgsea (1.17).
Additionally, we developed scripts and notebooks, which are openly available under a permissive license.
Detailed documentation of the analysis procedures was provided, along with a Docker image for replicating the runtime environment and a demo for testing the methods on authentic data.

*equation_definition*
Equation (@id): Mathematical equation representing a specific calculation or model used in the study.

*equation_definition*
Symbols:
- Python: Programming language used for data analysis and visualization.
- R: Statistical programming language for data analysis and visualization.
- CRISPR: Gene-editing technology used in the screening process.
- FACS: Fluorescence-activated cell sorting for analyzing and sorting cells based on their properties.
